
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../../RNAseq/rnaseq-nfcore/">
      
      
        <link rel="next" href="../../LCMS/lcms-analysis/">
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.16">
    
    
      
        <title>DADA2 pipeline - Mucosal Immunology Bioinformatics Hub</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.7e37652d.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheet/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      
  


  
  

<script id="__analytics">function __md_analytics(){function e(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],e("js",new Date),e("config","G-7HN0DB92MT"),document.addEventListener("DOMContentLoaded",(function(){document.forms.search&&document.forms.search.query.addEventListener("blur",(function(){this.value&&e("event","search",{search_term:this.value})}));document$.subscribe((function(){var t=document.forms.feedback;if(void 0!==t)for(var a of t.querySelectorAll("[type=submit]"))a.addEventListener("click",(function(a){a.preventDefault();var n=document.location.pathname,d=this.getAttribute("data-md-value");e("event","feedback",{page:n,data:d}),t.firstElementChild.disabled=!0;var r=t.querySelector(".md-feedback__note [data-md-value='"+d+"']");r&&(r.hidden=!1)})),t.hidden=!1})),location$.subscribe((function(t){e("config","G-7HN0DB92MT",{page_path:t.pathname})}))}));var t=document.createElement("script");t.async=!0,t.src="https://www.googletagmanager.com/gtag/js?id=G-7HN0DB92MT",document.getElementById("__analytics").insertAdjacentElement("afterEnd",t)}</script>
  
    <script>"undefined"!=typeof __md_analytics&&__md_analytics()</script>
  

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#dada2-pipeline-for-processing-raw-16s-sequencing-data" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Mucosal Immunology Bioinformatics Hub" class="md-header__button md-logo" aria-label="Mucosal Immunology Bioinformatics Hub" data-md-component="logo">
      
  <img src="../../assets/mil_logo.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Mucosal Immunology Bioinformatics Hub
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              DADA2 pipeline
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://mucosal-immunology-lab.github.io/bioinformatics-hub/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    Mucosal-Immunology-Lab/bioinformatic-hub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Mucosal Immunology Bioinformatics Hub" class="md-nav__button md-logo" aria-label="Mucosal Immunology Bioinformatics Hub" data-md-component="logo">
      
  <img src="../../assets/mil_logo.png" alt="logo">

    </a>
    Mucosal Immunology Bioinformatics Hub
  </label>
  
    <div class="md-nav__source">
      <a href="https://mucosal-immunology-lab.github.io/bioinformatics-hub/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    Mucosal-Immunology-Lab/bioinformatic-hub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../RNAseq/rnaseq-nfcore/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Bulk Transcriptomics
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" checked>
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Microbiome
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Microbiome
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_1" checked>
        
          
          <label class="md-nav__link" for="__nav_3_1" id="__nav_3_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    16S rRNA amplicon
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_1_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_3_1">
            <span class="md-nav__icon md-icon"></span>
            16S rRNA amplicon
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    DADA2 pipeline
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    DADA2 pipeline
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview" class="md-nav__link">
    <span class="md-ellipsis">
      Overview ğŸ“–
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#preparation" class="md-nav__link">
    <span class="md-ellipsis">
      Preparation ğŸ—ï¸
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Preparation ğŸ—ï¸">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#conda-environment" class="md-nav__link">
    <span class="md-ellipsis">
      Conda environment ğŸ
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sequencing-files" class="md-nav__link">
    <span class="md-ellipsis">
      Sequencing files ğŸ§¬ğŸ—’ï¸
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#silva-database" class="md-nav__link">
    <span class="md-ellipsis">
      SILVA database ğŸ¦ ğŸ—ƒï¸
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#install-raxml" class="md-nav__link">
    <span class="md-ellipsis">
      Install RAxML ğŸŒ³
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#usage" class="md-nav__link">
    <span class="md-ellipsis">
      Usage ğŸƒ
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Usage ğŸƒ">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#workflow-overview" class="md-nav__link">
    <span class="md-ellipsis">
      Workflow overview ğŸ“–
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Workflow overview ğŸ“–">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#demultiplexing" class="md-nav__link">
    <span class="md-ellipsis">
      Demultiplexing ğŸ—’ï¸ğŸª“
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quality-check" class="md-nav__link">
    <span class="md-ellipsis">
      Quality check ğŸ“‰ğŸ”
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quality-filtering-and-trimming" class="md-nav__link">
    <span class="md-ellipsis">
      Quality filtering and trimming ğŸ§¬âœ‚ï¸
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sequencing-error-model-generation" class="md-nav__link">
    <span class="md-ellipsis">
      Sequencing error model generation ğŸ”¬ğŸ“‰
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#count-table-generation" class="md-nav__link">
    <span class="md-ellipsis">
      Count table generation ğŸ§®
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#merge-runs" class="md-nav__link">
    <span class="md-ellipsis">
      Merge runs ğŸ¤ğŸ§ª
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#chimera-screening" class="md-nav__link">
    <span class="md-ellipsis">
      Chimera screening ğŸ¦„
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#reads-tracking" class="md-nav__link">
    <span class="md-ellipsis">
      Reads tracking ğŸ“ŠğŸ›¤ï¸
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#taxonomy-assignment" class="md-nav__link">
    <span class="md-ellipsis">
      Taxonomy Assignment ğŸ§¬ğŸ·ï¸
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#export-in-qiime-classic-otu-table-like-format" class="md-nav__link">
    <span class="md-ellipsis">
      Export in Qiime classic OTU table-like format ğŸ’¾ğŸ” 
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#create-a-phylogenetic-tree" class="md-nav__link">
    <span class="md-ellipsis">
      Create a phylogenetic tree ğŸŒ³ğŸ§¬
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#rights" class="md-nav__link">
    <span class="md-ellipsis">
      Rights
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    LCMS Analysis
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            LCMS Analysis
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../LCMS/lcms-analysis/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../LCMS/msdial-processing/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    1. Raw Data Processing with MS-DIAL
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../LCMS/pmp-quality-control/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    2. Peak Matrix Processing
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../LCMS/secondary-feature-annotation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    3. Secondary Feature Annotation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../LCMS/manual-peak-curation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    4. Manual Peak Curation
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Nextflow Pipelines
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Nextflow Pipelines
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../NextFlow/scRNAseq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Single-cell RNAseq
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../NextFlow/metagenomics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Metagenomics
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../NextFlow/dada2_16S/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    DADA2 16S rRNA
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Cross-omic Tools
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Cross-omic Tools
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6_1" >
        
          
          <label class="md-nav__link" for="__nav_6_1" id="__nav_6_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Differential testing
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_6_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6_1">
            <span class="md-nav__icon md-icon"></span>
            Differential testing
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../CrossOmicTools/DifferentialTesting/bio_limma/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Linear modelling with bio_limma
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" >
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Utilities
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            Utilities
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7_1" >
        
          
          <label class="md-nav__link" for="__nav_7_1" id="__nav_7_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Handling sequencing data
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_7_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7_1">
            <span class="md-nav__icon md-icon"></span>
            Handling sequencing data
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Utilities/data-best-practices/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Data best practices
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Utilities/convert-raw-novaseq-outputs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Converting NovaSeq outputs
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Utilities/sra-data-submission/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SRA sequencing data submission
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7_2" >
        
          
          <label class="md-nav__link" for="__nav_7_2" id="__nav_7_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Monash data storage
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_7_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7_2">
            <span class="md-nav__icon md-icon"></span>
            Monash data storage
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Utilities/market-storage/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Market storage
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Utilities/vault-storage/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Vault storage
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../PublicDatasets/public-datasets/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Public Datasets
    
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview" class="md-nav__link">
    <span class="md-ellipsis">
      Overview ğŸ“–
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#preparation" class="md-nav__link">
    <span class="md-ellipsis">
      Preparation ğŸ—ï¸
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Preparation ğŸ—ï¸">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#conda-environment" class="md-nav__link">
    <span class="md-ellipsis">
      Conda environment ğŸ
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sequencing-files" class="md-nav__link">
    <span class="md-ellipsis">
      Sequencing files ğŸ§¬ğŸ—’ï¸
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#silva-database" class="md-nav__link">
    <span class="md-ellipsis">
      SILVA database ğŸ¦ ğŸ—ƒï¸
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#install-raxml" class="md-nav__link">
    <span class="md-ellipsis">
      Install RAxML ğŸŒ³
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#usage" class="md-nav__link">
    <span class="md-ellipsis">
      Usage ğŸƒ
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Usage ğŸƒ">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#workflow-overview" class="md-nav__link">
    <span class="md-ellipsis">
      Workflow overview ğŸ“–
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Workflow overview ğŸ“–">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#demultiplexing" class="md-nav__link">
    <span class="md-ellipsis">
      Demultiplexing ğŸ—’ï¸ğŸª“
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quality-check" class="md-nav__link">
    <span class="md-ellipsis">
      Quality check ğŸ“‰ğŸ”
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quality-filtering-and-trimming" class="md-nav__link">
    <span class="md-ellipsis">
      Quality filtering and trimming ğŸ§¬âœ‚ï¸
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sequencing-error-model-generation" class="md-nav__link">
    <span class="md-ellipsis">
      Sequencing error model generation ğŸ”¬ğŸ“‰
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#count-table-generation" class="md-nav__link">
    <span class="md-ellipsis">
      Count table generation ğŸ§®
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#merge-runs" class="md-nav__link">
    <span class="md-ellipsis">
      Merge runs ğŸ¤ğŸ§ª
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#chimera-screening" class="md-nav__link">
    <span class="md-ellipsis">
      Chimera screening ğŸ¦„
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#reads-tracking" class="md-nav__link">
    <span class="md-ellipsis">
      Reads tracking ğŸ“ŠğŸ›¤ï¸
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#taxonomy-assignment" class="md-nav__link">
    <span class="md-ellipsis">
      Taxonomy Assignment ğŸ§¬ğŸ·ï¸
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#export-in-qiime-classic-otu-table-like-format" class="md-nav__link">
    <span class="md-ellipsis">
      Export in Qiime classic OTU table-like format ğŸ’¾ğŸ” 
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#create-a-phylogenetic-tree" class="md-nav__link">
    <span class="md-ellipsis">
      Create a phylogenetic tree ğŸŒ³ğŸ§¬
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#rights" class="md-nav__link">
    <span class="md-ellipsis">
      Rights
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  



<h1 id="dada2-pipeline-for-processing-raw-16s-sequencing-data">DADA2 pipeline for processing raw 16S sequencing data</h1>
<figure>
  <img alt="DADA2 logo" src="../../assets/Microbiome/16S/dada2_logo.png" width="300" />
</figure>
<h2 id="overview">Overview ğŸ“–</h2>
<p>Here we will run through the process for generating a sequencing variants table from raw FASTQ files generated by paired-end Illumina sequencing. For reproducibility, we also provide an <a href="../../assets/Microbiome/16S/DADA2_pipeline.Rmd"><code>Rmarkdown</code> file</a> that covers all the steps to produce the outputs required for assembling a <code>phyloseq</code> container object for your microbiome data.</p>
<table>
<thead>
<tr>
<th>Output</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>seqtab_nochim.rds</code></td>
<td>The clean counts table matrix</td>
</tr>
<tr>
<td><code>taxonomy_species.rds</code></td>
<td>The SILVA-assigned taxonomy table</td>
</tr>
<tr>
<td><code>tree.rds</code></td>
<td>The <em>de novo</em> phylogenetic tree</td>
</tr>
</tbody>
</table>
<div class="admonition tip">
<p class="admonition-title">We have a Nextflow pipeline for that!</p>
<p>If you are familiar with the <code>bash</code> command line interface, you may also want to try the <a href="../../NextFlow/dada2_16S/">Nextflow pipeline</a> we have built which will automate the processes contained in this guide.</p>
</div>
<h2 id="preparation">Preparation ğŸ—ï¸</h2>
<h3 id="conda-environment">Conda environment ğŸ</h3>
<p>Since we require the use of <code>illumina-utils</code> for demultiplexing our raw reads, it is recommended to create a conda environment to keep package versions separate from the global <code>base</code> environment. For faster environment creation, we will use the Miniforge3 package manager.</p>
<details class="question">
<summary>How can I install Miniforge3?</summary>
<div class="tabbed-set tabbed-alternate" data-tabs="1:3"><input checked="checked" id="__tabbed_1_1" name="__tabbed_1" type="radio" /><input id="__tabbed_1_2" name="__tabbed_1" type="radio" /><input id="__tabbed_1_3" name="__tabbed_1" type="radio" /><div class="tabbed-labels"><label for="__tabbed_1_1">Linux and MacOS ğŸ§ğŸ</label><label for="__tabbed_1_2">MacOS (Homebrew) ğŸº</label><label for="__tabbed_1_3">Windows ğŸªŸ</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<ul>
<li>Open a new terminal window, and download the appropriate installer for your computer's architecture using the <code>wget</code> command.</li>
</ul>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>wget<span class="w"> </span><span class="s2">&quot;https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-</span><span class="k">$(</span>uname<span class="k">)</span><span class="s2">-</span><span class="k">$(</span>uname<span class="w"> </span>-m<span class="k">)</span><span class="s2">.sh&quot;</span>
</span></code></pre></div>
<ul>
<li>Run the installation script:</li>
</ul>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>bash<span class="w"> </span>Miniforge3-<span class="k">$(</span>uname<span class="k">)</span>-<span class="k">$(</span>uname<span class="w"> </span>-m<span class="k">)</span>.sh
</span></code></pre></div>
<ul>
<li>The interactive installation will prompt you to initialize conda with your shell.</li>
</ul>
</div>
<div class="tabbed-block">
<ul>
<li>Ensure you have the <a href="https://brew.sh/">Homebrew</a> package manager installed.</li>
</ul>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>/bin/bash<span class="w"> </span>-c<span class="w"> </span><span class="s2">&quot;</span><span class="k">$(</span>curl<span class="w"> </span>-fsSL<span class="w"> </span>https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh<span class="k">)</span><span class="s2">&quot;</span>
</span></code></pre></div>
<ul>
<li>Install miniforge:</li>
</ul>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a>brew<span class="w"> </span>install<span class="w"> </span>--cask<span class="w"> </span>miniforge
</span></code></pre></div>
</div>
<div class="tabbed-block">
<ul>
<li>Download and run the <a href="https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-Windows-x86_64.exe">Windows installer</a>.</li>
<li>Follow the prompts, taking note to:<ul>
<li><code>Create start menu shortcuts</code></li>
<li><code>Add Miniforge3 to my PATH environment variable</code> (unselected by default)</li>
</ul>
</li>
</ul>
<div class="admonition info">
<p class="admonition-title">Info</p>
<p>The latter option above is not selected by default due to potential conflicts with other software. Without Miniforge3 on the path, the most convenient way to use the installed software (such as commands conda and mamba) will be via the "Miniforge Prompt" installed to the start menu.</p>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>There are known issues with the usage of special characters and spaces in the installation location, see for example <a href="https://github.com/conda-forge/miniforge/issues/484">#484</a> on the Miniforge3 GitHub. We recommend users install in a directory without any such characters in the name.</p>
</div>
</div>
</div>
</div>
</details>
<p>Let's create a new environment that houses the tools we need to run the pipeline.</p>
<div class="language-bash highlight"><span class="filename">Create a new environment for the DADA2 pipeline</span><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a><span class="c1"># Create a new conda environment named dada2</span>
</span><span id="__span-4-2"><a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a>mamba<span class="w"> </span>create<span class="w"> </span>-n<span class="w"> </span>dada2<span class="w"> </span><span class="nv">python</span><span class="o">=</span><span class="m">3</span>.8<span class="w"> </span>parallel<span class="w"> </span>git<span class="w"> </span>-y
</span><span id="__span-4-3"><a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a>
</span><span id="__span-4-4"><a id="__codelineno-4-4" name="__codelineno-4-4" href="#__codelineno-4-4"></a><span class="c1"># Activate the environment</span>
</span><span id="__span-4-5"><a id="__codelineno-4-5" name="__codelineno-4-5" href="#__codelineno-4-5"></a>mamba<span class="w"> </span>activate<span class="w"> </span>dada2
</span><span id="__span-4-6"><a id="__codelineno-4-6" name="__codelineno-4-6" href="#__codelineno-4-6"></a>
</span><span id="__span-4-7"><a id="__codelineno-4-7" name="__codelineno-4-7" href="#__codelineno-4-7"></a><span class="c1"># Install illumina-utils</span>
</span><span id="__span-4-8"><a id="__codelineno-4-8" name="__codelineno-4-8" href="#__codelineno-4-8"></a>pip<span class="w"> </span>install<span class="w"> </span>illumina-utils
</span></code></pre></div>
<h3 id="sequencing-files">Sequencing files ğŸ§¬ğŸ—’ï¸</h3>
<p>Before starting, make sure you have the four files listed below for each run you want to process:</p>
<table>
<thead>
<tr>
<th>File</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>R1.fastq.gz</code></td>
<td>FASTQ file for the forward read</td>
</tr>
<tr>
<td><code>R2.fastq.gz</code></td>
<td>FASTQ file for the reverse read</td>
</tr>
<tr>
<td><code>Index.fastq.gz</code></td>
<td>FASTQ file for the index read</td>
</tr>
<tr>
<td><code>barcode_to_sample_[runNN].txt</code></td>
<td>A text file with no column names that maps index barcodes to samples</td>
</tr>
</tbody>
</table>
<div class="admonition warning">
<p class="admonition-title">Things to pay attention to:</p>
<ul>
<li>FASTQ files must use the Phred+33 quality score format and sequences headers (@ lines) must fit the standard format of the CASAVA 1.8 output:</li>
</ul>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-5-1"><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a>@EAS139:136:FC706VJ:2:2104:15343:197393<span class="w"> </span><span class="m">1</span>:N:0:0
</span></code></pre></div>
<ul>
<li>The <code>barcode_to_sample_[runNN].txt</code> file must contain two tab-delimited columns: the first for samples names and the second for samples barcodes. Avoid special characters, and ensure you do not include column names.</li>
</ul>
</div>
<p>An example directory layout is shown below, and is the format that the <a href="../../assets/Microbiome/16S/DADA2_pipeline.Rmd"><code>Rmarkdown</code> file</a> will be expecting.
This layout future-proofs the directory by allowing other inputs, such as metadata or extra omics, to be stored in the same <code>input</code> folder without everything becoming too cluttered.</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-6-1"><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a>ProjectDirectory/
</span><span id="__span-6-2"><a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a>â”œâ”€â”€<span class="w"> </span>DADA2_pipeline.Rmd
</span><span id="__span-6-3"><a id="__codelineno-6-3" name="__codelineno-6-3" href="#__codelineno-6-3"></a>â””â”€â”€<span class="w"> </span>input/
</span><span id="__span-6-4"><a id="__codelineno-6-4" name="__codelineno-6-4" href="#__codelineno-6-4"></a><span class="w">    </span>â””â”€â”€<span class="w"> </span>01_dada2/
</span><span id="__span-6-5"><a id="__codelineno-6-5" name="__codelineno-6-5" href="#__codelineno-6-5"></a><span class="w">        </span>â””â”€â”€<span class="w"> </span>run_data/
</span><span id="__span-6-6"><a id="__codelineno-6-6" name="__codelineno-6-6" href="#__codelineno-6-6"></a><span class="w">            </span>â”œâ”€â”€<span class="w"> </span>run01/
</span><span id="__span-6-7"><a id="__codelineno-6-7" name="__codelineno-6-7" href="#__codelineno-6-7"></a><span class="w">            </span>â”‚<span class="w">   </span>â”œâ”€â”€<span class="w"> </span>R1.fastq.gz
</span><span id="__span-6-8"><a id="__codelineno-6-8" name="__codelineno-6-8" href="#__codelineno-6-8"></a><span class="w">            </span>â”‚<span class="w">   </span>â”œâ”€â”€<span class="w"> </span>R2.fastq.gz
</span><span id="__span-6-9"><a id="__codelineno-6-9" name="__codelineno-6-9" href="#__codelineno-6-9"></a><span class="w">            </span>â”‚<span class="w">   </span>â”œâ”€â”€<span class="w"> </span>Index.fastq.gz
</span><span id="__span-6-10"><a id="__codelineno-6-10" name="__codelineno-6-10" href="#__codelineno-6-10"></a><span class="w">            </span>â”‚<span class="w">   </span>â””â”€â”€<span class="w"> </span>barcode_to_sample_run01.txt
</span><span id="__span-6-11"><a id="__codelineno-6-11" name="__codelineno-6-11" href="#__codelineno-6-11"></a><span class="w">            </span>â”œâ”€â”€<span class="w"> </span>run02/<span class="w"> </span><span class="c1">#(if present)</span>
</span><span id="__span-6-12"><a id="__codelineno-6-12" name="__codelineno-6-12" href="#__codelineno-6-12"></a><span class="w">            </span>â”‚<span class="w">   </span>â”œâ”€â”€<span class="w"> </span>R1.fastq.gz
</span><span id="__span-6-13"><a id="__codelineno-6-13" name="__codelineno-6-13" href="#__codelineno-6-13"></a><span class="w">            </span>â”‚<span class="w">   </span>â”œâ”€â”€<span class="w"> </span>R2.fastq.gz
</span><span id="__span-6-14"><a id="__codelineno-6-14" name="__codelineno-6-14" href="#__codelineno-6-14"></a><span class="w">            </span>â”‚<span class="w">   </span>â”œâ”€â”€<span class="w"> </span>Index.fastq.gz
</span><span id="__span-6-15"><a id="__codelineno-6-15" name="__codelineno-6-15" href="#__codelineno-6-15"></a><span class="w">            </span>â”‚<span class="w">   </span>â””â”€â”€<span class="w"> </span>barcode_to_sample_run02.txt
</span><span id="__span-6-16"><a id="__codelineno-6-16" name="__codelineno-6-16" href="#__codelineno-6-16"></a><span class="w">            </span>â””â”€â”€<span class="w"> </span>...
</span></code></pre></div>
<h3 id="silva-database">SILVA database ğŸ¦ ğŸ—ƒï¸</h3>
<p>You will also require a DADA2-formatted reference database for assigning taxonomy and species-level annotation. The different options are available <a href="https://benjjneb.github.io/dada2/training.html">here</a>.</p>
<p>Make a note of the paths to each of these files, as you will need to enter these in the <code>Rmarkdown</code> file. We advise storing them in a dedicated databases folder rather than in your individual project directory, but that's up to you!</p>
<div class="admonition abstract">
<p class="admonition-title">Download the DADA2-formatted SILVA database</p>
<p>We recommend the SILVA database: <a href="https://zenodo.org/records/14169026">DADA2-formatted SILVA database version 138.2</a>.</p>
<ul>
<li>You will require the following database file, which can be downloaded directly by clicking the link.<ul>
<li><a href="https://zenodo.org/records/14169026/files/silva_nr99_v138.2_toSpecies_trainset.fa.gz?download=1"><code>silva_nr99_v138.2_toSpecies_trainset.fa.gz</code></a></li>
</ul>
</li>
</ul>
</div>
<h3 id="install-raxml">Install RAxML ğŸŒ³</h3>
<p><a href="https://cme.h-its.org/exelixis/web/software/raxml/">RAxML (Randomised Axelerated Maximum Likelihood)</a> is a widely used software for phylogenetic tree inference, and can be easily installed, assuming you have <strong>root user access</strong>.</p>
<details class="question">
<summary>How do I install RAxML?</summary>
<div class="tabbed-set tabbed-alternate" data-tabs="2:2"><input checked="checked" id="__tabbed_2_1" name="__tabbed_2" type="radio" /><input id="__tabbed_2_2" name="__tabbed_2" type="radio" /><div class="tabbed-labels"><label for="__tabbed_2_1">Linux ğŸ§</label><label for="__tabbed_2_2">MacOS (Homebrew) ğŸº</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="language-bash highlight"><span class="filename">Install multi-threaded RAxML</span><pre><span></span><code><span id="__span-7-1"><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a><span class="c1"># Install a C compiler and make if they are not already available</span>
</span><span id="__span-7-2"><a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a>sudo<span class="w"> </span>apt<span class="w"> </span>update
</span><span id="__span-7-3"><a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a>sudo<span class="w"> </span>apt<span class="w"> </span>install<span class="w"> </span>build-essential
</span><span id="__span-7-4"><a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a>
</span><span id="__span-7-5"><a id="__codelineno-7-5" name="__codelineno-7-5" href="#__codelineno-7-5"></a><span class="c1"># Download RAxML</span>
</span><span id="__span-7-6"><a id="__codelineno-7-6" name="__codelineno-7-6" href="#__codelineno-7-6"></a><span class="nb">cd</span><span class="w"> </span>~/Downloads
</span><span id="__span-7-7"><a id="__codelineno-7-7" name="__codelineno-7-7" href="#__codelineno-7-7"></a>git<span class="w"> </span>clone<span class="w"> </span>https://github.com/stamatak/standard-RAxML.git
</span><span id="__span-7-8"><a id="__codelineno-7-8" name="__codelineno-7-8" href="#__codelineno-7-8"></a><span class="nb">cd</span><span class="w"> </span>standard-RAxML
</span><span id="__span-7-9"><a id="__codelineno-7-9" name="__codelineno-7-9" href="#__codelineno-7-9"></a>
</span><span id="__span-7-10"><a id="__codelineno-7-10" name="__codelineno-7-10" href="#__codelineno-7-10"></a><span class="c1"># Compile the multi-threaded version</span>
</span><span id="__span-7-11"><a id="__codelineno-7-11" name="__codelineno-7-11" href="#__codelineno-7-11"></a>make<span class="w"> </span>-f<span class="w"> </span>Makefile.PTHREADS.gcc
</span><span id="__span-7-12"><a id="__codelineno-7-12" name="__codelineno-7-12" href="#__codelineno-7-12"></a>
</span><span id="__span-7-13"><a id="__codelineno-7-13" name="__codelineno-7-13" href="#__codelineno-7-13"></a><span class="c1"># Move the executable to a system directory</span>
</span><span id="__span-7-14"><a id="__codelineno-7-14" name="__codelineno-7-14" href="#__codelineno-7-14"></a>sudo<span class="w"> </span>mv<span class="w"> </span>raxmlHPC-PTHREADS<span class="w"> </span>/usr/bin/
</span><span id="__span-7-15"><a id="__codelineno-7-15" name="__codelineno-7-15" href="#__codelineno-7-15"></a>sudo<span class="w"> </span>chmod<span class="w"> </span>+x<span class="w"> </span>/usr/bin/raxmlHPC-PTHREADS
</span><span id="__span-7-16"><a id="__codelineno-7-16" name="__codelineno-7-16" href="#__codelineno-7-16"></a>
</span><span id="__span-7-17"><a id="__codelineno-7-17" name="__codelineno-7-17" href="#__codelineno-7-17"></a><span class="c1"># Check that the executable is correctly installed and accessible</span>
</span><span id="__span-7-18"><a id="__codelineno-7-18" name="__codelineno-7-18" href="#__codelineno-7-18"></a>raxmlHPC-PTHREADS<span class="w"> </span>-v<span class="w"> </span><span class="c1"># should show the version</span>
</span></code></pre></div>
</div>
<div class="tabbed-block">
<ul>
<li>Ensure you have the <a href="https://brew.sh/">Homebrew</a> package manager installed.</li>
</ul>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-8-1"><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a>/bin/bash<span class="w"> </span>-c<span class="w"> </span><span class="s2">&quot;</span><span class="k">$(</span>curl<span class="w"> </span>-fsSL<span class="w"> </span>https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh<span class="k">)</span><span class="s2">&quot;</span>
</span></code></pre></div>
<ul>
<li>Then you can install RAxML as follows:</li>
</ul>
<div class="language-bash highlight"><span class="filename">Install multi-threaded RAxML</span><pre><span></span><code><span id="__span-9-1"><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a><span class="c1"># Install a C compiler and make if they are not already available</span>
</span><span id="__span-9-2"><a id="__codelineno-9-2" name="__codelineno-9-2" href="#__codelineno-9-2"></a>brew<span class="w"> </span>install<span class="w"> </span>gcc<span class="w"> </span>make
</span><span id="__span-9-3"><a id="__codelineno-9-3" name="__codelineno-9-3" href="#__codelineno-9-3"></a>
</span><span id="__span-9-4"><a id="__codelineno-9-4" name="__codelineno-9-4" href="#__codelineno-9-4"></a><span class="c1"># Download RAxML</span>
</span><span id="__span-9-5"><a id="__codelineno-9-5" name="__codelineno-9-5" href="#__codelineno-9-5"></a><span class="nb">cd</span><span class="w"> </span>~/Downloads
</span><span id="__span-9-6"><a id="__codelineno-9-6" name="__codelineno-9-6" href="#__codelineno-9-6"></a>git<span class="w"> </span>clone<span class="w"> </span>https://github.com/stamatak/standard-RAxML.git
</span><span id="__span-9-7"><a id="__codelineno-9-7" name="__codelineno-9-7" href="#__codelineno-9-7"></a><span class="nb">cd</span><span class="w"> </span>standard-RAxML
</span><span id="__span-9-8"><a id="__codelineno-9-8" name="__codelineno-9-8" href="#__codelineno-9-8"></a>
</span><span id="__span-9-9"><a id="__codelineno-9-9" name="__codelineno-9-9" href="#__codelineno-9-9"></a><span class="c1"># Compile the multi-threaded version</span>
</span><span id="__span-9-10"><a id="__codelineno-9-10" name="__codelineno-9-10" href="#__codelineno-9-10"></a>make<span class="w"> </span>-f<span class="w"> </span>Makefile.PTHREADS.mac_gcc
</span><span id="__span-9-11"><a id="__codelineno-9-11" name="__codelineno-9-11" href="#__codelineno-9-11"></a>
</span><span id="__span-9-12"><a id="__codelineno-9-12" name="__codelineno-9-12" href="#__codelineno-9-12"></a><span class="c1"># Move the executable to a system directory</span>
</span><span id="__span-9-13"><a id="__codelineno-9-13" name="__codelineno-9-13" href="#__codelineno-9-13"></a>sudo<span class="w"> </span>mv<span class="w"> </span>raxmlHPC-PTHREADS<span class="w"> </span>/usr/local/bin
</span><span id="__span-9-14"><a id="__codelineno-9-14" name="__codelineno-9-14" href="#__codelineno-9-14"></a>sudo<span class="w"> </span>chmod<span class="w"> </span>+x<span class="w"> </span>/usr/local/bin/raxmlHPC-PTHREADS
</span><span id="__span-9-15"><a id="__codelineno-9-15" name="__codelineno-9-15" href="#__codelineno-9-15"></a>
</span><span id="__span-9-16"><a id="__codelineno-9-16" name="__codelineno-9-16" href="#__codelineno-9-16"></a><span class="c1"># If /usr/local/bin is not in your PATH variable, add it now</span>
</span><span id="__span-9-17"><a id="__codelineno-9-17" name="__codelineno-9-17" href="#__codelineno-9-17"></a><span class="nb">echo</span><span class="w"> </span><span class="s1">&#39;export PATH=&quot;$PATH:/usr/local/bin&quot;&#39;</span><span class="w"> </span>&gt;&gt;<span class="w"> </span>~/.zshrc
</span><span id="__span-9-18"><a id="__codelineno-9-18" name="__codelineno-9-18" href="#__codelineno-9-18"></a><span class="nb">source</span><span class="w"> </span>~/.zshrc
</span><span id="__span-9-19"><a id="__codelineno-9-19" name="__codelineno-9-19" href="#__codelineno-9-19"></a>
</span><span id="__span-9-20"><a id="__codelineno-9-20" name="__codelineno-9-20" href="#__codelineno-9-20"></a><span class="c1"># Verify installation</span>
</span><span id="__span-9-21"><a id="__codelineno-9-21" name="__codelineno-9-21" href="#__codelineno-9-21"></a>raxmlHPC-PTHREADS<span class="w"> </span>-v<span class="w"> </span><span class="c1"># should show the version</span>
</span></code></pre></div>
</div>
</div>
</div>
</details>
<h2 id="usage">Usage ğŸƒ</h2>
<p>Open the <a href="../../assets/Microbiome/16S/DADA2_pipeline.Rmd"><code>Rmarkdown</code> file</a> in RStudio and follow the instructions in the text and comments.
Hopefully this will all be quite self-explanatory.</p>
<p>At the end of the pipeline, your outputs will be in a handy location for further downstream processing.</p>
<table>
<thead>
<tr>
<th>Output</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>input/01_dada2/seqtab_nochim.rds</code></td>
<td>The clean counts table matrix</td>
</tr>
<tr>
<td><code>input/01_dada2/taxonomy_species.rds</code></td>
<td>The SILVA-assigned taxonomy table</td>
</tr>
<tr>
<td><code>input/01_dada2/tree.rds</code></td>
<td>The <em>de novo</em> phylogenetic tree</td>
</tr>
</tbody>
</table>
<details class="question">
<summary>What if my 16S data is already demultiplexed?</summary>
<p>This is no problem! Organise everything according to the directory layout above. Then, inside each run, create a folder called <code>demultiplexed</code> and store your files there. Everything else should work as planned, and you can skip the demultiplexing steps at the beginning of the pipeline.</p>
</details>
<h3 id="workflow-overview">Workflow overview ğŸ“–</h3>
<p>In this section, we will walk through the steps that are performed throughout the pipeline for reference.
Most of this information is present in the <code>Rmarkdown</code> file as well.</p>
<h4 id="demultiplexing">Demultiplexing ğŸ—’ï¸ğŸª“</h4>
<p>Demultiplexing is performed using the <code>iu-demultiplex</code> command from the illumina-utils FASTQ files processing toolbox.
If multiple runs have to be processed, this will be done in parallel. </p>
<p>The <code>-j</code> argument in the <code>parallel</code> command specifies the number of computing cores to use. You may edit it to your need (considering both available CPUs and memory).</p>
<div class="admonition warning">
<p class="admonition-title">File naming</p>
<p>Please ensure that the files names are consistent in each run. The 4 mandatory files are <code>R1.fastq.gz</code>, <code>R2.fastq.gz</code>, <code>Index.fastq.gz</code>, and <code>barcode_to_sample_[runNN].txt</code></p>
</div>
<details class="abstract">
<summary>Demultiplexing code</summary>
<ul>
<li>Firstly we want to check that we don't have any duplicated sample names in our <code>barcode_to_sample_[runNN].txt</code> mapping files.</li>
</ul>
<div class="language-r highlight"><span class="filename">Check repeating samples</span><pre><span></span><code><span id="__span-10-1"><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a><span class="c1"># Retrieve the barcode to sample mapping file paths</span>
</span><span id="__span-10-2"><a id="__codelineno-10-2" name="__codelineno-10-2" href="#__codelineno-10-2"></a><span class="n">run_data_fp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;run_data&#39;</span><span class="p">)</span>
</span><span id="__span-10-3"><a id="__codelineno-10-3" name="__codelineno-10-3" href="#__codelineno-10-3"></a><span class="n">barcode_files</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="n">run_data_fp</span><span class="p">,</span>
</span><span id="__span-10-4"><a id="__codelineno-10-4" name="__codelineno-10-4" href="#__codelineno-10-4"></a><span class="w">                            </span><span class="nf">list.files</span><span class="p">(</span><span class="n">run_data_fp</span><span class="p">,</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;barcode&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span>
</span><span id="__span-10-5"><a id="__codelineno-10-5" name="__codelineno-10-5" href="#__codelineno-10-5"></a>
</span><span id="__span-10-6"><a id="__codelineno-10-6" name="__codelineno-10-6" href="#__codelineno-10-6"></a><span class="c1"># Read in each of the mapping files to a data.frame</span>
</span><span id="__span-10-7"><a id="__codelineno-10-7" name="__codelineno-10-7" href="#__codelineno-10-7"></a><span class="n">sample_names_dir</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ldply</span><span class="p">(</span><span class="n">barcode_files</span><span class="p">,</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">f</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
</span><span id="__span-10-8"><a id="__codelineno-10-8" name="__codelineno-10-8" href="#__codelineno-10-8"></a><span class="w">    </span><span class="n">dat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read_tsv</span><span class="p">(</span><span class="n">f</span><span class="p">,</span><span class="w"> </span><span class="n">col_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;sample_id&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;barcode&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">col_types</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;cc&#39;</span><span class="p">)</span>
</span><span id="__span-10-9"><a id="__codelineno-10-9" name="__codelineno-10-9" href="#__codelineno-10-9"></a><span class="w">    </span><span class="kr">return</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>
</span><span id="__span-10-10"><a id="__codelineno-10-10" name="__codelineno-10-10" href="#__codelineno-10-10"></a><span class="p">})</span>
</span><span id="__span-10-11"><a id="__codelineno-10-11" name="__codelineno-10-11" href="#__codelineno-10-11"></a>
</span><span id="__span-10-12"><a id="__codelineno-10-12" name="__codelineno-10-12" href="#__codelineno-10-12"></a><span class="c1"># Check for duplicated sample names</span>
</span><span id="__span-10-13"><a id="__codelineno-10-13" name="__codelineno-10-13" href="#__codelineno-10-13"></a><span class="n">duplicate_sample_names_dir</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">sample_names_dir</span><span class="p">[</span><span class="nf">duplicated</span><span class="p">(</span><span class="n">sample_names_dir</span><span class="p">[,</span><span class="m">1</span><span class="p">])</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="nf">duplicated</span><span class="p">(</span><span class="n">sample_names_dir</span><span class="p">[,</span><span class="m">1</span><span class="p">],</span><span class="w"> </span><span class="n">fromLast</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">),]</span>
</span><span id="__span-10-14"><a id="__codelineno-10-14" name="__codelineno-10-14" href="#__codelineno-10-14"></a><span class="n">duplicate_sample_names_dir</span>
</span></code></pre></div>
<ul>
<li>If there are any names returned above, change them in your mapping file, and re-run the code above.</li>
<li>Then we can move on to the demultiplexing step itself, which is handled using <code>bash</code>, which assumes you are using a Unix command line interface.</li>
</ul>
<div class="language-bash highlight"><span class="filename">Demultiplex the runs</span><pre><span></span><code><span id="__span-11-1"><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a><span class="c1"># Activate conda environment if available</span>
</span><span id="__span-11-2"><a id="__codelineno-11-2" name="__codelineno-11-2" href="#__codelineno-11-2"></a><span class="k">if</span><span class="w"> </span><span class="nb">command</span><span class="w"> </span>-v<span class="w"> </span>conda<span class="w"> </span>&gt;/dev/null<span class="w"> </span><span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span><span class="p">;</span><span class="w"> </span><span class="k">then</span>
</span><span id="__span-11-3"><a id="__codelineno-11-3" name="__codelineno-11-3" href="#__codelineno-11-3"></a>conda<span class="w"> </span>activate<span class="w"> </span>dada2
</span><span id="__span-11-4"><a id="__codelineno-11-4" name="__codelineno-11-4" href="#__codelineno-11-4"></a><span class="k">fi</span>
</span><span id="__span-11-5"><a id="__codelineno-11-5" name="__codelineno-11-5" href="#__codelineno-11-5"></a>
</span><span id="__span-11-6"><a id="__codelineno-11-6" name="__codelineno-11-6" href="#__codelineno-11-6"></a>ls<span class="w"> </span>-d<span class="w"> </span>input/01_dada2/run_data/*<span class="w"> </span><span class="p">|</span><span class="w"> </span>parallel<span class="w"> </span>-j<span class="w"> </span>-2<span class="w"> </span><span class="s1">&#39;</span>
</span><span id="__span-11-7"><a id="__codelineno-11-7" name="__codelineno-11-7" href="#__codelineno-11-7"></a><span class="s1">run_dir=&quot;{}&quot;</span>
</span><span id="__span-11-8"><a id="__codelineno-11-8" name="__codelineno-11-8" href="#__codelineno-11-8"></a><span class="s1">outputdir=&quot;demultiplexed&quot;</span>
</span><span id="__span-11-9"><a id="__codelineno-11-9" name="__codelineno-11-9" href="#__codelineno-11-9"></a>
</span><span id="__span-11-10"><a id="__codelineno-11-10" name="__codelineno-11-10" href="#__codelineno-11-10"></a><span class="s1"># Create the output directory if it does not exist</span>
</span><span id="__span-11-11"><a id="__codelineno-11-11" name="__codelineno-11-11" href="#__codelineno-11-11"></a><span class="s1">if [[ ! -d &quot;${run_dir}/${outputdir}&quot; ]]; then</span>
</span><span id="__span-11-12"><a id="__codelineno-11-12" name="__codelineno-11-12" href="#__codelineno-11-12"></a><span class="s1">    mkdir &quot;${run_dir}/${outputdir}&quot;</span>
</span><span id="__span-11-13"><a id="__codelineno-11-13" name="__codelineno-11-13" href="#__codelineno-11-13"></a><span class="s1">fi</span>
</span><span id="__span-11-14"><a id="__codelineno-11-14" name="__codelineno-11-14" href="#__codelineno-11-14"></a>
</span><span id="__span-11-15"><a id="__codelineno-11-15" name="__codelineno-11-15" href="#__codelineno-11-15"></a><span class="s1"># Enable nullglob so that the for loop does not run if no .fastq.gz files are found</span>
</span><span id="__span-11-16"><a id="__codelineno-11-16" name="__codelineno-11-16" href="#__codelineno-11-16"></a><span class="s1">shopt -s nullglob</span>
</span><span id="__span-11-17"><a id="__codelineno-11-17" name="__codelineno-11-17" href="#__codelineno-11-17"></a><span class="s1">for f in &quot;${run_dir}&quot;/*.fastq.gz; do</span>
</span><span id="__span-11-18"><a id="__codelineno-11-18" name="__codelineno-11-18" href="#__codelineno-11-18"></a><span class="s1">    # Only gunzip if the uncompressed file does not exist</span>
</span><span id="__span-11-19"><a id="__codelineno-11-19" name="__codelineno-11-19" href="#__codelineno-11-19"></a><span class="s1">    if [ ! -f &quot;${f%.gz}&quot; ]; then</span>
</span><span id="__span-11-20"><a id="__codelineno-11-20" name="__codelineno-11-20" href="#__codelineno-11-20"></a><span class="s1">    gunzip &quot;$f&quot;</span>
</span><span id="__span-11-21"><a id="__codelineno-11-21" name="__codelineno-11-21" href="#__codelineno-11-21"></a><span class="s1">    fi</span>
</span><span id="__span-11-22"><a id="__codelineno-11-22" name="__codelineno-11-22" href="#__codelineno-11-22"></a><span class="s1">done</span>
</span><span id="__span-11-23"><a id="__codelineno-11-23" name="__codelineno-11-23" href="#__codelineno-11-23"></a>
</span><span id="__span-11-24"><a id="__codelineno-11-24" name="__codelineno-11-24" href="#__codelineno-11-24"></a><span class="s1"># Run demultiplexing; note that the glob is unquoted so it can expand properly</span>
</span><span id="__span-11-25"><a id="__codelineno-11-25" name="__codelineno-11-25" href="#__codelineno-11-25"></a><span class="s1">iu-demultiplex -s ${run_dir}/barcode_to_sample* \</span>
</span><span id="__span-11-26"><a id="__codelineno-11-26" name="__codelineno-11-26" href="#__codelineno-11-26"></a><span class="s1">                --r1 &quot;${run_dir}/R1.fastq&quot; \</span>
</span><span id="__span-11-27"><a id="__codelineno-11-27" name="__codelineno-11-27" href="#__codelineno-11-27"></a><span class="s1">                --r2 &quot;${run_dir}/R2.fastq&quot; \</span>
</span><span id="__span-11-28"><a id="__codelineno-11-28" name="__codelineno-11-28" href="#__codelineno-11-28"></a><span class="s1">                -i &quot;${run_dir}/Index.fastq&quot; \</span>
</span><span id="__span-11-29"><a id="__codelineno-11-29" name="__codelineno-11-29" href="#__codelineno-11-29"></a><span class="s1">                -x \</span>
</span><span id="__span-11-30"><a id="__codelineno-11-30" name="__codelineno-11-30" href="#__codelineno-11-30"></a><span class="s1">                -o &quot;${run_dir}/${outputdir}&quot;</span>
</span><span id="__span-11-31"><a id="__codelineno-11-31" name="__codelineno-11-31" href="#__codelineno-11-31"></a><span class="s1">&#39;</span>
</span></code></pre></div>
</details>
<h4 id="quality-check">Quality check ğŸ“‰ğŸ”</h4>
<p>The DADA2 <code>plotQualityProfile</code> function plots a visual summary of the distribution of quality scores as a function of sequence position for the input fastq file.</p>
<p>This can take minutes to hours depending on the number of samples and the resources available.</p>
<details class="abstract">
<summary>Quality check code</summary>
<div class="language-r highlight"><span class="filename">Run `plotQualityProfile` for each run</span><pre><span></span><code><span id="__span-12-1"><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a><span class="c1"># Identify the run directories</span>
</span><span id="__span-12-2"><a id="__codelineno-12-2" name="__codelineno-12-2" href="#__codelineno-12-2"></a><span class="n">runs.dirs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.dirs</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;run_data&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
</span><span id="__span-12-3"><a id="__codelineno-12-3" name="__codelineno-12-3" href="#__codelineno-12-3"></a><span class="n">runs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">basename</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">)</span>
</span><span id="__span-12-4"><a id="__codelineno-12-4" name="__codelineno-12-4" href="#__codelineno-12-4"></a>
</span><span id="__span-12-5"><a id="__codelineno-12-5" name="__codelineno-12-5" href="#__codelineno-12-5"></a><span class="c1"># Generate the quality profile plots</span>
</span><span id="__span-12-6"><a id="__codelineno-12-6" name="__codelineno-12-6" href="#__codelineno-12-6"></a><span class="n">plots</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">foreach</span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">runs</span><span class="p">),</span><span class="w"> </span><span class="n">.packages</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;ggplot2&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">%dopar%</span><span class="w"> </span><span class="p">{</span>
</span><span id="__span-12-7"><a id="__codelineno-12-7" name="__codelineno-12-7" href="#__codelineno-12-7"></a><span class="n">p</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">()</span>
</span><span id="__span-12-8"><a id="__codelineno-12-8" name="__codelineno-12-8" href="#__codelineno-12-8"></a><span class="n">p</span><span class="p">[[</span><span class="m">1</span><span class="p">]]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">plotQualityProfile</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;/demultiplexed&#39;</span><span class="p">),</span><span class="w"> </span>
</span><span id="__span-12-9"><a id="__codelineno-12-9" name="__codelineno-12-9" href="#__codelineno-12-9"></a><span class="w">                                    </span><span class="nf">list.files</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;/demultiplexed/&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;R1.fastq&#39;</span><span class="p">),</span><span class="w"> </span>
</span><span id="__span-12-10"><a id="__codelineno-12-10" name="__codelineno-12-10" href="#__codelineno-12-10"></a><span class="w">                                    </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;/&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1e+06</span><span class="p">,</span><span class="w"> </span><span class="n">aggregate</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
</span><span id="__span-12-11"><a id="__codelineno-12-11" name="__codelineno-12-11" href="#__codelineno-12-11"></a><span class="w">    </span><span class="nf">ggtitle</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="s">&#39;Forward reads |&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">runs</span><span class="p">[</span><span class="n">i</span><span class="p">]))</span>
</span><span id="__span-12-12"><a id="__codelineno-12-12" name="__codelineno-12-12" href="#__codelineno-12-12"></a><span class="n">p</span><span class="p">[[</span><span class="m">2</span><span class="p">]]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">plotQualityProfile</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;/demultiplexed&#39;</span><span class="p">),</span><span class="w"> </span>
</span><span id="__span-12-13"><a id="__codelineno-12-13" name="__codelineno-12-13" href="#__codelineno-12-13"></a><span class="w">                                    </span><span class="nf">list.files</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;/demultiplexed/&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;R2.fastq&#39;</span><span class="p">),</span><span class="w"> </span>
</span><span id="__span-12-14"><a id="__codelineno-12-14" name="__codelineno-12-14" href="#__codelineno-12-14"></a><span class="w">                                    </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;/&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1e+06</span><span class="p">,</span><span class="w"> </span><span class="n">aggregate</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
</span><span id="__span-12-15"><a id="__codelineno-12-15" name="__codelineno-12-15" href="#__codelineno-12-15"></a><span class="w">    </span><span class="nf">ggtitle</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="s">&#39;Reverse reads |&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">runs</span><span class="p">[</span><span class="n">i</span><span class="p">]))</span>
</span><span id="__span-12-16"><a id="__codelineno-12-16" name="__codelineno-12-16" href="#__codelineno-12-16"></a><span class="n">p</span>
</span><span id="__span-12-17"><a id="__codelineno-12-17" name="__codelineno-12-17" href="#__codelineno-12-17"></a><span class="p">}</span>
</span><span id="__span-12-18"><a id="__codelineno-12-18" name="__codelineno-12-18" href="#__codelineno-12-18"></a>
</span><span id="__span-12-19"><a id="__codelineno-12-19" name="__codelineno-12-19" href="#__codelineno-12-19"></a><span class="n">plots</span>
</span><span id="__span-12-20"><a id="__codelineno-12-20" name="__codelineno-12-20" href="#__codelineno-12-20"></a>
</span><span id="__span-12-21"><a id="__codelineno-12-21" name="__codelineno-12-21" href="#__codelineno-12-21"></a><span class="c1"># Store the quality profile in the run directory</span>
</span><span id="__span-12-22"><a id="__codelineno-12-22" name="__codelineno-12-22" href="#__codelineno-12-22"></a><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">runs</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
</span><span id="__span-12-23"><a id="__codelineno-12-23" name="__codelineno-12-23" href="#__codelineno-12-23"></a><span class="nf">saveRDS</span><span class="p">(</span><span class="n">plots</span><span class="p">[[</span><span class="n">i</span><span class="p">]],</span><span class="w"> </span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;quality_score.pdf.rds&#39;</span><span class="p">))</span>
</span><span id="__span-12-24"><a id="__codelineno-12-24" name="__codelineno-12-24" href="#__codelineno-12-24"></a><span class="nf">pdf</span><span class="p">(</span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;quality_score.pdf&#39;</span><span class="p">))</span>
</span><span id="__span-12-25"><a id="__codelineno-12-25" name="__codelineno-12-25" href="#__codelineno-12-25"></a><span class="nf">invisible</span><span class="p">(</span><span class="nf">lapply</span><span class="p">(</span><span class="n">plots</span><span class="p">[[</span><span class="n">i</span><span class="p">]],</span><span class="w"> </span><span class="n">print</span><span class="p">))</span>
</span><span id="__span-12-26"><a id="__codelineno-12-26" name="__codelineno-12-26" href="#__codelineno-12-26"></a><span class="nf">invisible</span><span class="p">(</span><span class="nf">dev.off</span><span class="p">())</span>
</span><span id="__span-12-27"><a id="__codelineno-12-27" name="__codelineno-12-27" href="#__codelineno-12-27"></a><span class="p">}</span>
</span></code></pre></div>
<div class="language-r highlight"><span class="filename">Combine plots for all runs</span><pre><span></span><code><span id="__span-13-1"><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a><span class="c1"># Recover variables for next chunk</span>
</span><span id="__span-13-2"><a id="__codelineno-13-2" name="__codelineno-13-2" href="#__codelineno-13-2"></a><span class="n">runs.dirs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.dirs</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;run_data&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
</span><span id="__span-13-3"><a id="__codelineno-13-3" name="__codelineno-13-3" href="#__codelineno-13-3"></a><span class="n">runs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">basename</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">)</span>
</span><span id="__span-13-4"><a id="__codelineno-13-4" name="__codelineno-13-4" href="#__codelineno-13-4"></a><span class="n">plots</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">foreach</span><span class="p">(</span><span class="n">i</span><span class="o">=</span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">runs</span><span class="p">))</span><span class="w"> </span><span class="o">%dopar%</span><span class="w"> </span><span class="p">{</span>
</span><span id="__span-13-5"><a id="__codelineno-13-5" name="__codelineno-13-5" href="#__codelineno-13-5"></a><span class="nf">readRDS</span><span class="p">(</span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;quality_score.pdf.rds&#39;</span><span class="p">))</span>
</span><span id="__span-13-6"><a id="__codelineno-13-6" name="__codelineno-13-6" href="#__codelineno-13-6"></a><span class="p">}</span>
</span><span id="__span-13-7"><a id="__codelineno-13-7" name="__codelineno-13-7" href="#__codelineno-13-7"></a>
</span><span id="__span-13-8"><a id="__codelineno-13-8" name="__codelineno-13-8" href="#__codelineno-13-8"></a><span class="n">nplot.pp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">4</span><span class="w">  </span><span class="c1"># number of plots per page</span>
</span><span id="__span-13-9"><a id="__codelineno-13-9" name="__codelineno-13-9" href="#__codelineno-13-9"></a><span class="n">ncol.pp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">2</span><span class="w">  </span><span class="c1"># number of columns in a page</span>
</span><span id="__span-13-10"><a id="__codelineno-13-10" name="__codelineno-13-10" href="#__codelineno-13-10"></a><span class="n">fig</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">foreach</span><span class="p">(</span><span class="n">i</span><span class="o">=</span><span class="nf">seq</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="nf">unlist</span><span class="p">(</span><span class="n">plots</span><span class="p">,</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)),</span><span class="w"> </span><span class="n">by</span><span class="o">=</span><span class="n">nplot.pp</span><span class="p">),</span><span class="w"> </span><span class="n">.packages</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;ggpubr&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">%dopar%</span><span class="w"> </span><span class="p">{</span>
</span><span id="__span-13-11"><a id="__codelineno-13-11" name="__codelineno-13-11" href="#__codelineno-13-11"></a><span class="nf">ggarrange</span><span class="p">(</span><span class="n">plotlist</span><span class="o">=</span><span class="nf">unlist</span><span class="p">(</span><span class="n">plots</span><span class="p">,</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)[</span><span class="n">i</span><span class="o">:</span><span class="p">(</span><span class="n">i</span><span class="o">+</span><span class="n">nplot.pp</span><span class="m">-1</span><span class="p">)],</span><span class="w"> </span><span class="n">ncol</span><span class="o">=</span><span class="n">ncol.pp</span><span class="p">,</span><span class="w"> </span><span class="n">nrow</span><span class="o">=</span><span class="n">nplot.pp</span><span class="o">/</span><span class="n">ncol.pp</span><span class="p">)</span>
</span><span id="__span-13-12"><a id="__codelineno-13-12" name="__codelineno-13-12" href="#__codelineno-13-12"></a><span class="p">}</span>
</span><span id="__span-13-13"><a id="__codelineno-13-13" name="__codelineno-13-13" href="#__codelineno-13-13"></a><span class="nf">invisible</span><span class="p">(</span><span class="nf">lapply</span><span class="p">(</span><span class="n">fig</span><span class="p">,</span><span class="w"> </span><span class="n">print</span><span class="p">))</span>
</span></code></pre></div>
<ul>
<li>In gray-scale is a heat map of the frequency of each quality score at each base position. The median quality score at each position is shown by the green line, and the quartiles of the quality score distribution by the orange lines. The reverse reads are usually of worse quality, especially at the end, which is common in Illumina sequencing.</li>
</ul>
<div class="language-r highlight"><span class="filename">Save outputs to PDF</span><pre><span></span><code><span id="__span-14-1"><a id="__codelineno-14-1" name="__codelineno-14-1" href="#__codelineno-14-1"></a><span class="c1"># Store the quality profile summary</span>
</span><span id="__span-14-2"><a id="__codelineno-14-2" name="__codelineno-14-2" href="#__codelineno-14-2"></a><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="nf">dir.exists</span><span class="p">(</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;figures&#39;</span><span class="p">)))</span><span class="w"> </span><span class="p">{</span>
</span><span id="__span-14-3"><a id="__codelineno-14-3" name="__codelineno-14-3" href="#__codelineno-14-3"></a><span class="nf">dir.create</span><span class="p">(</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;figures&#39;</span><span class="p">))</span>
</span><span id="__span-14-4"><a id="__codelineno-14-4" name="__codelineno-14-4" href="#__codelineno-14-4"></a><span class="p">}</span>
</span><span id="__span-14-5"><a id="__codelineno-14-5" name="__codelineno-14-5" href="#__codelineno-14-5"></a><span class="nf">pdf</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;figures&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;quality_score.pdf&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">paper</span><span class="o">=</span><span class="s">&#39;a4&#39;</span><span class="p">)</span>
</span><span id="__span-14-6"><a id="__codelineno-14-6" name="__codelineno-14-6" href="#__codelineno-14-6"></a><span class="nf">invisible</span><span class="p">(</span><span class="nf">lapply</span><span class="p">(</span><span class="n">fig</span><span class="p">,</span><span class="w"> </span><span class="n">print</span><span class="p">))</span>
</span><span id="__span-14-7"><a id="__codelineno-14-7" name="__codelineno-14-7" href="#__codelineno-14-7"></a><span class="nf">invisible</span><span class="p">(</span><span class="nf">dev.off</span><span class="p">())</span>
</span></code></pre></div>
</details>
<h4 id="quality-filtering-and-trimming">Quality filtering and trimming ğŸ§¬âœ‚ï¸</h4>
<p>The DADA2 <code>filterAndTrim</code> function trims sequences to a specified length, removes sequences shorter than that length, and filters based on the number of ambiguous bases, a minimum quality score, and the expected errors in a read. Based on the quality profiles, adjust the trimming (for each run). Your reads must still overlap after truncation in order to merge them later (the basic rule is <code>truncLen</code> must be large enough to maintain <code>20</code> + <code>biological.length.variation</code> nucleotides of overlap between them).</p>
<details class="abstract">
<summary>Quality filtering and trimming code</summary>
<div class="language-r highlight"><span class="filename">Set up the required parameters</span><pre><span></span><code><span id="__span-15-1"><a id="__codelineno-15-1" name="__codelineno-15-1" href="#__codelineno-15-1"></a><span class="c1"># Recover variables for next chunk</span>
</span><span id="__span-15-2"><a id="__codelineno-15-2" name="__codelineno-15-2" href="#__codelineno-15-2"></a><span class="n">runs.dirs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.dirs</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;run_data&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)</span>
</span><span id="__span-15-3"><a id="__codelineno-15-3" name="__codelineno-15-3" href="#__codelineno-15-3"></a><span class="n">runs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">basename</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">)</span>
</span><span id="__span-15-4"><a id="__codelineno-15-4" name="__codelineno-15-4" href="#__codelineno-15-4"></a>
</span><span id="__span-15-5"><a id="__codelineno-15-5" name="__codelineno-15-5" href="#__codelineno-15-5"></a><span class="c1"># Set up parameters for filtering and trimming (first parameter stands for R1, second for R2)</span>
</span><span id="__span-15-6"><a id="__codelineno-15-6" name="__codelineno-15-6" href="#__codelineno-15-6"></a><span class="n">truncLen</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">240</span><span class="p">,</span><span class="w"> </span><span class="m">240</span><span class="p">)</span>
</span><span id="__span-15-7"><a id="__codelineno-15-7" name="__codelineno-15-7" href="#__codelineno-15-7"></a><span class="n">maxEE</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">4</span><span class="p">,</span><span class="m">4</span><span class="p">)</span>
</span><span id="__span-15-8"><a id="__codelineno-15-8" name="__codelineno-15-8" href="#__codelineno-15-8"></a><span class="n">trimLeft</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">54</span><span class="p">,</span><span class="w"> </span><span class="m">54</span><span class="p">)</span>
</span><span id="__span-15-9"><a id="__codelineno-15-9" name="__codelineno-15-9" href="#__codelineno-15-9"></a><span class="n">truncQ</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">2</span><span class="p">,</span><span class="m">2</span><span class="p">)</span>
</span><span id="__span-15-10"><a id="__codelineno-15-10" name="__codelineno-15-10" href="#__codelineno-15-10"></a><span class="n">maxN</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">0</span><span class="p">)</span>
</span><span id="__span-15-11"><a id="__codelineno-15-11" name="__codelineno-15-11" href="#__codelineno-15-11"></a><span class="n">rm.phix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kc">TRUE</span>
</span></code></pre></div>
<div class="admonition info">
<p class="admonition-title">Tunable parameters</p>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>truncLen</code></td>
<td>Truncates reads to the specified length (R1, R2). Ensures uniform length but may discard short reads.</td>
</tr>
<tr>
<td><code>maxEE</code></td>
<td>Filters out reads with more than the specified number of expected errors. Lower values improve accuracy but reduce read count.</td>
</tr>
<tr>
<td><code>trimLeft</code></td>
<td>Removes the specified number of bases from the start of each read. Useful for removing primers and non-informative regions.</td>
</tr>
<tr>
<td><code>truncQ</code></td>
<td>Truncates reads at the first base with a quality score â‰¤ <code>truncQ</code>. Helps remove poor-quality tails.</td>
</tr>
<tr>
<td><code>maxN</code></td>
<td>Discards reads containing more than the specified number of ambiguous bases (<code>N</code>). Ensures high-quality sequences.</td>
</tr>
<tr>
<td><code>rm.phix</code></td>
<td>Removes reads matching the PhiX genome, commonly used as a sequencing control, to reduce contamination.</td>
</tr>
</tbody>
</table>
</div>
<div class="language-r highlight"><span class="filename">Perform filtering and trimming</span><pre><span></span><code><span id="__span-16-1"><a id="__codelineno-16-1" name="__codelineno-16-1" href="#__codelineno-16-1"></a><span class="c1"># For each run, store the filtered sequences in a new directory named &#39;filtered&#39;</span>
</span><span id="__span-16-2"><a id="__codelineno-16-2" name="__codelineno-16-2" href="#__codelineno-16-2"></a><span class="n">filterAndTrim.out</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">vector</span><span class="p">(</span><span class="s">&#39;list&#39;</span><span class="p">,</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">runs</span><span class="p">))</span>
</span><span id="__span-16-3"><a id="__codelineno-16-3" name="__codelineno-16-3" href="#__codelineno-16-3"></a><span class="kr">for</span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">runs</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
</span><span id="__span-16-4"><a id="__codelineno-16-4" name="__codelineno-16-4" href="#__codelineno-16-4"></a><span class="n">fwd.fn</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sort</span><span class="p">(</span><span class="nf">list.files</span><span class="p">(</span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;demultiplexed&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;R1.fastq&#39;</span><span class="p">))</span>
</span><span id="__span-16-5"><a id="__codelineno-16-5" name="__codelineno-16-5" href="#__codelineno-16-5"></a><span class="n">rev.fn</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sort</span><span class="p">(</span><span class="nf">list.files</span><span class="p">(</span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;demultiplexed&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;R2.fastq&#39;</span><span class="p">))</span>
</span><span id="__span-16-6"><a id="__codelineno-16-6" name="__codelineno-16-6" href="#__codelineno-16-6"></a><span class="n">filterAndTrim.out</span><span class="p">[[</span><span class="n">i</span><span class="p">]]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">filterAndTrim</span><span class="p">(</span>
</span><span id="__span-16-7"><a id="__codelineno-16-7" name="__codelineno-16-7" href="#__codelineno-16-7"></a><span class="w">                </span><span class="n">fwd</span><span class="o">=</span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;demultiplexed&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">fwd.fn</span><span class="p">),</span>
</span><span id="__span-16-8"><a id="__codelineno-16-8" name="__codelineno-16-8" href="#__codelineno-16-8"></a><span class="w">                </span><span class="n">filt</span><span class="o">=</span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;filtered&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">fwd.fn</span><span class="p">),</span>
</span><span id="__span-16-9"><a id="__codelineno-16-9" name="__codelineno-16-9" href="#__codelineno-16-9"></a><span class="w">                </span><span class="n">rev</span><span class="o">=</span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;demultiplexed&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">rev.fn</span><span class="p">),</span>
</span><span id="__span-16-10"><a id="__codelineno-16-10" name="__codelineno-16-10" href="#__codelineno-16-10"></a><span class="w">                </span><span class="n">filt.rev</span><span class="o">=</span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;filtered&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">rev.fn</span><span class="p">),</span>
</span><span id="__span-16-11"><a id="__codelineno-16-11" name="__codelineno-16-11" href="#__codelineno-16-11"></a><span class="w">                </span><span class="n">truncLen</span><span class="o">=</span><span class="n">truncLen</span><span class="p">,</span>
</span><span id="__span-16-12"><a id="__codelineno-16-12" name="__codelineno-16-12" href="#__codelineno-16-12"></a><span class="w">                </span><span class="n">trimLeft</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">trimLeft</span><span class="p">,</span>
</span><span id="__span-16-13"><a id="__codelineno-16-13" name="__codelineno-16-13" href="#__codelineno-16-13"></a><span class="w">                </span><span class="n">maxEE</span><span class="o">=</span><span class="n">maxEE</span><span class="p">,</span>
</span><span id="__span-16-14"><a id="__codelineno-16-14" name="__codelineno-16-14" href="#__codelineno-16-14"></a><span class="w">                </span><span class="n">truncQ</span><span class="o">=</span><span class="n">truncQ</span><span class="p">,</span>
</span><span id="__span-16-15"><a id="__codelineno-16-15" name="__codelineno-16-15" href="#__codelineno-16-15"></a><span class="w">                </span><span class="n">maxN</span><span class="o">=</span><span class="n">maxN</span><span class="p">,</span>
</span><span id="__span-16-16"><a id="__codelineno-16-16" name="__codelineno-16-16" href="#__codelineno-16-16"></a><span class="w">                </span><span class="n">rm.phix</span><span class="o">=</span><span class="n">rm.phix</span><span class="p">,</span>
</span><span id="__span-16-17"><a id="__codelineno-16-17" name="__codelineno-16-17" href="#__codelineno-16-17"></a><span class="w">                </span><span class="n">compress</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span>
</span><span id="__span-16-18"><a id="__codelineno-16-18" name="__codelineno-16-18" href="#__codelineno-16-18"></a><span class="w">                </span><span class="n">verbose</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span>
</span><span id="__span-16-19"><a id="__codelineno-16-19" name="__codelineno-16-19" href="#__codelineno-16-19"></a><span class="w">                </span><span class="n">multithread</span><span class="o">=</span><span class="n">nc</span><span class="p">)</span>
</span><span id="__span-16-20"><a id="__codelineno-16-20" name="__codelineno-16-20" href="#__codelineno-16-20"></a><span class="p">}</span>
</span><span id="__span-16-21"><a id="__codelineno-16-21" name="__codelineno-16-21" href="#__codelineno-16-21"></a>
</span><span id="__span-16-22"><a id="__codelineno-16-22" name="__codelineno-16-22" href="#__codelineno-16-22"></a><span class="c1"># Store the filtering report in the run directory</span>
</span><span id="__span-16-23"><a id="__codelineno-16-23" name="__codelineno-16-23" href="#__codelineno-16-23"></a><span class="n">filt.plots</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">foreach</span><span class="p">(</span><span class="n">i</span><span class="o">=</span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">runs</span><span class="p">),</span><span class="w"> </span><span class="n">.packages</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;ggplot2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;reshape2&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">%do%</span><span class="w"> </span><span class="p">{</span>
</span><span id="__span-16-24"><a id="__codelineno-16-24" name="__codelineno-16-24" href="#__codelineno-16-24"></a><span class="nf">saveRDS</span><span class="p">(</span><span class="n">filterAndTrim.out</span><span class="p">[[</span><span class="n">i</span><span class="p">]],</span><span class="w"> </span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;filtering_report.rds&#39;</span><span class="p">))</span>
</span><span id="__span-16-25"><a id="__codelineno-16-25" name="__codelineno-16-25" href="#__codelineno-16-25"></a><span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.data.frame</span><span class="p">(</span><span class="n">filterAndTrim.out</span><span class="p">[[</span><span class="n">i</span><span class="p">]])</span>
</span><span id="__span-16-26"><a id="__codelineno-16-26" name="__codelineno-16-26" href="#__codelineno-16-26"></a><span class="nf">row.names</span><span class="p">(</span><span class="n">data</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">gsub</span><span class="p">(</span><span class="s">&#39;_R1_001.fastq|-R1.fastq&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;&#39;</span><span class="p">,</span><span class="w"> </span><span class="nf">row.names</span><span class="p">(</span><span class="n">data</span><span class="p">))</span>
</span><span id="__span-16-27"><a id="__codelineno-16-27" name="__codelineno-16-27" href="#__codelineno-16-27"></a><span class="n">data</span><span class="o">$</span><span class="n">reads.in</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">reads.in</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">reads.out</span>
</span><span id="__span-16-28"><a id="__codelineno-16-28" name="__codelineno-16-28" href="#__codelineno-16-28"></a><span class="n">p</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="nf">melt</span><span class="p">(</span><span class="nf">as.matrix</span><span class="p">(</span><span class="n">data</span><span class="p">)),</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">Var1</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="n">value</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="o">=</span><span class="n">Var2</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
</span><span id="__span-16-29"><a id="__codelineno-16-29" name="__codelineno-16-29" href="#__codelineno-16-29"></a><span class="w">    </span><span class="nf">geom_bar</span><span class="p">(</span><span class="n">stat</span><span class="o">=</span><span class="s">&#39;identity&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
</span><span id="__span-16-30"><a id="__codelineno-16-30" name="__codelineno-16-30" href="#__codelineno-16-30"></a><span class="w">    </span><span class="nf">theme</span><span class="p">(</span><span class="n">axis.text.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">angle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">90</span><span class="p">,</span><span class="w"> </span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">vjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
</span><span id="__span-16-31"><a id="__codelineno-16-31" name="__codelineno-16-31" href="#__codelineno-16-31"></a><span class="w">    </span><span class="nf">labs</span><span class="p">(</span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">runs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;Samples&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;Reads&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
</span><span id="__span-16-32"><a id="__codelineno-16-32" name="__codelineno-16-32" href="#__codelineno-16-32"></a><span class="w">    </span><span class="nf">theme</span><span class="p">(</span><span class="n">axis.title.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_blank</span><span class="p">())</span>
</span><span id="__span-16-33"><a id="__codelineno-16-33" name="__codelineno-16-33" href="#__codelineno-16-33"></a><span class="nf">saveRDS</span><span class="p">(</span><span class="n">p</span><span class="p">,</span><span class="w"> </span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;filtering_report.pdf.rds&#39;</span><span class="p">))</span>
</span><span id="__span-16-34"><a id="__codelineno-16-34" name="__codelineno-16-34" href="#__codelineno-16-34"></a><span class="nf">pdf</span><span class="p">(</span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;filtering_report.pdf&#39;</span><span class="p">))</span>
</span><span id="__span-16-35"><a id="__codelineno-16-35" name="__codelineno-16-35" href="#__codelineno-16-35"></a><span class="nf">print</span><span class="p">(</span><span class="n">p</span><span class="p">)</span>
</span><span id="__span-16-36"><a id="__codelineno-16-36" name="__codelineno-16-36" href="#__codelineno-16-36"></a><span class="nf">invisible</span><span class="p">(</span><span class="nf">dev.off</span><span class="p">())</span>
</span><span id="__span-16-37"><a id="__codelineno-16-37" name="__codelineno-16-37" href="#__codelineno-16-37"></a><span class="n">p</span>
</span><span id="__span-16-38"><a id="__codelineno-16-38" name="__codelineno-16-38" href="#__codelineno-16-38"></a><span class="p">}</span>
</span><span id="__span-16-39"><a id="__codelineno-16-39" name="__codelineno-16-39" href="#__codelineno-16-39"></a><span class="nf">pdf</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;figures&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;filtering_report.pdf&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">12</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">8</span><span class="p">)</span>
</span><span id="__span-16-40"><a id="__codelineno-16-40" name="__codelineno-16-40" href="#__codelineno-16-40"></a><span class="nf">invisible</span><span class="p">(</span><span class="nf">lapply</span><span class="p">(</span><span class="n">filt.plots</span><span class="p">,</span><span class="w"> </span><span class="n">print</span><span class="p">))</span>
</span><span id="__span-16-41"><a id="__codelineno-16-41" name="__codelineno-16-41" href="#__codelineno-16-41"></a><span class="nf">invisible</span><span class="p">(</span><span class="nf">dev.off</span><span class="p">())</span>
</span></code></pre></div>
<div class="language-r highlight"><span class="filename">Save the filtering report to PDF</span><pre><span></span><code><span id="__span-17-1"><a id="__codelineno-17-1" name="__codelineno-17-1" href="#__codelineno-17-1"></a><span class="c1"># Recover variables for next chunck</span>
</span><span id="__span-17-2"><a id="__codelineno-17-2" name="__codelineno-17-2" href="#__codelineno-17-2"></a><span class="n">runs.dirs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.dirs</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;run_data&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)</span>
</span><span id="__span-17-3"><a id="__codelineno-17-3" name="__codelineno-17-3" href="#__codelineno-17-3"></a><span class="n">runs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">basename</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">)</span>
</span><span id="__span-17-4"><a id="__codelineno-17-4" name="__codelineno-17-4" href="#__codelineno-17-4"></a>
</span><span id="__span-17-5"><a id="__codelineno-17-5" name="__codelineno-17-5" href="#__codelineno-17-5"></a><span class="n">filt.plots</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">foreach</span><span class="p">(</span><span class="n">i</span><span class="o">=</span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">))</span><span class="w"> </span><span class="o">%dopar%</span><span class="w"> </span><span class="p">{</span>
</span><span id="__span-17-6"><a id="__codelineno-17-6" name="__codelineno-17-6" href="#__codelineno-17-6"></a><span class="nf">readRDS</span><span class="p">(</span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;filtering_report.pdf.rds&#39;</span><span class="p">))</span>
</span><span id="__span-17-7"><a id="__codelineno-17-7" name="__codelineno-17-7" href="#__codelineno-17-7"></a><span class="p">}</span>
</span><span id="__span-17-8"><a id="__codelineno-17-8" name="__codelineno-17-8" href="#__codelineno-17-8"></a>
</span><span id="__span-17-9"><a id="__codelineno-17-9" name="__codelineno-17-9" href="#__codelineno-17-9"></a><span class="nf">invisible</span><span class="p">(</span><span class="nf">lapply</span><span class="p">(</span><span class="n">filt.plots</span><span class="p">,</span><span class="w"> </span><span class="n">print</span><span class="p">))</span>
</span></code></pre></div>
<p>If too few reads are passing the filter, consider whether you can relax the <code>maxEE</code> parameter, or perhaps reduce the <code>truncLen</code> to remove the lower quality tail ends of reads.</p>
</details>
<h4 id="sequencing-error-model-generation">Sequencing error model generation ğŸ”¬ğŸ“‰</h4>
<p>The DADA2 algorithm makes use of a parametric error model, and every amplicon dataset has a different set of error rates. The <code>learnErrors</code> method learns this error model from the data, by alternating estimation of the error rates and inference of sample composition until they converge on a jointly consistent solution.</p>
<details class="abstract">
<summary>Sequencing error model code</summary>
<div class="language-r highlight"><span class="filename">Generate the sequencing error models</span><pre><span></span><code><span id="__span-18-1"><a id="__codelineno-18-1" name="__codelineno-18-1" href="#__codelineno-18-1"></a><span class="c1"># Recover variables</span>
</span><span id="__span-18-2"><a id="__codelineno-18-2" name="__codelineno-18-2" href="#__codelineno-18-2"></a><span class="n">runs.dirs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.dirs</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;run_data&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)</span>
</span><span id="__span-18-3"><a id="__codelineno-18-3" name="__codelineno-18-3" href="#__codelineno-18-3"></a><span class="n">runs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">basename</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">)</span>
</span><span id="__span-18-4"><a id="__codelineno-18-4" name="__codelineno-18-4" href="#__codelineno-18-4"></a>
</span><span id="__span-18-5"><a id="__codelineno-18-5" name="__codelineno-18-5" href="#__codelineno-18-5"></a><span class="c1"># Identify all of the forward and reverse sequencing files</span>
</span><span id="__span-18-6"><a id="__codelineno-18-6" name="__codelineno-18-6" href="#__codelineno-18-6"></a><span class="n">err.model</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">foreach</span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">runs</span><span class="p">),</span><span class="w"> </span><span class="n">.packages</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;ggplot2&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">%dopar%</span><span class="w"> </span><span class="p">{</span>
</span><span id="__span-18-7"><a id="__codelineno-18-7" name="__codelineno-18-7" href="#__codelineno-18-7"></a><span class="n">fwd.fn</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sort</span><span class="p">(</span><span class="nf">list.files</span><span class="p">(</span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;filtered&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;R1.fastq&#39;</span><span class="p">))</span>
</span><span id="__span-18-8"><a id="__codelineno-18-8" name="__codelineno-18-8" href="#__codelineno-18-8"></a><span class="n">rev.fn</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sort</span><span class="p">(</span><span class="nf">list.files</span><span class="p">(</span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;filtered&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;R2.fastq&#39;</span><span class="p">))</span>
</span><span id="__span-18-9"><a id="__codelineno-18-9" name="__codelineno-18-9" href="#__codelineno-18-9"></a><span class="n">err</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">()</span>
</span><span id="__span-18-10"><a id="__codelineno-18-10" name="__codelineno-18-10" href="#__codelineno-18-10"></a><span class="n">err</span><span class="p">[[</span><span class="m">1</span><span class="p">]]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">learnErrors</span><span class="p">(</span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;filtered&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">fwd.fn</span><span class="p">),</span><span class="w"> </span><span class="n">nbases</span><span class="o">=</span><span class="m">1e8</span><span class="p">,</span><span class="w"> </span><span class="n">multithread</span><span class="o">=</span><span class="n">nc</span><span class="p">)</span>
</span><span id="__span-18-11"><a id="__codelineno-18-11" name="__codelineno-18-11" href="#__codelineno-18-11"></a><span class="n">err</span><span class="p">[[</span><span class="m">2</span><span class="p">]]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">learnErrors</span><span class="p">(</span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;filtered&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">rev.fn</span><span class="p">),</span><span class="w"> </span><span class="n">nbases</span><span class="o">=</span><span class="m">1e8</span><span class="p">,</span><span class="w"> </span><span class="n">multithread</span><span class="o">=</span><span class="n">nc</span><span class="p">)</span>
</span><span id="__span-18-12"><a id="__codelineno-18-12" name="__codelineno-18-12" href="#__codelineno-18-12"></a><span class="n">err</span>
</span><span id="__span-18-13"><a id="__codelineno-18-13" name="__codelineno-18-13" href="#__codelineno-18-13"></a><span class="p">}</span>
</span><span id="__span-18-14"><a id="__codelineno-18-14" name="__codelineno-18-14" href="#__codelineno-18-14"></a><span class="c1"># Plot the error model</span>
</span><span id="__span-18-15"><a id="__codelineno-18-15" name="__codelineno-18-15" href="#__codelineno-18-15"></a><span class="n">err.plots</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">foreach</span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">runs</span><span class="p">),</span><span class="w"> </span><span class="n">.packages</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;ggplot2&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">%do%</span><span class="w"> </span><span class="p">{</span>
</span><span id="__span-18-16"><a id="__codelineno-18-16" name="__codelineno-18-16" href="#__codelineno-18-16"></a><span class="n">p</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">()</span>
</span><span id="__span-18-17"><a id="__codelineno-18-17" name="__codelineno-18-17" href="#__codelineno-18-17"></a><span class="n">p</span><span class="p">[[</span><span class="m">1</span><span class="p">]]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">plotErrors</span><span class="p">(</span><span class="n">err.model</span><span class="p">[[</span><span class="n">i</span><span class="p">]][[</span><span class="m">1</span><span class="p">]],</span><span class="w"> </span><span class="n">nominalQ</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
</span><span id="__span-18-18"><a id="__codelineno-18-18" name="__codelineno-18-18" href="#__codelineno-18-18"></a><span class="w">                </span><span class="nf">ggtitle</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="n">runs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;| forward reads&#39;</span><span class="p">))</span>
</span><span id="__span-18-19"><a id="__codelineno-18-19" name="__codelineno-18-19" href="#__codelineno-18-19"></a><span class="n">p</span><span class="p">[[</span><span class="m">2</span><span class="p">]]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">plotErrors</span><span class="p">(</span><span class="n">err.model</span><span class="p">[[</span><span class="n">i</span><span class="p">]][[</span><span class="m">2</span><span class="p">]],</span><span class="w"> </span><span class="n">nominalQ</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
</span><span id="__span-18-20"><a id="__codelineno-18-20" name="__codelineno-18-20" href="#__codelineno-18-20"></a><span class="w">                </span><span class="nf">ggtitle</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="n">runs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;| reverse reads&#39;</span><span class="p">))</span>
</span><span id="__span-18-21"><a id="__codelineno-18-21" name="__codelineno-18-21" href="#__codelineno-18-21"></a><span class="n">p</span>
</span><span id="__span-18-22"><a id="__codelineno-18-22" name="__codelineno-18-22" href="#__codelineno-18-22"></a><span class="p">}</span>
</span><span id="__span-18-23"><a id="__codelineno-18-23" name="__codelineno-18-23" href="#__codelineno-18-23"></a>
</span><span id="__span-18-24"><a id="__codelineno-18-24" name="__codelineno-18-24" href="#__codelineno-18-24"></a><span class="c1"># Store the error model in the run directory</span>
</span><span id="__span-18-25"><a id="__codelineno-18-25" name="__codelineno-18-25" href="#__codelineno-18-25"></a><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">runs</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
</span><span id="__span-18-26"><a id="__codelineno-18-26" name="__codelineno-18-26" href="#__codelineno-18-26"></a><span class="nf">saveRDS</span><span class="p">(</span><span class="n">err.model</span><span class="p">[[</span><span class="n">i</span><span class="p">]],</span><span class="w"> </span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;error_model.rds&#39;</span><span class="p">))</span>
</span><span id="__span-18-27"><a id="__codelineno-18-27" name="__codelineno-18-27" href="#__codelineno-18-27"></a><span class="nf">saveRDS</span><span class="p">(</span><span class="n">err.plots</span><span class="p">[[</span><span class="n">i</span><span class="p">]],</span><span class="w"> </span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;error_model.pdf.rds&#39;</span><span class="p">))</span>
</span><span id="__span-18-28"><a id="__codelineno-18-28" name="__codelineno-18-28" href="#__codelineno-18-28"></a><span class="nf">pdf</span><span class="p">(</span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;error_model.pdf&#39;</span><span class="p">))</span>
</span><span id="__span-18-29"><a id="__codelineno-18-29" name="__codelineno-18-29" href="#__codelineno-18-29"></a><span class="nf">invisible</span><span class="p">(</span><span class="nf">lapply</span><span class="p">(</span><span class="n">err.plots</span><span class="p">[[</span><span class="n">i</span><span class="p">]],</span><span class="w"> </span><span class="n">print</span><span class="p">))</span>
</span><span id="__span-18-30"><a id="__codelineno-18-30" name="__codelineno-18-30" href="#__codelineno-18-30"></a><span class="nf">invisible</span><span class="p">(</span><span class="nf">dev.off</span><span class="p">())</span>
</span><span id="__span-18-31"><a id="__codelineno-18-31" name="__codelineno-18-31" href="#__codelineno-18-31"></a><span class="p">}</span>
</span></code></pre></div>
<div class="language-r highlight"><span class="filename">Combine the error models into a condensed summary</span><pre><span></span><code><span id="__span-19-1"><a id="__codelineno-19-1" name="__codelineno-19-1" href="#__codelineno-19-1"></a><span class="c1"># Recover variables</span>
</span><span id="__span-19-2"><a id="__codelineno-19-2" name="__codelineno-19-2" href="#__codelineno-19-2"></a><span class="n">runs.dirs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.dirs</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;run_data&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)</span>
</span><span id="__span-19-3"><a id="__codelineno-19-3" name="__codelineno-19-3" href="#__codelineno-19-3"></a><span class="n">runs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">basename</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">)</span>
</span><span id="__span-19-4"><a id="__codelineno-19-4" name="__codelineno-19-4" href="#__codelineno-19-4"></a><span class="n">err.plots</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">foreach</span><span class="p">(</span><span class="n">i</span><span class="o">=</span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">runs</span><span class="p">))</span><span class="w"> </span><span class="o">%dopar%</span><span class="w"> </span><span class="p">{</span>
</span><span id="__span-19-5"><a id="__codelineno-19-5" name="__codelineno-19-5" href="#__codelineno-19-5"></a><span class="nf">readRDS</span><span class="p">(</span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;error_model.pdf.rds&#39;</span><span class="p">))</span>
</span><span id="__span-19-6"><a id="__codelineno-19-6" name="__codelineno-19-6" href="#__codelineno-19-6"></a><span class="p">}</span>
</span><span id="__span-19-7"><a id="__codelineno-19-7" name="__codelineno-19-7" href="#__codelineno-19-7"></a>
</span><span id="__span-19-8"><a id="__codelineno-19-8" name="__codelineno-19-8" href="#__codelineno-19-8"></a><span class="n">nplot.pp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">2</span><span class="w">  </span><span class="c1"># number of plots per page</span>
</span><span id="__span-19-9"><a id="__codelineno-19-9" name="__codelineno-19-9" href="#__codelineno-19-9"></a><span class="n">ncol.pp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">2</span><span class="w">  </span><span class="c1"># number of columns in a page</span>
</span><span id="__span-19-10"><a id="__codelineno-19-10" name="__codelineno-19-10" href="#__codelineno-19-10"></a><span class="n">fig</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">foreach</span><span class="p">(</span><span class="n">i</span><span class="o">=</span><span class="nf">seq</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="nf">unlist</span><span class="p">(</span><span class="n">err.plots</span><span class="p">,</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)),</span><span class="w"> </span><span class="n">by</span><span class="o">=</span><span class="n">nplot.pp</span><span class="p">),</span><span class="w"> </span><span class="n">.packages</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;ggpubr&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">%dopar%</span><span class="w"> </span><span class="p">{</span>
</span><span id="__span-19-11"><a id="__codelineno-19-11" name="__codelineno-19-11" href="#__codelineno-19-11"></a><span class="nf">ggarrange</span><span class="p">(</span><span class="n">plotlist</span><span class="o">=</span><span class="nf">unlist</span><span class="p">(</span><span class="n">err.plots</span><span class="p">,</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)[</span><span class="n">i</span><span class="o">:</span><span class="p">(</span><span class="n">i</span><span class="o">+</span><span class="n">nplot.pp</span><span class="m">-1</span><span class="p">)],</span><span class="w"> </span><span class="n">ncol</span><span class="o">=</span><span class="n">ncol.pp</span><span class="p">,</span><span class="w"> </span><span class="n">nrow</span><span class="o">=</span><span class="n">nplot.pp</span><span class="o">/</span><span class="n">ncol.pp</span><span class="p">)</span>
</span><span id="__span-19-12"><a id="__codelineno-19-12" name="__codelineno-19-12" href="#__codelineno-19-12"></a><span class="p">}</span>
</span><span id="__span-19-13"><a id="__codelineno-19-13" name="__codelineno-19-13" href="#__codelineno-19-13"></a><span class="nf">invisible</span><span class="p">(</span><span class="nf">lapply</span><span class="p">(</span><span class="n">fig</span><span class="p">,</span><span class="w"> </span><span class="n">print</span><span class="p">))</span>
</span></code></pre></div>
<p>Transitions (Aâ†’C, Aâ†’G, â€¦) are shown. Points are the observed error rates for each consensus quality score. The black line shows the estimated error rates after convergence of the machine-learning algorithm. The red line shows the error rates expected under the nominal definition of the Q-score. Here the estimated error rates (black line) are a good fit to the observed rates (points), and the error rates drop with increased quality as expected.</p>
<div class="language-r highlight"><span class="filename">Save the error model summary to PDF</span><pre><span></span><code><span id="__span-20-1"><a id="__codelineno-20-1" name="__codelineno-20-1" href="#__codelineno-20-1"></a><span class="c1"># Store the error model summary</span>
</span><span id="__span-20-2"><a id="__codelineno-20-2" name="__codelineno-20-2" href="#__codelineno-20-2"></a><span class="nf">pdf</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;figures&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;error_model.pdf&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">12</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">6</span><span class="p">)</span>
</span><span id="__span-20-3"><a id="__codelineno-20-3" name="__codelineno-20-3" href="#__codelineno-20-3"></a><span class="nf">invisible</span><span class="p">(</span><span class="nf">lapply</span><span class="p">(</span><span class="n">fig</span><span class="p">,</span><span class="w"> </span><span class="n">print</span><span class="p">))</span>
</span><span id="__span-20-4"><a id="__codelineno-20-4" name="__codelineno-20-4" href="#__codelineno-20-4"></a><span class="nf">invisible</span><span class="p">(</span><span class="nf">dev.off</span><span class="p">())</span>
</span></code></pre></div>
</details>
<h4 id="count-table-generation">Count table generation ğŸ§®</h4>
<p>A table with amplicon sequence variants is constructed. To avoid overloading memory, runs and samples are processed sequentialy.
The process starts with sequences dereplication, then it goes through Amplicon Sequence Variants (ASVs) inference and ends with Paired-Ends (PE) merging.</p>
<ul>
<li><strong>Dereplication</strong> combines all identical sequencing reads into into â€œunique sequencesâ€ with a corresponding â€œabundanceâ€ equal to the number of reads with that unique sequence. Dereplication in the DADA2 pipeline has one crucial addition from other pipelines: DADA2 retains a summary of the quality information associated with each unique sequence. The consensus quality profile of a unique sequence is the average of the positional qualities from the dereplicated reads. The consensus scores are then used by the error model of the dada function.</li>
<li><strong>Amplicon sequence variants (ASV) inference</strong> is the core stage of the DADA2 package (the <code>dada</code> function). It will assign all reads to an error-corrected sequence using the models of the error rates of the previous step.</li>
<li><strong>Paired-Ends (PE) merging</strong> performs a global ends-free alignment between paired forward and reverse reads and merges them together if they exactly overlap. It requires that the input forward and reverse reads are in the same order. Note that merging in the DADA2 pipeline happens after denoising, hence the strict requirement of exact overlap since it is expected that nearly all substitution errors have already been removed.</li>
</ul>
<details class="abstract">
<summary>Count table generation code</summary>
<div class="language-r highlight"><span class="filename">Generate the counts table</span><pre><span></span><code><span id="__span-21-1"><a id="__codelineno-21-1" name="__codelineno-21-1" href="#__codelineno-21-1"></a><span class="c1"># Recover variables</span>
</span><span id="__span-21-2"><a id="__codelineno-21-2" name="__codelineno-21-2" href="#__codelineno-21-2"></a><span class="n">runs.dirs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.dirs</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;run_data&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)</span>
</span><span id="__span-21-3"><a id="__codelineno-21-3" name="__codelineno-21-3" href="#__codelineno-21-3"></a><span class="n">runs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">basename</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">)</span>
</span><span id="__span-21-4"><a id="__codelineno-21-4" name="__codelineno-21-4" href="#__codelineno-21-4"></a><span class="n">err.model</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">foreach</span><span class="p">(</span><span class="n">i</span><span class="o">=</span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">runs</span><span class="p">))</span><span class="w"> </span><span class="o">%dopar%</span><span class="w"> </span><span class="p">{</span>
</span><span id="__span-21-5"><a id="__codelineno-21-5" name="__codelineno-21-5" href="#__codelineno-21-5"></a><span class="nf">readRDS</span><span class="p">(</span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;error_model.rds&#39;</span><span class="p">))</span>
</span><span id="__span-21-6"><a id="__codelineno-21-6" name="__codelineno-21-6" href="#__codelineno-21-6"></a><span class="p">}</span>
</span><span id="__span-21-7"><a id="__codelineno-21-7" name="__codelineno-21-7" href="#__codelineno-21-7"></a>
</span><span id="__span-21-8"><a id="__codelineno-21-8" name="__codelineno-21-8" href="#__codelineno-21-8"></a><span class="kr">for</span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">runs</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
</span><span id="__span-21-9"><a id="__codelineno-21-9" name="__codelineno-21-9" href="#__codelineno-21-9"></a>
</span><span id="__span-21-10"><a id="__codelineno-21-10" name="__codelineno-21-10" href="#__codelineno-21-10"></a><span class="n">fwd.fn</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sort</span><span class="p">(</span><span class="nf">list.files</span><span class="p">(</span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;filtered&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;R1.fastq&#39;</span><span class="p">))</span>
</span><span id="__span-21-11"><a id="__codelineno-21-11" name="__codelineno-21-11" href="#__codelineno-21-11"></a><span class="n">rev.fn</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sort</span><span class="p">(</span><span class="nf">list.files</span><span class="p">(</span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;filtered&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;R2.fastq&#39;</span><span class="p">))</span>
</span><span id="__span-21-12"><a id="__codelineno-21-12" name="__codelineno-21-12" href="#__codelineno-21-12"></a><span class="n">sample.names</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sapply</span><span class="p">(</span><span class="nf">strsplit</span><span class="p">(</span><span class="nf">basename</span><span class="p">(</span><span class="n">fwd.fn</span><span class="p">),</span><span class="w"> </span><span class="s">&#39;R1.fastq&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">`[`</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">)</span>
</span><span id="__span-21-13"><a id="__codelineno-21-13" name="__codelineno-21-13" href="#__codelineno-21-13"></a><span class="n">sample.names.rev</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sapply</span><span class="p">(</span><span class="nf">strsplit</span><span class="p">(</span><span class="nf">basename</span><span class="p">(</span><span class="n">rev.fn</span><span class="p">),</span><span class="w"> </span><span class="s">&#39;R2.fastq&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">`[`</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">)</span>
</span><span id="__span-21-14"><a id="__codelineno-21-14" name="__codelineno-21-14" href="#__codelineno-21-14"></a><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="nf">identical</span><span class="p">(</span><span class="n">sample.names</span><span class="p">,</span><span class="w"> </span><span class="n">sample.names.rev</span><span class="p">))</span><span class="w"> </span><span class="nf">stop</span><span class="p">(</span><span class="s">&#39;Forward and reverse files do not match.&#39;</span><span class="p">)</span>
</span><span id="__span-21-15"><a id="__codelineno-21-15" name="__codelineno-21-15" href="#__codelineno-21-15"></a><span class="nf">names</span><span class="p">(</span><span class="n">fwd.fn</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">sample.names</span>
</span><span id="__span-21-16"><a id="__codelineno-21-16" name="__codelineno-21-16" href="#__codelineno-21-16"></a><span class="nf">names</span><span class="p">(</span><span class="n">rev.fn</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">sample.names</span>
</span><span id="__span-21-17"><a id="__codelineno-21-17" name="__codelineno-21-17" href="#__codelineno-21-17"></a>
</span><span id="__span-21-18"><a id="__codelineno-21-18" name="__codelineno-21-18" href="#__codelineno-21-18"></a><span class="n">merged</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">vector</span><span class="p">(</span><span class="s">&#39;list&#39;</span><span class="p">,</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">sample.names</span><span class="p">))</span>
</span><span id="__span-21-19"><a id="__codelineno-21-19" name="__codelineno-21-19" href="#__codelineno-21-19"></a><span class="nf">names</span><span class="p">(</span><span class="n">merged</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">sample.names</span>
</span><span id="__span-21-20"><a id="__codelineno-21-20" name="__codelineno-21-20" href="#__codelineno-21-20"></a><span class="kr">for</span><span class="p">(</span><span class="n">j</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">sample.names</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
</span><span id="__span-21-21"><a id="__codelineno-21-21" name="__codelineno-21-21" href="#__codelineno-21-21"></a><span class="w">    </span><span class="n">derep</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">vector</span><span class="p">(</span><span class="s">&#39;list&#39;</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">)</span>
</span><span id="__span-21-22"><a id="__codelineno-21-22" name="__codelineno-21-22" href="#__codelineno-21-22"></a><span class="w">    </span><span class="n">derep</span><span class="p">[[</span><span class="m">1</span><span class="p">]]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">derepFastq</span><span class="p">(</span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;filtered&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">fwd.fn</span><span class="p">[</span><span class="n">j</span><span class="p">]))</span>
</span><span id="__span-21-23"><a id="__codelineno-21-23" name="__codelineno-21-23" href="#__codelineno-21-23"></a><span class="w">    </span><span class="n">derep</span><span class="p">[[</span><span class="m">2</span><span class="p">]]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">derepFastq</span><span class="p">(</span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;filtered&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">rev.fn</span><span class="p">[</span><span class="n">j</span><span class="p">]))</span>
</span><span id="__span-21-24"><a id="__codelineno-21-24" name="__codelineno-21-24" href="#__codelineno-21-24"></a><span class="w">    </span><span class="n">asv</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">vector</span><span class="p">(</span><span class="s">&#39;list&#39;</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">)</span>
</span><span id="__span-21-25"><a id="__codelineno-21-25" name="__codelineno-21-25" href="#__codelineno-21-25"></a><span class="w">    </span><span class="n">asv</span><span class="p">[[</span><span class="m">1</span><span class="p">]]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">dada</span><span class="p">(</span><span class="n">derep</span><span class="p">[[</span><span class="m">1</span><span class="p">]],</span><span class="w"> </span><span class="n">err</span><span class="o">=</span><span class="n">err.model</span><span class="p">[[</span><span class="n">i</span><span class="p">]][[</span><span class="m">1</span><span class="p">]],</span><span class="w"> </span><span class="n">pool</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">multithread</span><span class="o">=</span><span class="n">nc</span><span class="p">)</span>
</span><span id="__span-21-26"><a id="__codelineno-21-26" name="__codelineno-21-26" href="#__codelineno-21-26"></a><span class="w">    </span><span class="n">asv</span><span class="p">[[</span><span class="m">2</span><span class="p">]]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">dada</span><span class="p">(</span><span class="n">derep</span><span class="p">[[</span><span class="m">2</span><span class="p">]],</span><span class="w"> </span><span class="n">err</span><span class="o">=</span><span class="n">err.model</span><span class="p">[[</span><span class="n">i</span><span class="p">]][[</span><span class="m">2</span><span class="p">]],</span><span class="w"> </span><span class="n">pool</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">multithread</span><span class="o">=</span><span class="n">nc</span><span class="p">)</span>
</span><span id="__span-21-27"><a id="__codelineno-21-27" name="__codelineno-21-27" href="#__codelineno-21-27"></a><span class="w">    </span><span class="n">merged</span><span class="p">[[</span><span class="n">sample.names</span><span class="p">[</span><span class="n">j</span><span class="p">]]]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">mergePairs</span><span class="p">(</span><span class="n">asv</span><span class="p">[[</span><span class="m">1</span><span class="p">]],</span><span class="w"> </span><span class="n">derep</span><span class="p">[[</span><span class="m">1</span><span class="p">]],</span><span class="w"> </span><span class="n">asv</span><span class="p">[[</span><span class="m">2</span><span class="p">]],</span><span class="w"> </span><span class="n">derep</span><span class="p">[[</span><span class="m">2</span><span class="p">]])</span>
</span><span id="__span-21-28"><a id="__codelineno-21-28" name="__codelineno-21-28" href="#__codelineno-21-28"></a><span class="p">}</span>
</span><span id="__span-21-29"><a id="__codelineno-21-29" name="__codelineno-21-29" href="#__codelineno-21-29"></a>
</span><span id="__span-21-30"><a id="__codelineno-21-30" name="__codelineno-21-30" href="#__codelineno-21-30"></a><span class="n">st</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">makeSequenceTable</span><span class="p">(</span><span class="n">merged</span><span class="p">)</span>
</span><span id="__span-21-31"><a id="__codelineno-21-31" name="__codelineno-21-31" href="#__codelineno-21-31"></a><span class="nf">saveRDS</span><span class="p">(</span><span class="n">st</span><span class="p">,</span><span class="w"> </span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;seqtab.rds&#39;</span><span class="p">))</span>
</span><span id="__span-21-32"><a id="__codelineno-21-32" name="__codelineno-21-32" href="#__codelineno-21-32"></a><span class="p">}</span>
</span></code></pre></div>
<div class="admonition info">
<p class="admonition-title">Info</p>
<p>Most of your reads should successfully merge. If that is not the case, upstream parameters may need to be revisited.</p>
</div>
</details>
<h4 id="merge-runs">Merge runs ğŸ¤ğŸ§ª</h4>
<p>At this point, all of the individual per-run <code>seqtab</code> tables are merged into a single global <code>seqtab</code>.</p>
<details class="abstract">
<summary>Merge runs code</summary>
<div class="language-r highlight"><span class="filename">Merge runs</span><pre><span></span><code><span id="__span-22-1"><a id="__codelineno-22-1" name="__codelineno-22-1" href="#__codelineno-22-1"></a><span class="c1"># Recover variables</span>
</span><span id="__span-22-2"><a id="__codelineno-22-2" name="__codelineno-22-2" href="#__codelineno-22-2"></a><span class="n">runs.dirs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.dirs</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;run_data&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)</span>
</span><span id="__span-22-3"><a id="__codelineno-22-3" name="__codelineno-22-3" href="#__codelineno-22-3"></a>
</span><span id="__span-22-4"><a id="__codelineno-22-4" name="__codelineno-22-4" href="#__codelineno-22-4"></a><span class="n">seqtab.fps</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;seqtab.rds&#39;</span><span class="p">)</span>
</span><span id="__span-22-5"><a id="__codelineno-22-5" name="__codelineno-22-5" href="#__codelineno-22-5"></a><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">seqtab.fps</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
</span><span id="__span-22-6"><a id="__codelineno-22-6" name="__codelineno-22-6" href="#__codelineno-22-6"></a><span class="n">seqtab</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="n">seqtab.fps</span><span class="p">[[</span><span class="m">1</span><span class="p">]])</span>
</span><span id="__span-22-7"><a id="__codelineno-22-7" name="__codelineno-22-7" href="#__codelineno-22-7"></a><span class="p">}</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="p">{</span>
</span><span id="__span-22-8"><a id="__codelineno-22-8" name="__codelineno-22-8" href="#__codelineno-22-8"></a><span class="n">seqtab</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">mergeSequenceTables</span><span class="p">(</span><span class="n">tables</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">seqtab.fps</span><span class="p">)</span>
</span><span id="__span-22-9"><a id="__codelineno-22-9" name="__codelineno-22-9" href="#__codelineno-22-9"></a><span class="p">}</span>
</span><span id="__span-22-10"><a id="__codelineno-22-10" name="__codelineno-22-10" href="#__codelineno-22-10"></a>
</span><span id="__span-22-11"><a id="__codelineno-22-11" name="__codelineno-22-11" href="#__codelineno-22-11"></a><span class="c1"># Save data into a new directory named &#39;data&#39;</span>
</span><span id="__span-22-12"><a id="__codelineno-22-12" name="__codelineno-22-12" href="#__codelineno-22-12"></a><span class="nf">saveRDS</span><span class="p">(</span><span class="n">seqtab</span><span class="p">,</span><span class="w"> </span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;seqtab.rds&#39;</span><span class="p">))</span>
</span></code></pre></div>
</details>
<h4 id="chimera-screening">Chimera screening ğŸ¦„</h4>
<p>The <code>dada</code> algorithm models and removes substitution errors, but chimeras are another importance source of spurious sequences in amplicon sequencing. </p>
<div class="admonition example">
<p class="admonition-title">What are chimeras?</p>
<p>Chimeras are artifacts generated during PCR amplification when an incomplete amplicon acts as a primer for a subsequent extension step. This occurs when a partially synthesized DNA fragment anneals to a different template in later cycles, leading to the formation of a hybrid sequence. As a result, the final amplicon consists of segments from two distinct parental sequences, producing a misleading chimeric read that does not represent a true biological variant.</p>
</div>
<details class="abstract">
<summary>Chimera screening code</summary>
<div class="language-r highlight"><span class="filename">Screen and remove chimeric reads</span><pre><span></span><code><span id="__span-23-1"><a id="__codelineno-23-1" name="__codelineno-23-1" href="#__codelineno-23-1"></a><span class="c1"># Recover variables</span>
</span><span id="__span-23-2"><a id="__codelineno-23-2" name="__codelineno-23-2" href="#__codelineno-23-2"></a><span class="n">seqtab</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;seqtab.rds&#39;</span><span class="p">))</span>
</span><span id="__span-23-3"><a id="__codelineno-23-3" name="__codelineno-23-3" href="#__codelineno-23-3"></a>
</span><span id="__span-23-4"><a id="__codelineno-23-4" name="__codelineno-23-4" href="#__codelineno-23-4"></a><span class="c1"># Remove chimeric reads</span>
</span><span id="__span-23-5"><a id="__codelineno-23-5" name="__codelineno-23-5" href="#__codelineno-23-5"></a><span class="n">seqtab_nochim</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">removeBimeraDenovo</span><span class="p">(</span><span class="n">seqtab</span><span class="p">,</span><span class="w"> </span><span class="n">method</span><span class="o">=</span><span class="s">&#39;per-sample&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">multithread</span><span class="o">=</span><span class="n">nc</span><span class="p">,</span><span class="w"> </span><span class="n">verbose</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
</span><span id="__span-23-6"><a id="__codelineno-23-6" name="__codelineno-23-6" href="#__codelineno-23-6"></a><span class="nf">saveRDS</span><span class="p">(</span><span class="n">seqtab_nochim</span><span class="p">,</span><span class="w"> </span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;seqtab_nochim.rds&#39;</span><span class="p">))</span>
</span><span id="__span-23-7"><a id="__codelineno-23-7" name="__codelineno-23-7" href="#__codelineno-23-7"></a><span class="nf">fwrite</span><span class="p">(</span><span class="nf">as.data.frame</span><span class="p">(</span><span class="n">seqtab_nochim</span><span class="p">),</span><span class="w"> </span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;seqtab_nochim.txt&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;\t&#39;</span><span class="p">)</span>
</span><span id="__span-23-8"><a id="__codelineno-23-8" name="__codelineno-23-8" href="#__codelineno-23-8"></a>
</span><span id="__span-23-9"><a id="__codelineno-23-9" name="__codelineno-23-9" href="#__codelineno-23-9"></a><span class="c1"># Inspect distribution of sequence lengths after chimera removal</span>
</span><span id="__span-23-10"><a id="__codelineno-23-10" name="__codelineno-23-10" href="#__codelineno-23-10"></a><span class="n">distrib</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">table</span><span class="p">(</span><span class="nf">nchar</span><span class="p">(</span><span class="nf">getSequences</span><span class="p">(</span><span class="n">seqtab_nochim</span><span class="p">)))</span>
</span><span id="__span-23-11"><a id="__codelineno-23-11" name="__codelineno-23-11" href="#__codelineno-23-11"></a><span class="n">distrib_plot</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kr">function</span><span class="p">(){</span>
</span><span id="__span-23-12"><a id="__codelineno-23-12" name="__codelineno-23-12" href="#__codelineno-23-12"></a><span class="nf">plot</span><span class="p">(</span><span class="n">distrib</span><span class="p">,</span><span class="w"> </span><span class="n">xlab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;Read length&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">ylab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;Number of ASVs&#39;</span><span class="p">)</span>
</span><span id="__span-23-13"><a id="__codelineno-23-13" name="__codelineno-23-13" href="#__codelineno-23-13"></a><span class="p">}</span>
</span><span id="__span-23-14"><a id="__codelineno-23-14" name="__codelineno-23-14" href="#__codelineno-23-14"></a><span class="nf">saveRDS</span><span class="p">(</span><span class="n">distrib</span><span class="p">,</span><span class="w"> </span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;length_distribution.rds&#39;</span><span class="p">))</span>
</span><span id="__span-23-15"><a id="__codelineno-23-15" name="__codelineno-23-15" href="#__codelineno-23-15"></a><span class="nf">pdf</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;figures&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;length_distribution.pdf&#39;</span><span class="p">))</span>
</span><span id="__span-23-16"><a id="__codelineno-23-16" name="__codelineno-23-16" href="#__codelineno-23-16"></a><span class="nf">distrib_plot</span><span class="p">()</span>
</span><span id="__span-23-17"><a id="__codelineno-23-17" name="__codelineno-23-17" href="#__codelineno-23-17"></a><span class="nf">invisible</span><span class="p">(</span><span class="nf">dev.off</span><span class="p">())</span>
</span></code></pre></div>
<div class="language-r highlight"><span class="filename">Check the post-chimera removal dimensions and inspect length distributions</span><pre><span></span><code><span id="__span-24-1"><a id="__codelineno-24-1" name="__codelineno-24-1" href="#__codelineno-24-1"></a><span class="c1"># Recover variables</span>
</span><span id="__span-24-2"><a id="__codelineno-24-2" name="__codelineno-24-2" href="#__codelineno-24-2"></a><span class="n">seqtab</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;seqtab.rds&#39;</span><span class="p">))</span>
</span><span id="__span-24-3"><a id="__codelineno-24-3" name="__codelineno-24-3" href="#__codelineno-24-3"></a><span class="n">seqtab_nochim</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;seqtab_nochim.rds&#39;</span><span class="p">))</span>
</span><span id="__span-24-4"><a id="__codelineno-24-4" name="__codelineno-24-4" href="#__codelineno-24-4"></a><span class="n">distrib</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;length_distribution.rds&#39;</span><span class="p">))</span>
</span><span id="__span-24-5"><a id="__codelineno-24-5" name="__codelineno-24-5" href="#__codelineno-24-5"></a><span class="n">distrib.plot</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kr">function</span><span class="p">(){</span>
</span><span id="__span-24-6"><a id="__codelineno-24-6" name="__codelineno-24-6" href="#__codelineno-24-6"></a><span class="nf">plot</span><span class="p">(</span><span class="n">distrib</span><span class="p">,</span><span class="w"> </span><span class="n">xlab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;Read length&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">ylab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;Number of ASVs&#39;</span><span class="p">)</span>
</span><span id="__span-24-7"><a id="__codelineno-24-7" name="__codelineno-24-7" href="#__codelineno-24-7"></a><span class="p">}</span>
</span><span id="__span-24-8"><a id="__codelineno-24-8" name="__codelineno-24-8" href="#__codelineno-24-8"></a>
</span><span id="__span-24-9"><a id="__codelineno-24-9" name="__codelineno-24-9" href="#__codelineno-24-9"></a><span class="c1"># Check the dimensions of the table before chimera removal</span>
</span><span id="__span-24-10"><a id="__codelineno-24-10" name="__codelineno-24-10" href="#__codelineno-24-10"></a><span class="nf">dim</span><span class="p">(</span><span class="n">seqtab</span><span class="p">)</span>
</span><span id="__span-24-11"><a id="__codelineno-24-11" name="__codelineno-24-11" href="#__codelineno-24-11"></a>
</span><span id="__span-24-12"><a id="__codelineno-24-12" name="__codelineno-24-12" href="#__codelineno-24-12"></a><span class="c1"># Check the dimensions of the table after chimera removal</span>
</span><span id="__span-24-13"><a id="__codelineno-24-13" name="__codelineno-24-13" href="#__codelineno-24-13"></a><span class="nf">dim</span><span class="p">(</span><span class="n">seqtab_nochim</span><span class="p">)</span>
</span><span id="__span-24-14"><a id="__codelineno-24-14" name="__codelineno-24-14" href="#__codelineno-24-14"></a>
</span><span id="__span-24-15"><a id="__codelineno-24-15" name="__codelineno-24-15" href="#__codelineno-24-15"></a><span class="c1"># Check the distribution of sequence lengths</span>
</span><span id="__span-24-16"><a id="__codelineno-24-16" name="__codelineno-24-16" href="#__codelineno-24-16"></a><span class="nf">distrib.plot</span><span class="p">()</span>
</span></code></pre></div>
</details>
<h4 id="reads-tracking">Reads tracking ğŸ“ŠğŸ›¤ï¸</h4>
<p>As a final check of our progress, we look at the number of reads that made it through each step in the pipeline. </p>
<div class="admonition warning">
<p class="admonition-title">Double check the retention of reads</p>
<p>Outside of filtering (first step) there should be no step in which a majority of reads are lost. If a majority of reads failed to merge, you may need to revisit the truncLen parameter used in the filtering step and make sure that the truncated reads span your amplicon. If a majority of reads were removed as chimeric, you may need to revisit the removal of primers, as the ambiguous nucleotides in unremoved primers interfere with chimera identification.</p>
</div>
<details class="abstract">
<summary>Reads tracking code</summary>
<div class="language-r highlight"><span class="filename">Generate reads tracking plot and save to PDF</span><pre><span></span><code><span id="__span-25-1"><a id="__codelineno-25-1" name="__codelineno-25-1" href="#__codelineno-25-1"></a><span class="c1"># Recover variables</span>
</span><span id="__span-25-2"><a id="__codelineno-25-2" name="__codelineno-25-2" href="#__codelineno-25-2"></a><span class="n">seqtab</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;seqtab.rds&#39;</span><span class="p">))</span>
</span><span id="__span-25-3"><a id="__codelineno-25-3" name="__codelineno-25-3" href="#__codelineno-25-3"></a><span class="n">seqtab_nochim</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;seqtab_nochim.rds&#39;</span><span class="p">))</span>
</span><span id="__span-25-4"><a id="__codelineno-25-4" name="__codelineno-25-4" href="#__codelineno-25-4"></a>
</span><span id="__span-25-5"><a id="__codelineno-25-5" name="__codelineno-25-5" href="#__codelineno-25-5"></a><span class="n">track.plots</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">foreach</span><span class="p">(</span><span class="n">i</span><span class="o">=</span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">runs</span><span class="p">),</span><span class="w"> </span><span class="n">.packages</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;ggplot2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;reshape2&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">%do%</span><span class="w"> </span><span class="p">{</span>
</span><span id="__span-25-6"><a id="__codelineno-25-6" name="__codelineno-25-6" href="#__codelineno-25-6"></a><span class="n">filtering</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;filtering_report.rds&#39;</span><span class="p">))</span>
</span><span id="__span-25-7"><a id="__codelineno-25-7" name="__codelineno-25-7" href="#__codelineno-25-7"></a><span class="nf">row.names</span><span class="p">(</span><span class="n">filtering</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">gsub</span><span class="p">(</span><span class="s">&#39;_R1_001.fastq|-R1.fastq&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;&#39;</span><span class="p">,</span><span class="w"> </span><span class="nf">row.names</span><span class="p">(</span><span class="n">filtering</span><span class="p">))</span>
</span><span id="__span-25-8"><a id="__codelineno-25-8" name="__codelineno-25-8" href="#__codelineno-25-8"></a><span class="n">track</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">cbind</span><span class="p">(</span><span class="n">filtering</span><span class="p">[</span><span class="nf">row.names</span><span class="p">(</span><span class="n">filtering</span><span class="p">)</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">row.names</span><span class="p">(</span><span class="n">seqtab</span><span class="p">),],</span>
</span><span id="__span-25-9"><a id="__codelineno-25-9" name="__codelineno-25-9" href="#__codelineno-25-9"></a><span class="w">                </span><span class="nf">rowSums</span><span class="p">(</span><span class="n">seqtab</span><span class="p">[</span><span class="nf">row.names</span><span class="p">(</span><span class="n">seqtab</span><span class="p">)</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">row.names</span><span class="p">(</span><span class="n">filtering</span><span class="p">),</span><span class="w"> </span><span class="p">]),</span>
</span><span id="__span-25-10"><a id="__codelineno-25-10" name="__codelineno-25-10" href="#__codelineno-25-10"></a><span class="w">                </span><span class="nf">rowSums</span><span class="p">(</span><span class="n">seqtab_nochim</span><span class="p">[</span><span class="nf">row.names</span><span class="p">(</span><span class="n">seqtab_nochim</span><span class="p">)</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">row.names</span><span class="p">(</span><span class="n">filtering</span><span class="p">),</span><span class="w"> </span><span class="p">]))</span>
</span><span id="__span-25-11"><a id="__codelineno-25-11" name="__codelineno-25-11" href="#__codelineno-25-11"></a><span class="nf">colnames</span><span class="p">(</span><span class="n">track</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;Input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;Filtered&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;Merged&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;Non chimeric&#39;</span><span class="p">)</span>
</span><span id="__span-25-12"><a id="__codelineno-25-12" name="__codelineno-25-12" href="#__codelineno-25-12"></a><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">j</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="p">(</span><span class="nf">ncol</span><span class="p">(</span><span class="n">track</span><span class="p">)</span><span class="m">-1</span><span class="p">)</span><span class="o">:</span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
</span><span id="__span-25-13"><a id="__codelineno-25-13" name="__codelineno-25-13" href="#__codelineno-25-13"></a><span class="w">    </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">k</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="p">(</span><span class="n">j</span><span class="m">+1</span><span class="p">)</span><span class="o">:</span><span class="nf">ncol</span><span class="p">(</span><span class="n">track</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
</span><span id="__span-25-14"><a id="__codelineno-25-14" name="__codelineno-25-14" href="#__codelineno-25-14"></a><span class="w">    </span><span class="n">track</span><span class="p">[,</span><span class="w"> </span><span class="n">j</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">track</span><span class="p">[,</span><span class="w"> </span><span class="n">j</span><span class="p">]</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">track</span><span class="p">[,</span><span class="w"> </span><span class="n">k</span><span class="p">]</span>
</span><span id="__span-25-15"><a id="__codelineno-25-15" name="__codelineno-25-15" href="#__codelineno-25-15"></a><span class="w">    </span><span class="p">}</span>
</span><span id="__span-25-16"><a id="__codelineno-25-16" name="__codelineno-25-16" href="#__codelineno-25-16"></a><span class="p">}</span>
</span><span id="__span-25-17"><a id="__codelineno-25-17" name="__codelineno-25-17" href="#__codelineno-25-17"></a><span class="n">p</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="nf">melt</span><span class="p">(</span><span class="nf">as.matrix</span><span class="p">(</span><span class="n">track</span><span class="p">)),</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">Var1</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="n">value</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="o">=</span><span class="n">Var2</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
</span><span id="__span-25-18"><a id="__codelineno-25-18" name="__codelineno-25-18" href="#__codelineno-25-18"></a><span class="w">    </span><span class="nf">geom_bar</span><span class="p">(</span><span class="n">stat</span><span class="o">=</span><span class="s">&#39;identity&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
</span><span id="__span-25-19"><a id="__codelineno-25-19" name="__codelineno-25-19" href="#__codelineno-25-19"></a><span class="w">    </span><span class="nf">theme</span><span class="p">(</span><span class="n">axis.text.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">angle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">90</span><span class="p">,</span><span class="w"> </span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">vjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">4</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
</span><span id="__span-25-20"><a id="__codelineno-25-20" name="__codelineno-25-20" href="#__codelineno-25-20"></a><span class="w">    </span><span class="nf">labs</span><span class="p">(</span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">runs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;Samples&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;Reads&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">)</span>
</span><span id="__span-25-21"><a id="__codelineno-25-21" name="__codelineno-25-21" href="#__codelineno-25-21"></a><span class="nf">saveRDS</span><span class="p">(</span><span class="n">p</span><span class="p">,</span><span class="w"> </span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;read_tracking_report.pdf.rds&#39;</span><span class="p">))</span>
</span><span id="__span-25-22"><a id="__codelineno-25-22" name="__codelineno-25-22" href="#__codelineno-25-22"></a><span class="nf">pdf</span><span class="p">(</span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;read_tracking_report.pdf&#39;</span><span class="p">))</span>
</span><span id="__span-25-23"><a id="__codelineno-25-23" name="__codelineno-25-23" href="#__codelineno-25-23"></a><span class="nf">print</span><span class="p">(</span><span class="n">p</span><span class="p">)</span>
</span><span id="__span-25-24"><a id="__codelineno-25-24" name="__codelineno-25-24" href="#__codelineno-25-24"></a><span class="nf">invisible</span><span class="p">(</span><span class="nf">dev.off</span><span class="p">())</span>
</span><span id="__span-25-25"><a id="__codelineno-25-25" name="__codelineno-25-25" href="#__codelineno-25-25"></a><span class="n">p</span>
</span><span id="__span-25-26"><a id="__codelineno-25-26" name="__codelineno-25-26" href="#__codelineno-25-26"></a><span class="p">}</span>
</span><span id="__span-25-27"><a id="__codelineno-25-27" name="__codelineno-25-27" href="#__codelineno-25-27"></a><span class="nf">pdf</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;figures&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;read_tracking_report.pdf&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">8</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">6</span><span class="p">)</span>
</span><span id="__span-25-28"><a id="__codelineno-25-28" name="__codelineno-25-28" href="#__codelineno-25-28"></a><span class="nf">invisible</span><span class="p">(</span><span class="nf">lapply</span><span class="p">(</span><span class="n">track.plots</span><span class="p">,</span><span class="w"> </span><span class="n">print</span><span class="p">))</span>
</span><span id="__span-25-29"><a id="__codelineno-25-29" name="__codelineno-25-29" href="#__codelineno-25-29"></a><span class="nf">invisible</span><span class="p">(</span><span class="nf">dev.off</span><span class="p">())</span>
</span></code></pre></div>
<div class="language-r highlight"><span class="filename">View the reads tracking plot</span><pre><span></span><code><span id="__span-26-1"><a id="__codelineno-26-1" name="__codelineno-26-1" href="#__codelineno-26-1"></a><span class="c1"># Recover variables</span>
</span><span id="__span-26-2"><a id="__codelineno-26-2" name="__codelineno-26-2" href="#__codelineno-26-2"></a><span class="n">runs.dirs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.dirs</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;run_data&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)</span>
</span><span id="__span-26-3"><a id="__codelineno-26-3" name="__codelineno-26-3" href="#__codelineno-26-3"></a><span class="n">track.plots</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">foreach</span><span class="p">(</span><span class="n">i</span><span class="o">=</span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">))</span><span class="w"> </span><span class="o">%dopar%</span><span class="w"> </span><span class="p">{</span>
</span><span id="__span-26-4"><a id="__codelineno-26-4" name="__codelineno-26-4" href="#__codelineno-26-4"></a><span class="nf">readRDS</span><span class="p">(</span><span class="nf">file.path</span><span class="p">(</span><span class="n">runs.dirs</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&#39;read_tracking_report.pdf.rds&#39;</span><span class="p">))</span>
</span><span id="__span-26-5"><a id="__codelineno-26-5" name="__codelineno-26-5" href="#__codelineno-26-5"></a><span class="p">}</span>
</span><span id="__span-26-6"><a id="__codelineno-26-6" name="__codelineno-26-6" href="#__codelineno-26-6"></a>
</span><span id="__span-26-7"><a id="__codelineno-26-7" name="__codelineno-26-7" href="#__codelineno-26-7"></a><span class="nf">invisible</span><span class="p">(</span><span class="nf">lapply</span><span class="p">(</span><span class="n">track.plots</span><span class="p">,</span><span class="w"> </span><span class="n">print</span><span class="p">))</span>
</span></code></pre></div>
</details>
<h4 id="taxonomy-assignment">Taxonomy Assignment ğŸ§¬ğŸ·ï¸</h4>
<p>The DADA2 package provides a native implementation of the naive Bayesian classifier method for this purpose. 
The <code>assignTaxonomy</code> function takes as input a set of sequences to be classified and a training set of reference sequences with known taxonomy, and outputs taxonomic assignments with at least minBoot bootstrap confidence.</p>
<div class="admonition tip">
<p class="admonition-title">Update your SILVA database version</p>
<p>As of SILVA version 138.2, the DADA2-formatted version can assign to the species level in a single step. Please update your database file if you are using an older version.</p>
</div>
<details class="abstract">
<summary>Taxonomy assignment code</summary>
<div class="language-r highlight"><span class="filename">Assign taxonomy to the species level</span><pre><span></span><code><span id="__span-27-1"><a id="__codelineno-27-1" name="__codelineno-27-1" href="#__codelineno-27-1"></a><span class="c1"># Recover variables</span>
</span><span id="__span-27-2"><a id="__codelineno-27-2" name="__codelineno-27-2" href="#__codelineno-27-2"></a><span class="n">seqtab_nochim</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;seqtab_nochim.rds&#39;</span><span class="p">))</span>
</span><span id="__span-27-3"><a id="__codelineno-27-3" name="__codelineno-27-3" href="#__codelineno-27-3"></a>
</span><span id="__span-27-4"><a id="__codelineno-27-4" name="__codelineno-27-4" href="#__codelineno-27-4"></a><span class="c1"># Path to the DADA2-formatted reference database</span>
</span><span id="__span-27-5"><a id="__codelineno-27-5" name="__codelineno-27-5" href="#__codelineno-27-5"></a><span class="c1"># Replace the existing path with the path to your SILVA training set</span>
</span><span id="__span-27-6"><a id="__codelineno-27-6" name="__codelineno-27-6" href="#__codelineno-27-6"></a><span class="n">db_fp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;..&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;..&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;20_Databases&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;Silva&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;silva_nr99_v138.2_toSpecies_trainset.fa.gz&#39;</span><span class="p">)</span>
</span><span id="__span-27-7"><a id="__codelineno-27-7" name="__codelineno-27-7" href="#__codelineno-27-7"></a>
</span><span id="__span-27-8"><a id="__codelineno-27-8" name="__codelineno-27-8" href="#__codelineno-27-8"></a><span class="n">taxonomy</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">assignTaxonomy</span><span class="p">(</span><span class="n">seqtab_nochim</span><span class="p">,</span><span class="w"> </span><span class="n">db_fp</span><span class="p">,</span><span class="w"> </span><span class="n">minBoot</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">50</span><span class="p">,</span><span class="w"> </span><span class="n">tryRC</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">multithread</span><span class="o">=</span><span class="n">nc</span><span class="p">)</span>
</span><span id="__span-27-9"><a id="__codelineno-27-9" name="__codelineno-27-9" href="#__codelineno-27-9"></a><span class="nf">saveRDS</span><span class="p">(</span><span class="n">taxonomy</span><span class="p">,</span><span class="w"> </span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;taxonomy_species.rds&#39;</span><span class="p">))</span>
</span><span id="__span-27-10"><a id="__codelineno-27-10" name="__codelineno-27-10" href="#__codelineno-27-10"></a><span class="nf">fwrite</span><span class="p">(</span><span class="nf">as.data.frame</span><span class="p">(</span><span class="n">taxonomy</span><span class="p">),</span><span class="w"> </span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;taxonomy_species.txt&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;\t&#39;</span><span class="p">)</span>
</span></code></pre></div>
<div class="admonition info">
<p class="admonition-title">Trying the reverse complement</p>
<p>The <code>tryRC = TRUE</code> option ensures that both the original sequences and their reverse complements are considered when matching against the reference database. This is useful when the sequencing orientation is unknown or inconsistent, helping to improve taxonomic assignment accuracy by allowing for potential reverse-complement matches.</p>
</div>
</details>
<h4 id="export-in-qiime-classic-otu-table-like-format">Export in Qiime classic OTU table-like format ğŸ’¾ğŸ” </h4>
<p>The DADA2 pipeline provides results as a count table of ASVs per samples and a taxonomic classification of each ASV in two separate files. As detailed in the <a href="https://benjjneb.github.io/dada2/tutorial.html">DADA2 tutorial</a>, these two objects can easily be used with the phyloseq R package for subsequent data analysis.</p>
<ul>
<li>For compatibility with other data analysis tools, a count table in a tab-delimited text format matching the <a href="https://www.drive5.com/usearch/manual/qiime_classic.html">Qiime classic OTU table</a> format is also created.</li>
<li>The table contains samples in columns and ASVs in rows. The taxonomy at the species level is added as an extra 'taxonomy' column as well as a 'sequence' column. The first columns contains mock OTU IDs.</li>
</ul>
<details class="abstract">
<summary>Qiime-style OTU table export code</summary>
<div class="language-r highlight"><span class="filename">Export OTU table in classic Qiime format</span><pre><span></span><code><span id="__span-28-1"><a id="__codelineno-28-1" name="__codelineno-28-1" href="#__codelineno-28-1"></a><span class="c1"># Recover variables</span>
</span><span id="__span-28-2"><a id="__codelineno-28-2" name="__codelineno-28-2" href="#__codelineno-28-2"></a><span class="n">seqtab_nochim</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;seqtab_nochim.rds&#39;</span><span class="p">))</span>
</span><span id="__span-28-3"><a id="__codelineno-28-3" name="__codelineno-28-3" href="#__codelineno-28-3"></a><span class="n">taxonomy_species</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;taxonomy_species.rds&#39;</span><span class="p">))</span>
</span><span id="__span-28-4"><a id="__codelineno-28-4" name="__codelineno-28-4" href="#__codelineno-28-4"></a>
</span><span id="__span-28-5"><a id="__codelineno-28-5" name="__codelineno-28-5" href="#__codelineno-28-5"></a><span class="c1"># Define a function to create the classic OTU table</span>
</span><span id="__span-28-6"><a id="__codelineno-28-6" name="__codelineno-28-6" href="#__codelineno-28-6"></a><span class="n">dada2otu</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">seqtab</span><span class="o">=</span><span class="kc">NULL</span><span class="p">,</span><span class="w"> </span><span class="n">taxonomy</span><span class="o">=</span><span class="kc">NULL</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
</span><span id="__span-28-7"><a id="__codelineno-28-7" name="__codelineno-28-7" href="#__codelineno-28-7"></a><span class="n">out</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.data.frame</span><span class="p">(</span><span class="nf">cbind</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="nf">nrow</span><span class="p">(</span><span class="n">taxonomy</span><span class="p">)),</span><span class="w"> </span><span class="nf">t</span><span class="p">(</span><span class="nf">as.data.frame</span><span class="p">(</span><span class="n">seqtab</span><span class="p">)),</span><span class="w"> </span>
</span><span id="__span-28-8"><a id="__codelineno-28-8" name="__codelineno-28-8" href="#__codelineno-28-8"></a><span class="w">                            </span><span class="nf">apply</span><span class="p">(</span><span class="nf">as.data.frame</span><span class="p">(</span><span class="n">taxonomy</span><span class="p">),</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">paste</span><span class="p">,</span><span class="w"> </span><span class="n">collapse</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;; &#39;</span><span class="p">),</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">seqtab</span><span class="p">)))</span>
</span><span id="__span-28-9"><a id="__codelineno-28-9" name="__codelineno-28-9" href="#__codelineno-28-9"></a><span class="nf">row.names</span><span class="p">(</span><span class="n">out</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="nf">nrow</span><span class="p">(</span><span class="n">out</span><span class="p">))</span>
</span><span id="__span-28-10"><a id="__codelineno-28-10" name="__codelineno-28-10" href="#__codelineno-28-10"></a><span class="nf">names</span><span class="p">(</span><span class="n">out</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;#OTU ID&#39;</span><span class="p">,</span><span class="w"> </span><span class="nf">row.names</span><span class="p">(</span><span class="nf">as.data.frame</span><span class="p">(</span><span class="n">seqtab</span><span class="p">)),</span><span class="w"> </span><span class="s">&#39;taxonomy&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;sequence&#39;</span><span class="p">)</span>
</span><span id="__span-28-11"><a id="__codelineno-28-11" name="__codelineno-28-11" href="#__codelineno-28-11"></a><span class="kr">return</span><span class="p">(</span><span class="n">out</span><span class="p">)</span>
</span><span id="__span-28-12"><a id="__codelineno-28-12" name="__codelineno-28-12" href="#__codelineno-28-12"></a><span class="p">}</span>
</span><span id="__span-28-13"><a id="__codelineno-28-13" name="__codelineno-28-13" href="#__codelineno-28-13"></a>
</span><span id="__span-28-14"><a id="__codelineno-28-14" name="__codelineno-28-14" href="#__codelineno-28-14"></a><span class="c1"># Export OTU table</span>
</span><span id="__span-28-15"><a id="__codelineno-28-15" name="__codelineno-28-15" href="#__codelineno-28-15"></a><span class="nf">fwrite</span><span class="p">(</span><span class="nf">dada2otu</span><span class="p">(</span><span class="n">seqtab_nochim</span><span class="p">,</span><span class="w"> </span><span class="n">taxonomy.species</span><span class="p">),</span><span class="w"> </span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;otu_table.txt&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;\t&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">buffMB</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">100</span><span class="p">)</span>
</span><span id="__span-28-16"><a id="__codelineno-28-16" name="__codelineno-28-16" href="#__codelineno-28-16"></a>
</span><span id="__span-28-17"><a id="__codelineno-28-17" name="__codelineno-28-17" href="#__codelineno-28-17"></a><span class="c1"># Export count table</span>
</span><span id="__span-28-18"><a id="__codelineno-28-18" name="__codelineno-28-18" href="#__codelineno-28-18"></a><span class="n">total_counts</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.data.frame</span><span class="p">(</span><span class="nf">cbind</span><span class="p">(</span><span class="nf">row.names</span><span class="p">(</span><span class="nf">as.data.frame</span><span class="p">(</span><span class="n">seqtab_nochim</span><span class="p">)),</span><span class="w"> </span><span class="nf">rowSums</span><span class="p">(</span><span class="n">seqtab_nochim</span><span class="p">)))</span>
</span><span id="__span-28-19"><a id="__codelineno-28-19" name="__codelineno-28-19" href="#__codelineno-28-19"></a><span class="nf">names</span><span class="p">(</span><span class="n">total_counts</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;SampleID&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;Total count&#39;</span><span class="p">)</span>
</span><span id="__span-28-20"><a id="__codelineno-28-20" name="__codelineno-28-20" href="#__codelineno-28-20"></a><span class="nf">fwrite</span><span class="p">(</span><span class="n">total_counts</span><span class="p">,</span><span class="w"> </span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;total_counts.txt&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;\t&#39;</span><span class="p">)</span>
</span></code></pre></div>
</details>
<h4 id="create-a-phylogenetic-tree">Create a phylogenetic tree ğŸŒ³ğŸ§¬</h4>
<p>The DADA2 sequence inference method is reference-free, so we must construct the phylogenetic tree relating the inferred sequence variants de novo. We begin by performing a multiple-alignment using the <code>DECIPHER</code> R package. 
The <code>phangorn</code> R package is then used to construct a phylogenetic tree. Here we first construct a neighbor-joining tree, and then fit a GTR+G+I (Generalized time-reversible with Gamma rate variation) maximum likelihood tree using the neighbor-joining tree as a starting point. 
Adapted from Callahan <em>et al.</em> <a href="https://f1000research.com/articles/5-1492/v2">pipeline</a>. </p>
<details class="abstract">
<summary><em>De novo</em> phylogenetic tree code</summary>
<div class="language-r highlight"><span class="filename">Create phylogenetic tree using RAxML</span><pre><span></span><code><span id="__span-29-1"><a id="__codelineno-29-1" name="__codelineno-29-1" href="#__codelineno-29-1"></a><span class="c1"># Recover variables for next chunk</span>
</span><span id="__span-29-2"><a id="__codelineno-29-2" name="__codelineno-29-2" href="#__codelineno-29-2"></a><span class="n">seqtab_nochim</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;seqtab_nochim.rds&#39;</span><span class="p">))</span>
</span><span id="__span-29-3"><a id="__codelineno-29-3" name="__codelineno-29-3" href="#__codelineno-29-3"></a><span class="n">taxonomy_species</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;taxonomy_species.rds&#39;</span><span class="p">))</span>
</span><span id="__span-29-4"><a id="__codelineno-29-4" name="__codelineno-29-4" href="#__codelineno-29-4"></a>
</span><span id="__span-29-5"><a id="__codelineno-29-5" name="__codelineno-29-5" href="#__codelineno-29-5"></a><span class="c1"># Extract ASV sequences</span>
</span><span id="__span-29-6"><a id="__codelineno-29-6" name="__codelineno-29-6" href="#__codelineno-29-6"></a><span class="n">seqs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">seqtab_nochim</span><span class="p">)</span>
</span><span id="__span-29-7"><a id="__codelineno-29-7" name="__codelineno-29-7" href="#__codelineno-29-7"></a><span class="nf">names</span><span class="p">(</span><span class="n">seqs</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">seqs</span>
</span><span id="__span-29-8"><a id="__codelineno-29-8" name="__codelineno-29-8" href="#__codelineno-29-8"></a>
</span><span id="__span-29-9"><a id="__codelineno-29-9" name="__codelineno-29-9" href="#__codelineno-29-9"></a><span class="c1"># Align sequences</span>
</span><span id="__span-29-10"><a id="__codelineno-29-10" name="__codelineno-29-10" href="#__codelineno-29-10"></a><span class="n">alignment</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">AlignSeqs</span><span class="p">(</span><span class="nf">DNAStringSet</span><span class="p">(</span><span class="n">seqs</span><span class="p">),</span><span class="w"> </span><span class="n">anchor</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NA</span><span class="p">)</span>
</span><span id="__span-29-11"><a id="__codelineno-29-11" name="__codelineno-29-11" href="#__codelineno-29-11"></a><span class="n">phang_align</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">phyDat</span><span class="p">(</span><span class="nf">as</span><span class="p">(</span><span class="n">alignment</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;matrix&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">type</span><span class="w"> </span><span class="o">=</span><span class="s">&#39;DNA&#39;</span><span class="p">)</span>
</span><span id="__span-29-12"><a id="__codelineno-29-12" name="__codelineno-29-12" href="#__codelineno-29-12"></a>
</span><span id="__span-29-13"><a id="__codelineno-29-13" name="__codelineno-29-13" href="#__codelineno-29-13"></a><span class="c1"># Convert phyDat class to DNAbin format</span>
</span><span id="__span-29-14"><a id="__codelineno-29-14" name="__codelineno-29-14" href="#__codelineno-29-14"></a><span class="n">alignment_dnabin</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.DNAbin</span><span class="p">(</span><span class="n">phang_align</span><span class="p">)</span>
</span><span id="__span-29-15"><a id="__codelineno-29-15" name="__codelineno-29-15" href="#__codelineno-29-15"></a>
</span><span id="__span-29-16"><a id="__codelineno-29-16" name="__codelineno-29-16" href="#__codelineno-29-16"></a><span class="c1"># Prepare suitable names for RAxML</span>
</span><span id="__span-29-17"><a id="__codelineno-29-17" name="__codelineno-29-17" href="#__codelineno-29-17"></a><span class="n">seq_names_tidy</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span><span class="n">original</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">alignment_dnabin</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
</span><span id="__span-29-18"><a id="__codelineno-29-18" name="__codelineno-29-18" href="#__codelineno-29-18"></a><span class="nf">mutate</span><span class="p">(</span><span class="n">raxml_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;seq&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
</span><span id="__span-29-19"><a id="__codelineno-29-19" name="__codelineno-29-19" href="#__codelineno-29-19"></a><span class="nf">mutate</span><span class="p">(</span><span class="n">raxml_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">gsub</span><span class="p">(</span><span class="s">&#39;\\.&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;_&#39;</span><span class="p">,</span><span class="w"> </span><span class="nf">make.unique</span><span class="p">(</span><span class="n">raxml_names</span><span class="p">)))</span>
</span><span id="__span-29-20"><a id="__codelineno-29-20" name="__codelineno-29-20" href="#__codelineno-29-20"></a><span class="nf">rownames</span><span class="p">(</span><span class="n">alignment_dnabin</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">seq_names_tidy</span><span class="o">$</span><span class="n">raxml_names</span>
</span><span id="__span-29-21"><a id="__codelineno-29-21" name="__codelineno-29-21" href="#__codelineno-29-21"></a>
</span><span id="__span-29-22"><a id="__codelineno-29-22" name="__codelineno-29-22" href="#__codelineno-29-22"></a><span class="c1"># Run RAxML using the raxml function</span>
</span><span id="__span-29-23"><a id="__codelineno-29-23" name="__codelineno-29-23" href="#__codelineno-29-23"></a><span class="c1"># Set threads to the number of **physical** cores you have</span>
</span><span id="__span-29-24"><a id="__codelineno-29-24" name="__codelineno-29-24" href="#__codelineno-29-24"></a><span class="n">fitGTR</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">raxml</span><span class="p">(</span><span class="n">DNAbin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">alignment_dnabin</span><span class="p">,</span>
</span><span id="__span-29-25"><a id="__codelineno-29-25" name="__codelineno-29-25" href="#__codelineno-29-25"></a><span class="w">                </span><span class="n">m</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;GTRGAMMAI&#39;</span><span class="p">,</span>
</span><span id="__span-29-26"><a id="__codelineno-29-26" name="__codelineno-29-26" href="#__codelineno-29-26"></a><span class="w">                </span><span class="n">f</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;a&#39;</span><span class="p">,</span>
</span><span id="__span-29-27"><a id="__codelineno-29-27" name="__codelineno-29-27" href="#__codelineno-29-27"></a><span class="w">                </span><span class="n">N</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">100</span><span class="p">,</span>
</span><span id="__span-29-28"><a id="__codelineno-29-28" name="__codelineno-29-28" href="#__codelineno-29-28"></a><span class="w">                </span><span class="n">p</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">12345</span><span class="p">,</span>
</span><span id="__span-29-29"><a id="__codelineno-29-29" name="__codelineno-29-29" href="#__codelineno-29-29"></a><span class="w">                </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">12345</span><span class="p">,</span>
</span><span id="__span-29-30"><a id="__codelineno-29-30" name="__codelineno-29-30" href="#__codelineno-29-30"></a><span class="w">                </span><span class="n">threads</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">24</span><span class="p">,</span><span class="w"> </span>
</span><span id="__span-29-31"><a id="__codelineno-29-31" name="__codelineno-29-31" href="#__codelineno-29-31"></a><span class="w">                </span><span class="n">exec</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;/usr/bin/raxmlHPC-PTHREADS&#39;</span><span class="p">,</span>
</span><span id="__span-29-32"><a id="__codelineno-29-32" name="__codelineno-29-32" href="#__codelineno-29-32"></a><span class="w">                </span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;dada2&#39;</span><span class="p">)</span>
</span><span id="__span-29-33"><a id="__codelineno-29-33" name="__codelineno-29-33" href="#__codelineno-29-33"></a>
</span><span id="__span-29-34"><a id="__codelineno-29-34" name="__codelineno-29-34" href="#__codelineno-29-34"></a><span class="c1"># Move the best tree to your dada2 input folder from the project directory</span>
</span><span id="__span-29-35"><a id="__codelineno-29-35" name="__codelineno-29-35" href="#__codelineno-29-35"></a><span class="c1"># Remove the other unnecessary files</span>
</span><span id="__span-29-36"><a id="__codelineno-29-36" name="__codelineno-29-36" href="#__codelineno-29-36"></a><span class="c1"># Read in the resulting tree from the RAxML output file</span>
</span><span id="__span-29-37"><a id="__codelineno-29-37" name="__codelineno-29-37" href="#__codelineno-29-37"></a><span class="nf">file.copy</span><span class="p">(</span><span class="n">from</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;RAxML_bestTree.dada2&#39;</span><span class="p">),</span>
</span><span id="__span-29-38"><a id="__codelineno-29-38" name="__codelineno-29-38" href="#__codelineno-29-38"></a><span class="w">        </span><span class="n">to</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;RAxML_bestTree.dada2&#39;</span><span class="p">))</span>
</span><span id="__span-29-39"><a id="__codelineno-29-39" name="__codelineno-29-39" href="#__codelineno-29-39"></a><span class="nf">file.remove</span><span class="p">(</span><span class="nf">list.files</span><span class="p">(</span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;RAxML|dada2.phy&#39;</span><span class="p">))</span>
</span><span id="__span-29-40"><a id="__codelineno-29-40" name="__codelineno-29-40" href="#__codelineno-29-40"></a><span class="n">tree</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.tree</span><span class="p">(</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;RAxML_bestTree.dada2&#39;</span><span class="p">))</span>
</span><span id="__span-29-41"><a id="__codelineno-29-41" name="__codelineno-29-41" href="#__codelineno-29-41"></a>
</span><span id="__span-29-42"><a id="__codelineno-29-42" name="__codelineno-29-42" href="#__codelineno-29-42"></a><span class="c1"># Correct the tip labels</span>
</span><span id="__span-29-43"><a id="__codelineno-29-43" name="__codelineno-29-43" href="#__codelineno-29-43"></a><span class="n">tip_labels</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span><span class="n">raxml_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">tree</span><span class="o">$</span><span class="n">tip.label</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
</span><span id="__span-29-44"><a id="__codelineno-29-44" name="__codelineno-29-44" href="#__codelineno-29-44"></a><span class="nf">left_join</span><span class="p">(</span><span class="n">seq_names_tidy</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;raxml_names&#39;</span><span class="p">)</span>
</span><span id="__span-29-45"><a id="__codelineno-29-45" name="__codelineno-29-45" href="#__codelineno-29-45"></a><span class="n">tree</span><span class="o">$</span><span class="n">tip.label</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">tip_labels</span><span class="o">$</span><span class="n">original</span>
</span><span id="__span-29-46"><a id="__codelineno-29-46" name="__codelineno-29-46" href="#__codelineno-29-46"></a>
</span><span id="__span-29-47"><a id="__codelineno-29-47" name="__codelineno-29-47" href="#__codelineno-29-47"></a><span class="c1"># Force dichotomy (resolve any multifurcations)</span>
</span><span id="__span-29-48"><a id="__codelineno-29-48" name="__codelineno-29-48" href="#__codelineno-29-48"></a><span class="n">tree_dich</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ape</span><span class="o">::</span><span class="nf">multi2di</span><span class="p">(</span><span class="n">tree</span><span class="p">)</span>
</span><span id="__span-29-49"><a id="__codelineno-29-49" name="__codelineno-29-49" href="#__codelineno-29-49"></a>
</span><span id="__span-29-50"><a id="__codelineno-29-50" name="__codelineno-29-50" href="#__codelineno-29-50"></a><span class="c1"># Re-root the tree explicitly using an outgroup (i.e. the first tip)</span>
</span><span id="__span-29-51"><a id="__codelineno-29-51" name="__codelineno-29-51" href="#__codelineno-29-51"></a><span class="n">tree_rooted</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">root</span><span class="p">(</span><span class="n">tree_dich</span><span class="p">,</span>
</span><span id="__span-29-52"><a id="__codelineno-29-52" name="__codelineno-29-52" href="#__codelineno-29-52"></a><span class="w">                    </span><span class="n">outgroup</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">tree_dich</span><span class="o">$</span><span class="n">tip.label</span><span class="p">[</span><span class="m">1</span><span class="p">],</span>
</span><span id="__span-29-53"><a id="__codelineno-29-53" name="__codelineno-29-53" href="#__codelineno-29-53"></a><span class="w">                    </span><span class="n">resolve.root</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
</span><span id="__span-29-54"><a id="__codelineno-29-54" name="__codelineno-29-54" href="#__codelineno-29-54"></a>
</span><span id="__span-29-55"><a id="__codelineno-29-55" name="__codelineno-29-55" href="#__codelineno-29-55"></a><span class="c1"># Save to disk</span>
</span><span id="__span-29-56"><a id="__codelineno-29-56" name="__codelineno-29-56" href="#__codelineno-29-56"></a><span class="nf">saveRDS</span><span class="p">(</span><span class="n">tree_rooted</span><span class="p">,</span><span class="w"> </span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;input&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_dada2&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;tree.rds&#39;</span><span class="p">))</span>
</span></code></pre></div>
</details>
<h2 id="rights">Rights</h2>
<ul>
<li>Copyright Â©ï¸ 2025 Mucosal Immunology Lab, Monash University, Melbourne, Australia and Service de Pneumologie, Centre Hospitalier Universitaire Vaudois (CHUV), Switzerland.</li>
<li>Licence: This pipeline is provided under the MIT license.</li>
<li>Pipeline authors: A. Rapin, C. Pattaroni, A. Butler, B.J. Marsland, M. Macowan</li>
<li>This document: M. Macowan</li>
</ul>
<div class="admonition info">
<ul>
<li>Illumina Utils: <a href="https://github.com/merenlab/illumina-utils">Meren Lab</a></li>
<li>DADA2: <a href="https://github.com/benjjneb/dada2">Benjamin Callahan et al.</a></li>
<li>SILVA: <a href="https://www.arb-silva.de/">arb-silva.de</a></li>
<li>RAxML: <a href="https://cme.h-its.org/exelixis/web/software/raxml/">Exelixis Lab</a></li>
<li>phangorn: <a href="https://github.com/KlausVigo/phangorn">Klaus Schliep et al.</a></li>
<li>ips: <a href="https://github.com/heibl/ips">Christoph Heibl et al.</a></li>
</ul>
</div>









  




                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": ["content.code.copy"], "search": "../../assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.50899def.min.js"></script>
      
    
  </body>
</html>