
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../convert-raw-novaseq-outputs/">
      
      
        <link rel="next" href="../market-storage/">
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.16">
    
    
      
        <title>SRA sequencing data submission - Mucosal Immunology Bioinformatics Hub</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.7e37652d.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheet/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      
  


  
  

<script id="__analytics">function __md_analytics(){function e(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],e("js",new Date),e("config","G-7HN0DB92MT"),document.addEventListener("DOMContentLoaded",(function(){document.forms.search&&document.forms.search.query.addEventListener("blur",(function(){this.value&&e("event","search",{search_term:this.value})}));document$.subscribe((function(){var t=document.forms.feedback;if(void 0!==t)for(var a of t.querySelectorAll("[type=submit]"))a.addEventListener("click",(function(a){a.preventDefault();var n=document.location.pathname,d=this.getAttribute("data-md-value");e("event","feedback",{page:n,data:d}),t.firstElementChild.disabled=!0;var r=t.querySelector(".md-feedback__note [data-md-value='"+d+"']");r&&(r.hidden=!1)})),t.hidden=!1})),location$.subscribe((function(t){e("config","G-7HN0DB92MT",{page_path:t.pathname})}))}));var t=document.createElement("script");t.async=!0,t.src="https://www.googletagmanager.com/gtag/js?id=G-7HN0DB92MT",document.getElementById("__analytics").insertAdjacentElement("afterEnd",t)}</script>
  
    <script>"undefined"!=typeof __md_analytics&&__md_analytics()</script>
  

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#sra-sequencing-data-submission" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Mucosal Immunology Bioinformatics Hub" class="md-header__button md-logo" aria-label="Mucosal Immunology Bioinformatics Hub" data-md-component="logo">
      
  <img src="../../assets/mil_logo.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Mucosal Immunology Bioinformatics Hub
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              SRA sequencing data submission
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://mucosal-immunology-lab.github.io/bioinformatics-hub/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    Mucosal-Immunology-Lab/bioinformatic-hub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Mucosal Immunology Bioinformatics Hub" class="md-nav__button md-logo" aria-label="Mucosal Immunology Bioinformatics Hub" data-md-component="logo">
      
  <img src="../../assets/mil_logo.png" alt="logo">

    </a>
    Mucosal Immunology Bioinformatics Hub
  </label>
  
    <div class="md-nav__source">
      <a href="https://mucosal-immunology-lab.github.io/bioinformatics-hub/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    Mucosal-Immunology-Lab/bioinformatic-hub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../RNAseq/rnaseq-nfcore/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Bulk Transcriptomics
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Microbiome
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Microbiome
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_1" >
        
          
          <label class="md-nav__link" for="__nav_3_1" id="__nav_3_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    16S rRNA amplicon
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_1">
            <span class="md-nav__icon md-icon"></span>
            16S rRNA amplicon
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Microbiome/dada2_pipeline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    DADA2 pipeline
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    LCMS Analysis
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            LCMS Analysis
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../LCMS/lcms-analysis/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../LCMS/msdial-processing/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    1. Raw Data Processing with MS-DIAL
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../LCMS/pmp-quality-control/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    2. Peak Matrix Processing
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../LCMS/secondary-feature-annotation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    3. Secondary Feature Annotation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../LCMS/manual-peak-curation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    4. Manual Peak Curation
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Nextflow Pipelines
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Nextflow Pipelines
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../NextFlow/scRNAseq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Single-cell RNAseq
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../NextFlow/metagenomics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Metagenomics
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../NextFlow/dada2_16S/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    DADA2 16S rRNA
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Cross-omic Tools
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Cross-omic Tools
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6_1" >
        
          
          <label class="md-nav__link" for="__nav_6_1" id="__nav_6_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Differential testing
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_6_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6_1">
            <span class="md-nav__icon md-icon"></span>
            Differential testing
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../CrossOmicTools/DifferentialTesting/bio_limma/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Linear modelling with bio_limma
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" checked>
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Utilities
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            Utilities
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7_1" checked>
        
          
          <label class="md-nav__link" for="__nav_7_1" id="__nav_7_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Handling sequencing data
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_7_1_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_7_1">
            <span class="md-nav__icon md-icon"></span>
            Handling sequencing data
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../data-best-practices/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Data best practices
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../convert-raw-novaseq-outputs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Converting NovaSeq outputs
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    SRA sequencing data submission
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    SRA sequencing data submission
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#process-overview" class="md-nav__link">
    <span class="md-ellipsis">
      Process overview
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#register-a-bioproject" class="md-nav__link">
    <span class="md-ellipsis">
      Register a BioProject üìî
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#register-biosamples" class="md-nav__link">
    <span class="md-ellipsis">
      Register BioSamples üß™
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Register BioSamples üß™">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#microbiome-data" class="md-nav__link">
    <span class="md-ellipsis">
      Microbiome data ü¶†
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#transcriptomics-data" class="md-nav__link">
    <span class="md-ellipsis">
      Transcriptomics data üë®üê≠
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#submit-data-to-sra" class="md-nav__link">
    <span class="md-ellipsis">
      Submit data to SRA üì§
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7_2" >
        
          
          <label class="md-nav__link" for="__nav_7_2" id="__nav_7_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Monash data storage
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_7_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7_2">
            <span class="md-nav__icon md-icon"></span>
            Monash data storage
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../market-storage/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Market storage
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../vault-storage/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Vault storage
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../PublicDatasets/public-datasets/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Public Datasets
    
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#process-overview" class="md-nav__link">
    <span class="md-ellipsis">
      Process overview
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#register-a-bioproject" class="md-nav__link">
    <span class="md-ellipsis">
      Register a BioProject üìî
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#register-biosamples" class="md-nav__link">
    <span class="md-ellipsis">
      Register BioSamples üß™
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Register BioSamples üß™">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#microbiome-data" class="md-nav__link">
    <span class="md-ellipsis">
      Microbiome data ü¶†
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#transcriptomics-data" class="md-nav__link">
    <span class="md-ellipsis">
      Transcriptomics data üë®üê≠
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#submit-data-to-sra" class="md-nav__link">
    <span class="md-ellipsis">
      Submit data to SRA üì§
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  



<h1 id="sra-sequencing-data-submission">SRA sequencing data submission</h1>
<p>A guide to submitting sequencing data to the National Center for Biotechnology Information (NCBI) sequencing read archive (SRA) database. Includes information on uploading data to the SRA using the high-speed Aspera Connect tool.</p>
<div class="admonition warning">
<p class="admonition-title">Patient-derived sequencing files</p>
<p>If your samples are derived from humans, ensure that your <strong>file names include no reference to patient identifiers</strong>. Once uploaded to the SRA database, it is very difficult to change the names of files, and requires directly contacting the database to arrange for removal of files and for you to reupload the data. It also involves a difficult process of them re-mapping the new uploads to your existing SRA metadata files.</p>
<p>Also ensure that you only include the absolute minimum amount of metadata, in a manner that protects patient confidentiality. Absolutely no information should be unique to one single patient in your cohort, even an age (if you have a patient with a unique age, this should be replaced with <code>NA</code> for the purposes of SRA submission). For manuscripts, you can include a phrase indicating that further metadata is available upon reasonable request. <strong>The important thing here is to not infringe on patient privacy and confidentiality.</strong></p>
<p>Things you could potentially include:</p>
<ul>
<li>Modified and anonymised patient ID</li>
<li>Sampling group</li>
<li>Timepoint (not exact days or months)</li>
<li>Sex</li>
<li>Collection year (no exact dates)</li>
<li>Tissue</li>
</ul>
</div>
<h2 id="process-overview">Process overview</h2>
<ol>
<li>Register a BioProject</li>
<li>Register BioSamples for the related BioProject</li>
<li>Submit data to SRA</li>
</ol>
<h2 id="register-a-bioproject">Register a BioProject üìî</h2>
<p>The BioProject is an important element that can link together different types of sequencing data, and represents all the sequencing data for a given experiment.</p>
<p>Go to the <a href="https://submit.ncbi.nlm.nih.gov/">SRA submission</a> website to register a new BioProject.</p>
<ul>
<li>Sample scope: Multispecies (if you have microbiome data)</li>
<li>Target description: Bacterial 16S metagenomics (change if you have shotgun metagenomics and/or host transcriptomics)</li>
<li>Organism name: Human (change if using mouse or rat data)</li>
<li>Project type: Metagenome (add transcriptome if you also have host transcriptomics)</li>
</ul>
<h2 id="register-biosamples">Register BioSamples üß™</h2>
<h3 id="microbiome-data">Microbiome data ü¶†</h3>
<p>Microbiome samples will be registered as <em>MIMARKS Specimen</em> samples. On the <strong>BioSample Attributes</strong> tab, download the BioSample metadata Excel template, and complete it accordingly before uploading. Be very careful with the required field formats. You can double check ontology using the <a href="https://www.ebi.ac.uk/ols4/">EMBL-EBI Ontology Lookup Service</a>.</p>
<ul>
<li>Use the BioProject accession number previously generated</li>
<li><strong>Organism</strong>: <code>human metagenome</code> (or as appropriate)</li>
<li><strong>Env broad scale</strong>: <code>host-associated</code></li>
<li><strong>Env local scale</strong>: <code>mammalia-associated habitat</code></li>
<li><strong>Env medium</strong>: (as appropriate &ndash; e.g. <code>bronchial brushing</code>, <code>stool</code>, etc.)</li>
<li><strong>Strain, isolate, cultivar, ecotype</strong>: <code>NA</code></li>
<li><strong>Host</strong>: (as appropriate &ndash; e.g. <code>Homo sapiens</code>, <code>Mus musculus</code>, etc.)</li>
<li><strong>Latitude/Longitude</strong>: (as appropriate &ndash; e.g. <code>37.8457 S 144.9819 E</code> for the Alfred Hospital, Melbourne)</li>
<li>Add any other relevant host information in the table, as well as the host tissue samples</li>
<li>Any other column which is not relevant can be set to <code>NA</code></li>
</ul>
<p>The <strong>SRA Metadata</strong> tab is what will join everything together. Once again, download the provided Excel template, and fill everything in carefully.</p>
<ul>
<li><strong>Sample name</strong>: the base name of your samples</li>
<li><strong>Library ID</strong>: you may have named your files differently than your sample names &ndash; provide this if so, otherwise you can repeat the sample name</li>
<li><strong>Title</strong>: a short description of the sample in the form "<code>{methodology}</code> of <code>{organism}</code>: <code>{sample_info}</code>" &ndash; e.g. "Shotgun metagenomics of Homo sapiens: childhood bronchial brushing".</li>
<li><strong>Library strategy</strong>: <code>WGS</code></li>
<li><strong>Library source</strong>: <code>METAGENOMIC</code></li>
<li><strong>Library selection</strong>: <code>RANDOM</code></li>
<li><strong>Library layout</strong>: <code>paired</code></li>
<li><strong>Platform</strong>: <code>ILLUMINA</code></li>
<li><strong>Instrument model</strong>: <code>Illumina NovaSeq 6000</code></li>
<li><strong>Design description</strong>: <code>NA</code></li>
<li><strong>Filetype</strong>: <code>fastq</code></li>
<li><strong>Filename</strong>: the file name of the forward reads</li>
<li><strong>Filename2</strong>: the file name of the reverse reads</li>
</ul>
<h3 id="transcriptomics-data">Transcriptomics data üë®üê≠</h3>
<p>Host transcriptomics samples will be registered as either <em>HUMAN</em> or <em>Model organism or animal</em> samples. On the <strong>BioSample Attributes</strong> tab, download the BioSample metadata Excel template, and complete it accordingly before uploading. Be very careful with the required field formats. You can double check ontology using the <a href="https://www.ebi.ac.uk/ols4/">EMBL-EBI Ontology Lookup Service</a>.</p>
<ul>
<li>Use the BioProject accession number previously generated</li>
<li><strong>Organism</strong>: <code>Homo sapiens</code> (or <code>Mus musculus</code>/<code>Rattus norvegicus</code> as appropriate)</li>
<li><strong>Isolate</strong>: NA</li>
<li><strong>Age</strong>: fill this in, but leave <code>NA</code> for human samples if it would result in a unique combination of metadata variables with potential to allow identification of any individual.</li>
<li><strong>Biomaterial provider</strong>: enter the lab, organisation etc. that provided the samples</li>
<li><strong>Collection date</strong>: do not enter any exact dates for human samples</li>
<li><strong>Geo loc name</strong>: country in which samples were collected</li>
<li><strong>Sex</strong>: provide sex of host</li>
<li><strong>Tissue</strong>: specify tissue origin of samples</li>
<li>Add any other relevant data, such as sampling group</li>
</ul>
<p>As above, the <strong>SRA Metadata</strong> tab is where the magic will happen :magic_wand:<img alt="‚ú®" class="emojione" src="https://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/png/2728.png" title=":sparkles:" />. Once again, download the provided Excel template, and fill everything in carefully.</p>
<ul>
<li><strong>Sample name</strong>: the base name of your samples</li>
<li><strong>Library ID</strong>: you may have named your files differently than your sample names &ndash; provide this if so, otherwise you can repeat the sample name</li>
<li><strong>Title</strong>: a short description of the sample in the form "<code>{methodology}</code> of <code>{organism}</code>: <code>{sample_info}</code>" &ndash; e.g. "RNA-Seq of Homo sapiens: childhood bronchial brushing".</li>
<li><strong>Library strategy</strong>: <code>RNA-Seq</code></li>
<li><strong>Library source</strong>: <code>TRANSCRIPTOMIC</code></li>
<li><strong>Library selection</strong>: <code>RANDOM</code></li>
<li><strong>Library layout</strong>: <code>paired</code></li>
<li><strong>Platform</strong>: <code>ILLUMINA</code></li>
<li><strong>Instrument model</strong>: <code>Illumina NovaSeq 6000</code></li>
<li><strong>Design description</strong>: <code>NA</code></li>
<li><strong>Filetype</strong>: <code>fastq</code></li>
<li><strong>Filename</strong>: the file name of the forward reads</li>
<li><strong>Filename2</strong>: the file name of the reverse reads</li>
</ul>
<h2 id="submit-data-to-sra">Submit data to SRA üì§</h2>
<div class="admonition question">
<p class="admonition-title">Which upload option should I choose?</p>
<p>You can choose either of the following upload options, and each has pros and cons.</p>
<ul>
<li><strong>Aspera Connect</strong> (at least with NCBI) only allows sequential uploads, but the upload speed is significantly faster.</li>
<li><strong>Filezilla</strong> allows parallel uploads according to your settings, but upload speed is typically slower.</li>
</ul>
</div>
<div class="tabbed-set tabbed-alternate" data-tabs="1:2"><input checked="checked" id="__tabbed_1_1" name="__tabbed_1" type="radio" /><input id="__tabbed_1_2" name="__tabbed_1" type="radio" /><div class="tabbed-labels"><label for="__tabbed_1_1">Aspera Connect</label><label for="__tabbed_1_2">FileZilla ü¶ñ</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<p>The IBM Aspera Connect tool allows for much faster uploads than FileZilla, and is a good alternative for large files.</p>
<p><strong>Linux process</strong> üêß</p>
<p>The process described here is for Linux, but is similar for Windows and MacOS operating systems. More information is provided on the <a href="https://www.ibm.com/docs/en/aspera-connect/4.2?topic=suc-installation">IBM website</a>.</p>
<ul>
<li>Download the <a href="https://www.ibm.com/aspera/connect/">Aspera Connect software</a>.</li>
<li>Open a new terminal window (<code>Ctrl+Alt+T</code>)</li>
<li>Navigate to downloads, extract the <code>tar.gz</code> file.</li>
<li>Run the install script.</li>
</ul>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="c1"># Extract the file</span>
</span><span id="__span-0-2"><a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a>tar<span class="w"> </span>-zxvf<span class="w"> </span>ibm-aspera-connect-version+platform.tar.gz
</span><span id="__span-0-3"><a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a><span class="c1"># Run the install script</span>
</span><span id="__span-0-4"><a id="__codelineno-0-4" name="__codelineno-0-4" href="#__codelineno-0-4"></a>./ibm-aspera-connect-version+platform.sh
</span></code></pre></div>
<ul>
<li>Add the Aspera Connect bin folder to your PATH variable (reopen terminal to apply changes).</li>
</ul>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a><span class="c1"># Add folder to PATH</span>
</span><span id="__span-1-2"><a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a><span class="nb">echo</span><span class="w"> </span><span class="s1">&#39;export PATH=$PATH:/home/{user}/.aspera/connect/bin/ &gt;&gt; ~/.bashrc&#39;</span>
</span></code></pre></div>
<ul>
<li>Download the NCBI Aspera Connect <a href="https://submit.ncbi.nlm.nih.gov/preload/aspera_key/">key file</a>.</li>
<li>Navigate to the parent folder of the folder containing the files you want to upload to the SRA database, and create a new bash script.</li>
</ul>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a><span class="c1"># Create a new bash script file</span>
</span><span id="__span-2-2"><a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a>touch<span class="w"> </span>upload_seq_data.sh
</span></code></pre></div>
<ul>
<li>Add the following code to the bash script file. <ul>
<li>The <code>-i</code> argument is the path to the key file, and must be given as a full path (not a relative one).</li>
<li>The <code>-d</code> argument specifies that the directory will be created if it doesn't exist.</li>
<li>You can adjust the maximum upload speed using the <code>-l500m</code> argument, where <code>500</code> is the speed in Mbps. You could increase or decrease as desired.</li>
<li>Add the folder containing the data to upload, which can be relative to the folder containing the bash script.</li>
<li>Next provide the upload folder provided by NCBI, which will be user-specific, and <strong>ensure you provide a project folder</strong> at the end of this. Data will not be available if it is uploaded into the main uploads folder.</li>
</ul>
</li>
</ul>
<div class="admonition warning">
<p class="admonition-title">Seriously... remember to specify a new folder üëÄ</p>
<p>If you don't specify a folder (<code>{name-of-project}</code>) after the main directory path, your files will be uploaded to the main user directory.
Your files will then not be available for any SRA submission, and you will have to re-upload everything.</p>
<p><code>subasp@upload.ncbi.nlm.nih.gov:uploads/{user-specific-ID}/{name-of-project}</code></p>
</div>
<div class="language-bash highlight"><span class="filename">upload_seq_data.sh</span><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a><span class="ch">#!/bin/bash</span>
</span><span id="__span-3-2"><a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a>ascp<span class="w"> </span>-i<span class="w"> </span><span class="o">{</span>/full/path/to/key-file/aspera.openssh<span class="o">}</span><span class="w"> </span>-QT<span class="w"> </span>-l500m<span class="w"> </span>-k1<span class="w"> </span>-d<span class="w"> </span><span class="o">{</span>./name-of-seq-data-folder<span class="o">}</span><span class="w"> </span>subasp@upload.ncbi.nlm.nih.gov:uploads/<span class="o">{</span>user-specific-ID<span class="o">}</span>/<span class="o">{</span>name-of-project<span class="o">}</span>
</span></code></pre></div>
<ul>
<li>Run the bash script, and upload all files. The default settings will allow you to resume uploads if they are interrupted, and it will not overwrite files that are identical in the destination folder.</li>
</ul>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a><span class="c1"># Run script</span>
</span><span id="__span-4-2"><a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a>bash<span class="w"> </span>upload_seq_data.sh
</span></code></pre></div>
</div>
<div class="tabbed-block">
<p>Using <a href="https://filezilla-project.org/">FileZilla</a> is more effective when you have large files and/or a large number of files.</p>
<p>In FileZilla, open the sites manager and connect to NCBI as follows:
- Protocol: <code>FTP</code>
- Host: <code>ftp-private.ncbi.nlm.nih.gov</code>
- Username: <code>subftp</code>
- Password: this is your user-specific NCBI password given when you submit your data</p>
<p>In the <code>Advanced</code> tab next to the <code>General</code> tab, set the <code>Default remote directory</code> field to the directory specified by NCBI. This will looks something like: <code>/uploads/{username}_{uniqueID}</code>.</p>
<p>Select connect, and gain access to your account folder on the NCBI FTP server.</p>
<p><strong>Create a new project folder</strong> within the main upload folder, and enter the folder. Add your files to the upload queue, and begin the upload process.</p>
</div>
</div>
</div>









  




                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": ["content.code.copy"], "search": "../../assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.50899def.min.js"></script>
      
    
  </body>
</html>