
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../pmp-quality-control/">
      
      
        <link rel="next" href="../manual-peak-curation/">
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.16">
    
    
      
        <title>3. Secondary Feature Annotation - Mucosal Immunology Bioinformatics Hub</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.7e37652d.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheet/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      
  


  
  

<script id="__analytics">function __md_analytics(){function e(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],e("js",new Date),e("config","G-7HN0DB92MT"),document.addEventListener("DOMContentLoaded",(function(){document.forms.search&&document.forms.search.query.addEventListener("blur",(function(){this.value&&e("event","search",{search_term:this.value})}));document$.subscribe((function(){var t=document.forms.feedback;if(void 0!==t)for(var a of t.querySelectorAll("[type=submit]"))a.addEventListener("click",(function(a){a.preventDefault();var n=document.location.pathname,d=this.getAttribute("data-md-value");e("event","feedback",{page:n,data:d}),t.firstElementChild.disabled=!0;var r=t.querySelector(".md-feedback__note [data-md-value='"+d+"']");r&&(r.hidden=!1)})),t.hidden=!1})),location$.subscribe((function(t){e("config","G-7HN0DB92MT",{page_path:t.pathname})}))}));var t=document.createElement("script");t.async=!0,t.src="https://www.googletagmanager.com/gtag/js?id=G-7HN0DB92MT",document.getElementById("__analytics").insertAdjacentElement("afterEnd",t)}</script>
  
    <script>"undefined"!=typeof __md_analytics&&__md_analytics()</script>
  

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#secondary-ms1-feature-annotation" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Mucosal Immunology Bioinformatics Hub" class="md-header__button md-logo" aria-label="Mucosal Immunology Bioinformatics Hub" data-md-component="logo">
      
  <img src="../../assets/mil_logo.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Mucosal Immunology Bioinformatics Hub
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              3. Secondary Feature Annotation
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://mucosal-immunology-lab.github.io/bioinformatics-hub/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    Mucosal-Immunology-Lab/bioinformatic-hub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Mucosal Immunology Bioinformatics Hub" class="md-nav__button md-logo" aria-label="Mucosal Immunology Bioinformatics Hub" data-md-component="logo">
      
  <img src="../../assets/mil_logo.png" alt="logo">

    </a>
    Mucosal Immunology Bioinformatics Hub
  </label>
  
    <div class="md-nav__source">
      <a href="https://mucosal-immunology-lab.github.io/bioinformatics-hub/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    Mucosal-Immunology-Lab/bioinformatic-hub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../RNAseq/rnaseq-nfcore/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Bulk Transcriptomics
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Microbiome
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Microbiome
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_1" >
        
          
          <label class="md-nav__link" for="__nav_3_1" id="__nav_3_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    16S rRNA amplicon
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_1">
            <span class="md-nav__icon md-icon"></span>
            16S rRNA amplicon
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Microbiome/dada2_pipeline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    DADA2 pipeline
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" checked>
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    LCMS Analysis
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            LCMS Analysis
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../lcms-analysis/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../msdial-processing/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    1. Raw Data Processing with MS-DIAL
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pmp-quality-control/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    2. Peak Matrix Processing
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    3. Secondary Feature Annotation
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    3. Secondary Feature Annotation
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#secondary-annotation-with-hmdb" class="md-nav__link">
    <span class="md-ellipsis">
      Secondary annotation with HMDB üìò
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Secondary annotation with HMDB üìò">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#appending-hmdb-annotations-to-summarizedexperiment-objects" class="md-nav__link">
    <span class="md-ellipsis">
      Appending HMDB annotations to SummarizedExperiment objects
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#secondary-annotation-with-lipid-maps" class="md-nav__link">
    <span class="md-ellipsis">
      Secondary annotation with LIPID MAPS üßà
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Secondary annotation with LIPID MAPS üßà">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#appending-lipid-maps-annotations-to-summarizedexperiment-objects" class="md-nav__link">
    <span class="md-ellipsis">
      Appending LIPID MAPS annotations to SummarizedExperiment objects
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#comparing-annotations-from-different-databases" class="md-nav__link">
    <span class="md-ellipsis">
      Comparing annotations from different databases ‚öñÔ∏èüìö
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#keeping-only-annotated-features" class="md-nav__link">
    <span class="md-ellipsis">
      Keeping only annotated features üè∑Ô∏è‚úÖ
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#next-steps" class="md-nav__link">
    <span class="md-ellipsis">
      Next steps ‚û°Ô∏è
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#rights" class="md-nav__link">
    <span class="md-ellipsis">
      Rights
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../manual-peak-curation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    4. Manual Peak Curation
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Nextflow Pipelines
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Nextflow Pipelines
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../NextFlow/scRNAseq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Single-cell RNAseq
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../NextFlow/metagenomics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Metagenomics
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../NextFlow/dada2_16S/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    DADA2 16S rRNA
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Cross-omic Tools
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Cross-omic Tools
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6_1" >
        
          
          <label class="md-nav__link" for="__nav_6_1" id="__nav_6_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Differential testing
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_6_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6_1">
            <span class="md-nav__icon md-icon"></span>
            Differential testing
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../CrossOmicTools/DifferentialTesting/bio_limma/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Linear modelling with bio_limma
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" >
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Utilities
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            Utilities
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7_1" >
        
          
          <label class="md-nav__link" for="__nav_7_1" id="__nav_7_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Handling sequencing data
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_7_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7_1">
            <span class="md-nav__icon md-icon"></span>
            Handling sequencing data
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Utilities/data-best-practices/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Data best practices
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Utilities/convert-raw-novaseq-outputs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Converting NovaSeq outputs
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Utilities/sra-data-submission/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SRA sequencing data submission
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7_2" >
        
          
          <label class="md-nav__link" for="__nav_7_2" id="__nav_7_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Monash data storage
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_7_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7_2">
            <span class="md-nav__icon md-icon"></span>
            Monash data storage
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Utilities/market-storage/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Market storage
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Utilities/vault-storage/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Vault storage
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../PublicDatasets/public-datasets/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Public Datasets
    
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#secondary-annotation-with-hmdb" class="md-nav__link">
    <span class="md-ellipsis">
      Secondary annotation with HMDB üìò
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Secondary annotation with HMDB üìò">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#appending-hmdb-annotations-to-summarizedexperiment-objects" class="md-nav__link">
    <span class="md-ellipsis">
      Appending HMDB annotations to SummarizedExperiment objects
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#secondary-annotation-with-lipid-maps" class="md-nav__link">
    <span class="md-ellipsis">
      Secondary annotation with LIPID MAPS üßà
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Secondary annotation with LIPID MAPS üßà">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#appending-lipid-maps-annotations-to-summarizedexperiment-objects" class="md-nav__link">
    <span class="md-ellipsis">
      Appending LIPID MAPS annotations to SummarizedExperiment objects
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#comparing-annotations-from-different-databases" class="md-nav__link">
    <span class="md-ellipsis">
      Comparing annotations from different databases ‚öñÔ∏èüìö
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#keeping-only-annotated-features" class="md-nav__link">
    <span class="md-ellipsis">
      Keeping only annotated features üè∑Ô∏è‚úÖ
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#next-steps" class="md-nav__link">
    <span class="md-ellipsis">
      Next steps ‚û°Ô∏è
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#rights" class="md-nav__link">
    <span class="md-ellipsis">
      Rights
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  



<h1 id="secondary-ms1-feature-annotation">Secondary MS1 feature annotation</h1>
<p>After processing LCMS data through the MS-DIAL pipeline, secondary annotations for MS1 features can be obtained by matching mass data to external databases. </p>
<details class="question">
<summary>What do you mean by MS1 data?</summary>
<ul>
<li><strong>What is MS1 data?</strong><ul>
<li>MS1 data represents the initial scan in mass spectrometry, where ions are detected based on their mass-to-charge ratio (m/z). It provides an overview of all molecular features in a sample, including their m/z values and intensities, enabling detection and quantification of compounds.</li>
</ul>
</li>
<li><strong>How does that differ from MS/MS (MS2) data?</strong><ul>
<li>While MS1 provides a broad profile of ions without structural information, MS/MS involves selecting specific ions from the MS1 scan (precursor ions), fragmenting them, and analysing the resulting product ions. This fragmentation reveals structural details, enabling precise compound identification and differentiation of isomers.</li>
</ul>
</li>
</ul>
</details>
<div class="admonition info">
<p class="admonition-title">Version 2 of the secondary annotation functions available</p>
<p>Please update your script files for the <a href="../../assets/LCMS/LCMS_preprocessing/add_hmdb.R"><code>add_hmdb()</code></a> and <a href="../../assets/LCMS/LCMS_preprocessing/add_lmsd.R"><code>add_lmsd()</code></a> functions for new runs.
Updated with additional adduct values for automated adduct-specific correction to monoisotopic masses.
This is important for accurate secondary assignments.</p>
<p><em>17 May, 2025</em></p>
</div>
<h2 id="secondary-annotation-with-hmdb">Secondary annotation with HMDB üìò</h2>
<p>You should currently have a <code>SummarizedExperiment</code> object that has been pre-processed with <code>pmp</code>. The next stage is to match the MS1 mass data for each feature to the Human Metabolome Database (HMDB) database file.</p>
<div class="admonition abstract">
<p class="admonition-title">Prepared HMDB database file</p>
<p>Download the prepared HMDB database file in RDS format <a href="../../assets/LCMS/LCMS_preprocessing/hmdb_metabolites_detect_quant_v5_20231102.rds">here</a>.</p>
<ul>
<li>Version 5.0 (November 2023) &ndash; pre-formatted and filtered to include only annotated or documented features.</li>
</ul>
</div>
<h3 id="appending-hmdb-annotations-to-summarizedexperiment-objects">Appending HMDB annotations to <code>SummarizedExperiment</code> objects</h3>
<p>The first step is to load the formatted HMDB <code>data.frame</code> into your R session.</p>
<div class="language-r highlight"><span class="filename">Load HMDB database into your R session</span><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="c1"># Load HMDB dataset</span>
</span><span id="__span-0-2"><a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a><span class="n">hmdb_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;hmdb&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;hmdb_metabolites_detect_quant_v5_20231102.rds&#39;</span><span class="p">))</span>
</span></code></pre></div>
<p>Then, using the <a href="../../assets/LCMS/LCMS_preprocessing/add_hmdb.R"><code>add_hmdb()</code></a> function, we can search the HMDB annotations in the <code>data.frame</code> and add them to our <code>SummarizedExperiment</code> objects, as shown below with an example stool metabolomics object.</p>
<details class="info">
<summary>Parameters for <code>add_hmdb()</code></summary>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>metab_SE</code></td>
<td>The <code>SummarizedExperiment</code> object you want to annotate with HMDB.</td>
</tr>
<tr>
<td><code>hmdb</code></td>
<td>The HMDB database object.</td>
</tr>
<tr>
<td><code>mass_tol</code></td>
<td>Default: <code>0.002</code> &ndash; the mass tolerance allowed for annotation.</td>
</tr>
<tr>
<td><code>cores</code></td>
<td>The number of parallel processes to use if desired.</td>
</tr>
</tbody>
</table>
<div class="admonition warning">
<p class="admonition-title">Run time</p>
<p>Running <code>add_hmdb()</code>, especially without parallel processing, can take a very long time.</p>
</div>
</details>
<div class="admonition question">
<p class="admonition-title">Which <code>SummarizedExperiment</code> should I use?</p>
<p>If you are using the output of the <code>pmp_preprocess()</code> function, you should extract and annotate the <code>glog_results</code>. The standard practice is to use the glog-transformed data from here on.</p>
</div>
<div class="language-r highlight"><span class="filename">Add HMDB secondary annotations</span><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a><span class="c1"># Extract the glog data from the pmp_preprocess output</span>
</span><span id="__span-1-2"><a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a><span class="n">metab_stool_glog</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">metab_stool_pmp</span><span class="o">$</span><span class="n">glog_results</span>
</span><span id="__span-1-3"><a id="__codelineno-1-3" name="__codelineno-1-3" href="#__codelineno-1-3"></a>
</span><span id="__span-1-4"><a id="__codelineno-1-4" name="__codelineno-1-4" href="#__codelineno-1-4"></a><span class="c1"># Search annotations in HMDB and add to the SE objects</span>
</span><span id="__span-1-5"><a id="__codelineno-1-5" name="__codelineno-1-5" href="#__codelineno-1-5"></a><span class="n">metab_stool_glog</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">add_hmdb</span><span class="p">(</span><span class="n">metab_SE</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">metab_stool_glog</span><span class="p">,</span>
</span><span id="__span-1-6"><a id="__codelineno-1-6" name="__codelineno-1-6" href="#__codelineno-1-6"></a><span class="w">                             </span><span class="n">hmdb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">hmdb_df</span><span class="p">,</span><span class="w"> </span>
</span><span id="__span-1-7"><a id="__codelineno-1-7" name="__codelineno-1-7" href="#__codelineno-1-7"></a><span class="w">                             </span><span class="n">mass_tol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.002</span><span class="p">,</span>
</span><span id="__span-1-8"><a id="__codelineno-1-8" name="__codelineno-1-8" href="#__codelineno-1-8"></a><span class="w">                             </span><span class="n">cores</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">6</span><span class="p">)</span>
</span></code></pre></div>
<h2 id="secondary-annotation-with-lipid-maps">Secondary annotation with LIPID MAPS üßà</h2>
<p>After processing lipidomics data through MS-DIAL, you can enhance the annotations of MS1 features by leveraging the LIPID MAPS Structure Database (LMSD). At this point, you should have a <code>SummarizedExperiment</code> object containing preliminary annotations and those from the HMDB database. The next step involves matching the MS1 mass data of each feature to the entries in the LMSD database.</p>
<div class="admonition abstract">
<p class="admonition-title">Prepared LMSD database file</p>
<p>Download the prepared LMSD database file in RDS format <a href="../../assets/LCMS/LCMS_preprocessing/LMSD_231107.rds">here</a>.</p>
<ul>
<li>Version 2022-02-16</li>
</ul>
</div>
<div class="admonition question">
<p class="admonition-title">Should I run this section?</p>
<p>You should definitely run this step if you have <strong>lipidomics</strong> data to process.
If you are processing metabolomics data, you can skip this section.</p>
</div>
<h3 id="appending-lipid-maps-annotations-to-summarizedexperiment-objects">Appending LIPID MAPS annotations to <code>SummarizedExperiment</code> objects</h3>
<p>The first step is to load the formatted LMSD <code>data.frame</code> into your R session.</p>
<div class="language-r highlight"><span class="filename">Load LMSD database into your R session</span><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a><span class="c1"># Load LMSD dataset</span>
</span><span id="__span-2-2"><a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a><span class="n">lmsd_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="n">here</span><span class="o">::</span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;lmsd&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;LMSD_231107.rds&#39;</span><span class="p">))</span>
</span></code></pre></div>
<p>Then, using the <a href="../../assets/LCMS/LCMS_preprocessing/add_lmsd.R"><code>add_lmsd()</code></a> function, we can search the LIPID MAPS annotations in the <code>data.frame</code> and add them to our <code>SummarizedExperiment</code> objects, as shown below with an example stool metabolomics object.</p>
<details class="info">
<summary>Parameters for <code>add_lmsd()</code></summary>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>metab_SE</code></td>
<td>The <code>SummarizedExperiment</code> object you want to annotate with HMDB.</td>
</tr>
<tr>
<td><code>lmsd</code></td>
<td>The LMSD database object.</td>
</tr>
<tr>
<td><code>mass_tol</code></td>
<td>Default: <code>0.002</code> &ndash; the mass tolerance allowed for annotation.</td>
</tr>
<tr>
<td><code>cores</code></td>
<td>The number of parallel processes to use if desired.</td>
</tr>
</tbody>
</table>
<div class="admonition warning">
<p class="admonition-title">Run time</p>
<p>Running <code>add_lmsd()</code>, especially without parallel processing, can take a very long time.</p>
</div>
</details>
<div class="language-r highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a><span class="c1"># Search annotations in LMSD and add to the SE objects</span>
</span><span id="__span-3-2"><a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a><span class="c1"># Create list for all distinct Lipid Maps matching mz in tolerance range 0.002, an aggregated df of distinct lipids and a df to replace SummarizedExperiment metadata [rowData(metab_glog)]</span>
</span><span id="__span-3-3"><a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a><span class="n">lmsd_ann_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">add_lmsd</span><span class="p">(</span><span class="n">metab_SE</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">metab_stool_glog</span><span class="p">,</span><span class="w"> </span>
</span><span id="__span-3-4"><a id="__codelineno-3-4" name="__codelineno-3-4" href="#__codelineno-3-4"></a><span class="w">                          </span><span class="n">lmsd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">lmsd_df</span><span class="p">,</span><span class="w"> </span>
</span><span id="__span-3-5"><a id="__codelineno-3-5" name="__codelineno-3-5" href="#__codelineno-3-5"></a><span class="w">                          </span><span class="n">mass_tol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.002</span><span class="p">,</span>
</span><span id="__span-3-6"><a id="__codelineno-3-6" name="__codelineno-3-6" href="#__codelineno-3-6"></a><span class="w">                          </span><span class="n">cores</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">6</span><span class="p">)</span><span class="w"> </span>
</span><span id="__span-3-7"><a id="__codelineno-3-7" name="__codelineno-3-7" href="#__codelineno-3-7"></a>
</span><span id="__span-3-8"><a id="__codelineno-3-8" name="__codelineno-3-8" href="#__codelineno-3-8"></a><span class="c1"># Use metadata_lmsd_table to replace the existing SE object metadata</span>
</span><span id="__span-3-9"><a id="__codelineno-3-9" name="__codelineno-3-9" href="#__codelineno-3-9"></a><span class="nf">rowData</span><span class="p">(</span><span class="n">metab_stool_glog</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">lmsd_ann_list</span><span class="o">$</span><span class="n">metadata_lmsd_table</span>
</span></code></pre></div>
<h2 id="comparing-annotations-from-different-databases">Comparing annotations from different databases ‚öñÔ∏èüìö</h2>
<p>To compare the assigned annotations from each of the methods the <a href="../../assets/LCMS/LCMS_preprocessing/compare_annotations_SE.R">compare_annotations_SE()</a> function. It will produce a <code>data.frame</code> containing only features with at least one annotation, and allow us see whether the annotations typically agree with each other.</p>
<details class="info">
<summary>Parameters for <code>compare_annotations_SE()</code></summary>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>metab_SE</code></td>
<td>The <code>SummarizedExperiment</code> object with secondary annotations. These should include HMDB for metabolomics data, and both HMDB <strong>and</strong> LMSD for lipidomics data.</td>
</tr>
<tr>
<td><code>mode</code></td>
<td>Either <code>'metabolomics'</code> or <code>'lipidomics'</code> depending on your dataset.</td>
</tr>
<tr>
<td><code>agg_lmsd_ann</code></td>
<td>The aggregated LMSD annotations you generated using the <code>add_lmsd()</code> function, i.e. <code>lmsd_ann_list$agg_lmsd_df</code>. <strong>Only required for <code>mode</code> = <code>'lipidomics'</code></strong></td>
</tr>
</tbody>
</table>
</details>
<div class="tabbed-set tabbed-alternate" data-tabs="1:2"><input checked="checked" id="__tabbed_1_1" name="__tabbed_1" type="radio" /><input id="__tabbed_1_2" name="__tabbed_1" type="radio" /><div class="tabbed-labels"><label for="__tabbed_1_1">Metabolomics</label><label for="__tabbed_1_2">Lipidomics</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="language-r highlight"><span class="filename">Compare metabolite annotations</span><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a><span class="c1"># Prepare data.frame with alignment IDs and annotations, and filter for at least one annotation</span>
</span><span id="__span-4-2"><a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a><span class="n">anno_df_metab</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">compare_annotations_SE</span><span class="p">(</span><span class="n">metab_SE</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">metab_stool_glog</span><span class="p">,</span>
</span><span id="__span-4-3"><a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a><span class="w">                                        </span><span class="n">mode</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;metabolomics&#39;</span><span class="p">)</span>
</span></code></pre></div>
</div>
<div class="tabbed-block">
<div class="language-r highlight"><span class="filename">Compare lipid annotations</span><pre><span></span><code><span id="__span-5-1"><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a><span class="c1"># Prepare data.frame with alignment IDs and annotations, and filter for at least one annotation</span>
</span><span id="__span-5-2"><a id="__codelineno-5-2" name="__codelineno-5-2" href="#__codelineno-5-2"></a><span class="n">anno_df_lipid</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">compare_annotations_SE</span><span class="p">(</span><span class="n">metab_SE</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">lipid_stool_glog</span><span class="p">,</span>
</span><span id="__span-5-3"><a id="__codelineno-5-3" name="__codelineno-5-3" href="#__codelineno-5-3"></a><span class="w">                                        </span><span class="n">mode</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;lipidomics&#39;</span><span class="p">,</span>
</span><span id="__span-5-4"><a id="__codelineno-5-4" name="__codelineno-5-4" href="#__codelineno-5-4"></a><span class="w">                                        </span><span class="n">agg_lmsd_ann</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">lmsd_ann_list</span><span class="o">$</span><span class="n">agg_lmsd_df</span><span class="p">)</span>
</span></code></pre></div>
</div>
</div>
</div>
<h2 id="keeping-only-annotated-features">Keeping only annotated features üè∑Ô∏è‚úÖ</h2>
<p>From here, we can filter our <code>SummarizedExperiment</code> object for features with at least one annotation.
While the other features likely represent interesting metabolites and lipids, without an available annotation, they won't be interpretable downstream.</p>
<p>We can achieve this providing our <code>SummarizedExperiment</code> object to the <a href="../../assets/LCMS/LCMS_preprocessing/keep_annotated_SE.R"><code>keep_annotated_SE()</code></a> function, which will output a filtered <code>SummarizedExperiment</code> object.</p>
<details class="info">
<summary>Parameters for <code>keep_annotated_SE()</code></summary>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>metab_SE</code></td>
<td>The <code>SummarizedExperiment</code> object with secondary annotations. These should include HMDB for metabolomics data, and both HMDB <strong>and</strong> LMSD for lipidomics data.</td>
</tr>
<tr>
<td><code>mode</code></td>
<td>Either <code>'metabolomics'</code> or <code>'lipidomics'</code> depending on your dataset.</td>
</tr>
</tbody>
</table>
</details>
<div class="tabbed-set tabbed-alternate" data-tabs="2:2"><input checked="checked" id="__tabbed_2_1" name="__tabbed_2" type="radio" /><input id="__tabbed_2_2" name="__tabbed_2" type="radio" /><div class="tabbed-labels"><label for="__tabbed_2_1">Metabolomics</label><label for="__tabbed_2_2">Lipidomics</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<p>The function will create a new <code>rowData</code> element called <code>shortname</code>, and also assign this value as the preferred row name.</p>
<ul>
<li>It uses the following naming hierarchy to decide an appropriate name: HMDB &gt; MS-DIAL.</li>
</ul>
<div class="language-r highlight"><span class="filename">Remove unannotated features</span><pre><span></span><code><span id="__span-6-1"><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a><span class="c1"># Keep only annotated rows and generate shortname column</span>
</span><span id="__span-6-2"><a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a><span class="n">metab_stool_glog</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">keep_annotated</span><span class="p">(</span><span class="n">metab_SE</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">metab_stool_glog</span><span class="p">,</span>
</span><span id="__span-6-3"><a id="__codelineno-6-3" name="__codelineno-6-3" href="#__codelineno-6-3"></a><span class="w">                                   </span><span class="n">mode</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;metabolomics&#39;</span><span class="p">)</span>
</span><span id="__span-6-4"><a id="__codelineno-6-4" name="__codelineno-6-4" href="#__codelineno-6-4"></a>
</span><span id="__span-6-5"><a id="__codelineno-6-5" name="__codelineno-6-5" href="#__codelineno-6-5"></a><span class="c1"># Save the object</span>
</span><span id="__span-6-6"><a id="__codelineno-6-6" name="__codelineno-6-6" href="#__codelineno-6-6"></a><span class="nf">saveRDS</span><span class="p">(</span><span class="n">metab_stool_glog</span><span class="p">,</span><span class="w"> </span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;output&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_Preprocessing&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;metab_stool_glog_anno.rds&#39;</span><span class="p">))</span>
</span></code></pre></div>
<div class="admonition question">
<p class="admonition-title">Are the other annotations still there?</p>
<p>Yes! While the shorter names are succinct and useful for plotting, you can view the additional annotations at any time and alter as required.</p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-7-1"><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a><span class="c1"># Get HMDB and KEGG annotations</span>
</span><span id="__span-7-2"><a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a><span class="n">hmdb_annotations</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rowData</span><span class="p">(</span><span class="n">metab_stool_glog</span><span class="p">)</span><span class="o">$</span><span class="n">HMDB</span>
</span><span id="__span-7-3"><a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a><span class="n">kegg_annotations</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rowData</span><span class="p">(</span><span class="n">metab_stool_glog</span><span class="p">)</span><span class="o">$</span><span class="n">KEGG</span>
</span></code></pre></div>
</div>
</div>
<div class="tabbed-block">
<p>The function will create a new <code>rowData</code> element called <code>shortname</code>, and also assign this value as the preferred row name.</p>
<ul>
<li>It uses the following naming hierarchy to decide an appropriate name: MS-DIAL &gt; LMSD &gt; HMDB.</li>
<li>MS-DIAL has its own lipid database that is effective and, because it can also utilise MS/MS data in its annotations, is preferred here.</li>
</ul>
<div class="language-r highlight"><span class="filename">Remove unannotated features</span><pre><span></span><code><span id="__span-8-1"><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a><span class="c1"># Keep only annotated rows and generate shortname column</span>
</span><span id="__span-8-2"><a id="__codelineno-8-2" name="__codelineno-8-2" href="#__codelineno-8-2"></a><span class="n">lipid_stool_glog</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">keep_annotated</span><span class="p">(</span><span class="n">metab_SE</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">lipid_stool_glog</span><span class="p">,</span>
</span><span id="__span-8-3"><a id="__codelineno-8-3" name="__codelineno-8-3" href="#__codelineno-8-3"></a><span class="w">                                   </span><span class="n">mode</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;lipidomics&#39;</span><span class="p">)</span>
</span><span id="__span-8-4"><a id="__codelineno-8-4" name="__codelineno-8-4" href="#__codelineno-8-4"></a>
</span><span id="__span-8-5"><a id="__codelineno-8-5" name="__codelineno-8-5" href="#__codelineno-8-5"></a><span class="c1"># Save the object</span>
</span><span id="__span-8-6"><a id="__codelineno-8-6" name="__codelineno-8-6" href="#__codelineno-8-6"></a><span class="nf">saveRDS</span><span class="p">(</span><span class="n">lipid_stool_glog</span><span class="p">,</span><span class="w"> </span><span class="nf">here</span><span class="p">(</span><span class="s">&#39;output&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;01_Preprocessing&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;lipid_stool_glog_anno.rds&#39;</span><span class="p">))</span>
</span></code></pre></div>
<div class="admonition question">
<p class="admonition-title">Are the other annotations still there?</p>
<p>Yes! While the shorter names are succinct and useful for plotting, you can view the additional annotations at any time and alter as required.</p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-9-1"><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a><span class="c1"># Get HMDB and KEGG annotations</span>
</span><span id="__span-9-2"><a id="__codelineno-9-2" name="__codelineno-9-2" href="#__codelineno-9-2"></a><span class="n">hmdb_annotations</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rowData</span><span class="p">(</span><span class="n">lipid_stool_glog</span><span class="p">)</span><span class="o">$</span><span class="n">HMDB</span>
</span><span id="__span-9-3"><a id="__codelineno-9-3" name="__codelineno-9-3" href="#__codelineno-9-3"></a><span class="n">kegg_annotations</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rowData</span><span class="p">(</span><span class="n">lipid_stool_glog</span><span class="p">)</span><span class="o">$</span><span class="n">KEGG</span>
</span></code></pre></div>
</div>
</div>
</div>
</div>
<h2 id="next-steps">Next steps ‚û°Ô∏è</h2>
<p>You now have a normalised, imputed, dataset that has undergone secondary annotation and been filtered for annotated features.
It is now time to proceed to <a href="../manual-peak-curation/">manual curation of the annotated spectra</a>.</p>
<h2 id="rights">Rights</h2>
<ul>
<li>Copyright ¬©Ô∏è 2024 Mucosal Immunology lab, Monash University, Melbourne, Australia.</li>
<li><a href="https://pubmed.ncbi.nlm.nih.gov/34986597/">HMDB version 5.0</a>: Wishart DS, Guo A, Oler E, Wang F, Anjum A, Peters H, Dizon R, Sayeeda Z, Tian S, Lee BL, Berjanskii M. HMDB 5.0: the human metabolome database for 2022. <em>Nucleic acids research</em>. 2022 Jan 7;50(D1):D622-31.</li>
<li>License: This pipeline is provided under the MIT license.</li>
<li>Authors: M. Macowan and C. Pattaroni</li>
</ul>









  




                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": ["content.code.copy"], "search": "../../assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.50899def.min.js"></script>
      
    
  </body>
</html>