
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../../LCMS/manual-peak-curation/">
      
      
        <link rel="next" href="../metagenomics/">
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.16">
    
    
      
        <title>Single-cell RNAseq - Mucosal Immunology Bioinformatics Hub</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.7e37652d.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheet/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      
  


  
  

<script id="__analytics">function __md_analytics(){function e(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],e("js",new Date),e("config","G-7HN0DB92MT"),document.addEventListener("DOMContentLoaded",(function(){document.forms.search&&document.forms.search.query.addEventListener("blur",(function(){this.value&&e("event","search",{search_term:this.value})}));document$.subscribe((function(){var t=document.forms.feedback;if(void 0!==t)for(var a of t.querySelectorAll("[type=submit]"))a.addEventListener("click",(function(a){a.preventDefault();var n=document.location.pathname,d=this.getAttribute("data-md-value");e("event","feedback",{page:n,data:d}),t.firstElementChild.disabled=!0;var r=t.querySelector(".md-feedback__note [data-md-value='"+d+"']");r&&(r.hidden=!1)})),t.hidden=!1})),location$.subscribe((function(t){e("config","G-7HN0DB92MT",{page_path:t.pathname})}))}));var t=document.createElement("script");t.async=!0,t.src="https://www.googletagmanager.com/gtag/js?id=G-7HN0DB92MT",document.getElementById("__analytics").insertAdjacentElement("afterEnd",t)}</script>
  
    <script>"undefined"!=typeof __md_analytics&&__md_analytics()</script>
  

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#single-cell-rnaseq-fastq-pre-processing" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Mucosal Immunology Bioinformatics Hub" class="md-header__button md-logo" aria-label="Mucosal Immunology Bioinformatics Hub" data-md-component="logo">
      
  <img src="../../assets/mil_logo.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Mucosal Immunology Bioinformatics Hub
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Single-cell RNAseq
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://mucosal-immunology-lab.github.io/bioinformatics-hub/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    Mucosal-Immunology-Lab/bioinformatic-hub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Mucosal Immunology Bioinformatics Hub" class="md-nav__button md-logo" aria-label="Mucosal Immunology Bioinformatics Hub" data-md-component="logo">
      
  <img src="../../assets/mil_logo.png" alt="logo">

    </a>
    Mucosal Immunology Bioinformatics Hub
  </label>
  
    <div class="md-nav__source">
      <a href="https://mucosal-immunology-lab.github.io/bioinformatics-hub/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    Mucosal-Immunology-Lab/bioinformatic-hub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../RNAseq/rnaseq-nfcore/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Bulk Transcriptomics
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Microbiome
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Microbiome
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_1" >
        
          
          <label class="md-nav__link" for="__nav_3_1" id="__nav_3_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    16S rRNA amplicon
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_1">
            <span class="md-nav__icon md-icon"></span>
            16S rRNA amplicon
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Microbiome/dada2_pipeline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    DADA2 pipeline
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    LCMS Analysis
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            LCMS Analysis
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../LCMS/lcms-analysis/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../LCMS/msdial-processing/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    1. Raw Data Processing with MS-DIAL
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../LCMS/pmp-quality-control/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    2. Peak Matrix Processing
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../LCMS/secondary-feature-annotation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    3. Secondary Feature Annotation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../LCMS/manual-peak-curation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    4. Manual Peak Curation
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" checked>
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Nextflow Pipelines
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Nextflow Pipelines
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Single-cell RNAseq
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Single-cell RNAseq
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#introduction" class="md-nav__link">
    <span class="md-ellipsis">
      Introduction
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#usage" class="md-nav__link">
    <span class="md-ellipsis">
      Usage
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Usage">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#download-the-repository" class="md-nav__link">
    <span class="md-ellipsis">
      Download the repository 📁
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#create-the-conda-environment" class="md-nav__link">
    <span class="md-ellipsis">
      Create the conda environment 🐍
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#prepare-the-genome" class="md-nav__link">
    <span class="md-ellipsis">
      Prepare the genome 🧬
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#prepare-your-sample-sheet" class="md-nav__link">
    <span class="md-ellipsis">
      Prepare your sample sheet ✏️
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#running-the-pipeline" class="md-nav__link">
    <span class="md-ellipsis">
      Running the pipeline 🏃
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Running the pipeline 🏃">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#customisation" class="md-nav__link">
    <span class="md-ellipsis">
      Customisation ⚙️
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Outputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#collected-export-files" class="md-nav__link">
    <span class="md-ellipsis">
      Collected export files 📦
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#reports" class="md-nav__link">
    <span class="md-ellipsis">
      Reports 📄
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#starsolo" class="md-nav__link">
    <span class="md-ellipsis">
      STARsolo ⭐
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#rights" class="md-nav__link">
    <span class="md-ellipsis">
      Rights
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../metagenomics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Metagenomics
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../dada2_16S/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    DADA2 16S rRNA
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Cross-omic Tools
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Cross-omic Tools
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6_1" >
        
          
          <label class="md-nav__link" for="__nav_6_1" id="__nav_6_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Differential testing
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_6_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6_1">
            <span class="md-nav__icon md-icon"></span>
            Differential testing
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../CrossOmicTools/DifferentialTesting/bio_limma/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Linear modelling with bio_limma
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" >
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Utilities
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            Utilities
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7_1" >
        
          
          <label class="md-nav__link" for="__nav_7_1" id="__nav_7_1_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Handling sequencing data
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_7_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7_1">
            <span class="md-nav__icon md-icon"></span>
            Handling sequencing data
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Utilities/data-best-practices/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Data best practices
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Utilities/convert-raw-novaseq-outputs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Converting NovaSeq outputs
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Utilities/sra-data-submission/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SRA sequencing data submission
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7_2" >
        
          
          <label class="md-nav__link" for="__nav_7_2" id="__nav_7_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Monash data storage
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_7_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7_2">
            <span class="md-nav__icon md-icon"></span>
            Monash data storage
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Utilities/market-storage/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Market storage
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Utilities/vault-storage/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Vault storage
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../PublicDatasets/public-datasets/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Public Datasets
    
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#introduction" class="md-nav__link">
    <span class="md-ellipsis">
      Introduction
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#usage" class="md-nav__link">
    <span class="md-ellipsis">
      Usage
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Usage">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#download-the-repository" class="md-nav__link">
    <span class="md-ellipsis">
      Download the repository 📁
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#create-the-conda-environment" class="md-nav__link">
    <span class="md-ellipsis">
      Create the conda environment 🐍
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#prepare-the-genome" class="md-nav__link">
    <span class="md-ellipsis">
      Prepare the genome 🧬
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#prepare-your-sample-sheet" class="md-nav__link">
    <span class="md-ellipsis">
      Prepare your sample sheet ✏️
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#running-the-pipeline" class="md-nav__link">
    <span class="md-ellipsis">
      Running the pipeline 🏃
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Running the pipeline 🏃">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#customisation" class="md-nav__link">
    <span class="md-ellipsis">
      Customisation ⚙️
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#outputs" class="md-nav__link">
    <span class="md-ellipsis">
      Outputs
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Outputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#collected-export-files" class="md-nav__link">
    <span class="md-ellipsis">
      Collected export files 📦
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#reports" class="md-nav__link">
    <span class="md-ellipsis">
      Reports 📄
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#starsolo" class="md-nav__link">
    <span class="md-ellipsis">
      STARsolo ⭐
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#rights" class="md-nav__link">
    <span class="md-ellipsis">
      Rights
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  



<h1 id="single-cell-rnaseq-fastq-pre-processing">Single-cell RNAseq FASTQ pre-processing</h1>
<h2 id="introduction">Introduction</h2>
<p><a href="https://github.com/mucosal-immunology-lab/nf-mucimmuno/tree/main/scRNAseq"><strong>nf-mucimmuno/scRNAseq</strong></a> is a bioinformatics pipeline that can be used to run quality control steps and alignment to a host genome using STARsolo. It takes a samplesheet and FASTQ files as input, performs FastQC, trimming and alignment, and produces an output <code>.tar.gz</code> archive containing the collected outputs from STARsolo, ready for further processing downstream in R. MultiQC is run on the FastQC outputs both before and after TrimGalore! for visual inspection of sample quality &ndash; output <code>.html</code> files are collected in the results.</p>
<p><img align="center" alt="nf-mucimmuno/scRNAseq" src="../../assets/NextFlow/nf-mucimmuno_scRNAseq.png" /></p>
<div class="admonition info">
<p class="admonition-title">Citation</p>
<p>If you use this workflow and end up publishing something, please consider including a reference to our work! 😎🙏</p>
<p>Macowan, M., Pattaroni, C., Cardwell, B. A., &amp; Marsland, B. (2024). nf-mucimmuno/scRNAseq (0.1.0). Zenodo. <a href="https://doi.org/10.5281/zenodo.14868176">https://doi.org/10.5281/zenodo.14868176</a></p>
</div>
<h2 id="usage">Usage</h2>
<h3 id="download-the-repository">Download the repository 📁</h3>
<p>This repository contains the relevant Nextflow workflow components, including a conda environment and submodules, to run the pipeline. To retrieve this repository alone, run the <code>retrieve_me.sh</code> script above.</p>
<div class="admonition warning">
<p class="admonition-title">Git version requirements</p>
<p>Git <code>sparse-checkout</code> is required to retrieve just the <strong>nf-mucimmuno/scRNAseq</strong> pipeline. It was only introduced to Git in version 2.27.0, so ensure that the loaded version is high enough (or that there is a version loaded on the cluster at all). As of July 2024, the M3 MASSIVE cluster has version 2.38.1 available.</p>
</div>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="c1"># Check git version</span>
</span><span id="__span-0-2"><a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a>git<span class="w"> </span>--version
</span><span id="__span-0-3"><a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a>
</span><span id="__span-0-4"><a id="__codelineno-0-4" name="__codelineno-0-4" href="#__codelineno-0-4"></a><span class="c1"># Load git module if not loaded or insufficient version</span>
</span><span id="__span-0-5"><a id="__codelineno-0-5" name="__codelineno-0-5" href="#__codelineno-0-5"></a>module<span class="w"> </span>load<span class="w"> </span>git/2.38.1
</span></code></pre></div>
<p>First, create a new bash script file.</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a><span class="c1"># Create and edit a new file with nano</span>
</span><span id="__span-1-2"><a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a>nano<span class="w"> </span>retrieve_me.sh
</span></code></pre></div>
<p>Add the contents to the file, save, and close.</p>
<div class="language-bash highlight"><span class="filename">retrieve_me.sh</span><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a><span class="ch">#!/bin/bash</span>
</span><span id="__span-2-2"><a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a>
</span><span id="__span-2-3"><a id="__codelineno-2-3" name="__codelineno-2-3" href="#__codelineno-2-3"></a><span class="c1"># Define variables</span>
</span><span id="__span-2-4"><a id="__codelineno-2-4" name="__codelineno-2-4" href="#__codelineno-2-4"></a><span class="nv">REPO_URL</span><span class="o">=</span><span class="s2">&quot;https://github.com/mucosal-immunology-lab/nf-mucimmuno&quot;</span>
</span><span id="__span-2-5"><a id="__codelineno-2-5" name="__codelineno-2-5" href="#__codelineno-2-5"></a><span class="nv">REPO_DIR</span><span class="o">=</span><span class="s2">&quot;nf-mucimmuno&quot;</span>
</span><span id="__span-2-6"><a id="__codelineno-2-6" name="__codelineno-2-6" href="#__codelineno-2-6"></a><span class="nv">SUBFOLDER</span><span class="o">=</span><span class="s2">&quot;scRNAseq&quot;</span>
</span><span id="__span-2-7"><a id="__codelineno-2-7" name="__codelineno-2-7" href="#__codelineno-2-7"></a>
</span><span id="__span-2-8"><a id="__codelineno-2-8" name="__codelineno-2-8" href="#__codelineno-2-8"></a><span class="c1"># Clone the repository with sparse checkout</span>
</span><span id="__span-2-9"><a id="__codelineno-2-9" name="__codelineno-2-9" href="#__codelineno-2-9"></a>git<span class="w"> </span>clone<span class="w"> </span>--no-checkout<span class="w"> </span><span class="nv">$REPO_URL</span>
</span><span id="__span-2-10"><a id="__codelineno-2-10" name="__codelineno-2-10" href="#__codelineno-2-10"></a><span class="nb">cd</span><span class="w"> </span><span class="nv">$REPO_DIR</span>
</span><span id="__span-2-11"><a id="__codelineno-2-11" name="__codelineno-2-11" href="#__codelineno-2-11"></a>
</span><span id="__span-2-12"><a id="__codelineno-2-12" name="__codelineno-2-12" href="#__codelineno-2-12"></a><span class="c1"># Initialize sparse-checkout and set the desired subfolder</span>
</span><span id="__span-2-13"><a id="__codelineno-2-13" name="__codelineno-2-13" href="#__codelineno-2-13"></a>git<span class="w"> </span>sparse-checkout<span class="w"> </span>init<span class="w"> </span>--cone
</span><span id="__span-2-14"><a id="__codelineno-2-14" name="__codelineno-2-14" href="#__codelineno-2-14"></a>git<span class="w"> </span>sparse-checkout<span class="w"> </span><span class="nb">set</span><span class="w"> </span><span class="nv">$SUBFOLDER</span>
</span><span id="__span-2-15"><a id="__codelineno-2-15" name="__codelineno-2-15" href="#__codelineno-2-15"></a>
</span><span id="__span-2-16"><a id="__codelineno-2-16" name="__codelineno-2-16" href="#__codelineno-2-16"></a><span class="c1"># Checkout the files in the subfolder</span>
</span><span id="__span-2-17"><a id="__codelineno-2-17" name="__codelineno-2-17" href="#__codelineno-2-17"></a>git<span class="w"> </span>checkout<span class="w"> </span>main
</span><span id="__span-2-18"><a id="__codelineno-2-18" name="__codelineno-2-18" href="#__codelineno-2-18"></a>
</span><span id="__span-2-19"><a id="__codelineno-2-19" name="__codelineno-2-19" href="#__codelineno-2-19"></a><span class="c1"># Move the folder into the main folder and delete the parent</span>
</span><span id="__span-2-20"><a id="__codelineno-2-20" name="__codelineno-2-20" href="#__codelineno-2-20"></a>mv<span class="w"> </span><span class="nv">$SUBFOLDER</span><span class="w"> </span>../
</span><span id="__span-2-21"><a id="__codelineno-2-21" name="__codelineno-2-21" href="#__codelineno-2-21"></a><span class="nb">cd</span><span class="w"> </span>..
</span><span id="__span-2-22"><a id="__codelineno-2-22" name="__codelineno-2-22" href="#__codelineno-2-22"></a>rm<span class="w"> </span>-rf<span class="w"> </span><span class="nv">$REPO_DIR</span>
</span><span id="__span-2-23"><a id="__codelineno-2-23" name="__codelineno-2-23" href="#__codelineno-2-23"></a>
</span><span id="__span-2-24"><a id="__codelineno-2-24" name="__codelineno-2-24" href="#__codelineno-2-24"></a><span class="c1"># Extract the larger gzipped CLS files</span>
</span><span id="__span-2-25"><a id="__codelineno-2-25" name="__codelineno-2-25" href="#__codelineno-2-25"></a>gunzip<span class="w"> </span>-r<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$SUBFOLDER</span><span class="s2">/modules/starsolo/CLS&quot;</span>
</span><span id="__span-2-26"><a id="__codelineno-2-26" name="__codelineno-2-26" href="#__codelineno-2-26"></a>
</span><span id="__span-2-27"><a id="__codelineno-2-27" name="__codelineno-2-27" href="#__codelineno-2-27"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Subfolder &#39;</span><span class="nv">$SUBFOLDER</span><span class="s2">&#39; has been downloaded successfully.&quot;</span>
</span></code></pre></div>
<p>Then run the script to retrieve the repository into a new folder called <code>scRNAseq</code>, which will house your workflow files and results.</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a><span class="c1"># Run the script</span>
</span><span id="__span-3-2"><a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a>bash<span class="w"> </span>retrieve_me.sh
</span></code></pre></div>
<h3 id="create-the-conda-environment">Create the conda environment 🐍</h3>
<p>To create the conda environment, use the provided environment <code>.yaml</code> file. Then activate it to access required functions.</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a><span class="c1"># Create the environment</span>
</span><span id="__span-4-2"><a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a>mamba<span class="w"> </span>env<span class="w"> </span>create<span class="w"> </span>-f<span class="w"> </span>environment.yaml
</span><span id="__span-4-3"><a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a>
</span><span id="__span-4-4"><a id="__codelineno-4-4" name="__codelineno-4-4" href="#__codelineno-4-4"></a><span class="c1"># Activate the environment</span>
</span><span id="__span-4-5"><a id="__codelineno-4-5" name="__codelineno-4-5" href="#__codelineno-4-5"></a>mamba<span class="w"> </span>activate<span class="w"> </span>nextflow-scrnaseq
</span></code></pre></div>
<h3 id="prepare-the-genome">Prepare the genome 🧬</h3>
<p>Create a new folder somewhere to store your genome files. Enter the new folder, and run the relevant code depending on your host organism. Run these steps in an interactive session with ~48GB RAM and 16 cores, or submit them as an sbatch job.</p>
<div class="admonition warning">
<p class="admonition-title">Do these databases exist already?</p>
<p>Please check if these are already available somewhere before regenerating them yourself!</p>
</div>
<p>STAR should be loaded already via the conda environment for the genome indexing step. We will set <code>--sjdbOverhang</code> to 79 to be suitable for use with the longer <code>R2</code> FASTQ data resulting from BD Rhapsody single cell sequencing. This may require alteration for other platforms. <strong>Essentially, you just need to set <code>--sjdbOverhang</code> to the length of your R2 sequences minus 1.</strong></p>
<div class="tabbed-set tabbed-alternate" data-tabs="1:2"><input checked="checked" id="__tabbed_1_1" name="__tabbed_1" type="radio" /><input id="__tabbed_1_2" name="__tabbed_1" type="radio" /><div class="tabbed-labels"><label for="__tabbed_1_1">Human genome files 👨👩</label><label for="__tabbed_1_2">Mouse genome files 🐁</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="language-bash highlight"><span class="filename">01_retrieve_human_genome.sh</span><pre><span></span><code><span id="__span-5-1"><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a><span class="ch">#!/bin/bash</span>
</span><span id="__span-5-2"><a id="__codelineno-5-2" name="__codelineno-5-2" href="#__codelineno-5-2"></a><span class="nv">VERSION</span><span class="o">=</span><span class="m">111</span>
</span><span id="__span-5-3"><a id="__codelineno-5-3" name="__codelineno-5-3" href="#__codelineno-5-3"></a>wget<span class="w"> </span>-L<span class="w"> </span>ftp://ftp.ensembl.org/pub/release-<span class="nv">$VERSION</span>/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa.gz
</span><span id="__span-5-4"><a id="__codelineno-5-4" name="__codelineno-5-4" href="#__codelineno-5-4"></a>wget<span class="w"> </span>-L<span class="w"> </span>ftp://ftp.ensembl.org/pub/release-<span class="nv">$VERSION</span>/gtf/homo_sapiens/Homo_sapiens.GRCh38.<span class="nv">$VERSION</span>.gtf.gz
</span><span id="__span-5-5"><a id="__codelineno-5-5" name="__codelineno-5-5" href="#__codelineno-5-5"></a>gunzip<span class="w"> </span>*
</span></code></pre></div>
</div>
<div class="tabbed-block">
<div class="language-bash highlight"><span class="filename">01_retrieve_mouse_genome.sh</span><pre><span></span><code><span id="__span-6-1"><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a><span class="ch">#!/bin/bash</span>
</span><span id="__span-6-2"><a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a><span class="nv">VERSION</span><span class="o">=</span><span class="m">111</span>
</span><span id="__span-6-3"><a id="__codelineno-6-3" name="__codelineno-6-3" href="#__codelineno-6-3"></a>wget<span class="w"> </span>-L<span class="w"> </span>ftp://ftp.ensembl.org/pub/release-<span class="nv">$VERSION</span>/fasta/mus_musculus/dna/Mus_musculus.GRCm39.dna_sm.primary_assembly.fa.gz
</span><span id="__span-6-4"><a id="__codelineno-6-4" name="__codelineno-6-4" href="#__codelineno-6-4"></a>wget<span class="w"> </span>-L<span class="w"> </span>ftp://ftp.ensembl.org/pub/release-<span class="nv">$VERSION</span>/gtf/mus_musculus/Mus_musculus.GRCm39.<span class="nv">$VERSION</span>.gtf.gz
</span><span id="__span-6-5"><a id="__codelineno-6-5" name="__codelineno-6-5" href="#__codelineno-6-5"></a>gunzip<span class="w"> </span>*
</span></code></pre></div>
</div>
</div>
</div>
<p>Then use STAR to prepare the genome index.</p>
<div class="tabbed-set tabbed-alternate" data-tabs="2:2"><input checked="checked" id="__tabbed_2_1" name="__tabbed_2" type="radio" /><input id="__tabbed_2_2" name="__tabbed_2" type="radio" /><div class="tabbed-labels"><label for="__tabbed_2_1">Human genome files 👨👩</label><label for="__tabbed_2_2">Mouse genome files 🐁</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="language-bash highlight"><span class="filename">02_index_human_genome.sh</span><pre><span></span><code><span id="__span-7-1"><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a><span class="ch">#!/bin/bash</span>
</span><span id="__span-7-2"><a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a><span class="nv">VERSION</span><span class="o">=</span><span class="m">111</span>
</span><span id="__span-7-3"><a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a>STAR<span class="w"> </span><span class="se">\</span>
</span><span id="__span-7-4"><a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a><span class="w">    </span>--runThreadN<span class="w"> </span><span class="m">16</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-7-5"><a id="__codelineno-7-5" name="__codelineno-7-5" href="#__codelineno-7-5"></a><span class="w">    </span>--genomeDir<span class="w"> </span><span class="s2">&quot;STARgenomeIndex79/&quot;</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-7-6"><a id="__codelineno-7-6" name="__codelineno-7-6" href="#__codelineno-7-6"></a><span class="w">    </span>--runMode<span class="w"> </span>genomeGenerate<span class="w"> </span><span class="se">\</span>
</span><span id="__span-7-7"><a id="__codelineno-7-7" name="__codelineno-7-7" href="#__codelineno-7-7"></a><span class="w">    </span>--genomeFastaFiles<span class="w"> </span><span class="s2">&quot;Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa&quot;</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-7-8"><a id="__codelineno-7-8" name="__codelineno-7-8" href="#__codelineno-7-8"></a><span class="w">    </span>--sjdbGTFfile<span class="w"> </span><span class="s2">&quot;Homo_sapiens.GRCh38.</span><span class="nv">$VERSION</span><span class="s2">.gtf&quot;</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-7-9"><a id="__codelineno-7-9" name="__codelineno-7-9" href="#__codelineno-7-9"></a><span class="w">    </span>--sjdbOverhang<span class="w"> </span><span class="m">79</span>
</span></code></pre></div>
</div>
<div class="tabbed-block">
<div class="language-bash highlight"><span class="filename">02_index_mouse_genome.sh</span><pre><span></span><code><span id="__span-8-1"><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a><span class="ch">#!/bin/bash</span>
</span><span id="__span-8-2"><a id="__codelineno-8-2" name="__codelineno-8-2" href="#__codelineno-8-2"></a><span class="nv">VERSION</span><span class="o">=</span><span class="m">111</span>
</span><span id="__span-8-3"><a id="__codelineno-8-3" name="__codelineno-8-3" href="#__codelineno-8-3"></a>STAR<span class="w"> </span><span class="se">\</span>
</span><span id="__span-8-4"><a id="__codelineno-8-4" name="__codelineno-8-4" href="#__codelineno-8-4"></a><span class="w">    </span>--runThreadN<span class="w"> </span><span class="m">16</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-8-5"><a id="__codelineno-8-5" name="__codelineno-8-5" href="#__codelineno-8-5"></a><span class="w">    </span>--genomeDir<span class="w"> </span><span class="s2">&quot;STARgenomeIndex79/&quot;</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-8-6"><a id="__codelineno-8-6" name="__codelineno-8-6" href="#__codelineno-8-6"></a><span class="w">    </span>--runMode<span class="w"> </span>genomeGenerate<span class="w"> </span><span class="se">\</span>
</span><span id="__span-8-7"><a id="__codelineno-8-7" name="__codelineno-8-7" href="#__codelineno-8-7"></a><span class="w">    </span>--genomeFastaFiles<span class="w"> </span><span class="s2">&quot;Mus_musculus.GRCm39.dna_sm.primary_assembly.fa&quot;</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-8-8"><a id="__codelineno-8-8" name="__codelineno-8-8" href="#__codelineno-8-8"></a><span class="w">    </span>--sjdbGTFfile<span class="w"> </span><span class="s2">&quot;Mus_musculus.GRCm39.</span><span class="nv">$VERSION</span><span class="s2">.gtf&quot;</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-8-9"><a id="__codelineno-8-9" name="__codelineno-8-9" href="#__codelineno-8-9"></a><span class="w">    </span>--sjdbOverhang<span class="w"> </span><span class="m">79</span>
</span></code></pre></div>
</div>
</div>
</div>
<h3 id="prepare-your-sample-sheet">Prepare your sample sheet ✏️</h3>
<p>This pipeline requires a sample sheet to identify where your FASTQ files are located, and which cell label sequences (CLS) are being utilised.</p>
<p>More information about the CLS tags used with BD Rhapsody single-cell RNAseq library preparation can be found here:</p>
<ul>
<li><a href="https://www.bdbiosciences.com/content/dam/bdb/marketing-documents/products-pdf-folder/software-informatics/rhapsody-sequence-analysis-pipeline/Rhapsody-Sequence-Analysis-Pipeline-UG.pdf">BD Rhapsody Sequence Analysis Pipeline &ndash; User's Guide</a></li>
<li><a href="https://bd-rhapsody-public.s3.amazonaws.com/CellLabel/rhapsody_cell_label.py.txt">BD Rhapsody Cell Label Structure &ndash; Python Script</a></li>
</ul>
<p>More information about the CLS tags used with 10X Chromium single-cell RNAseq library preparation can be found here:</p>
<ul>
<li><a href="https://teichlab.github.io/scg_lib_structs/methods_html/10xChromium3.html">10X Chromium Single Cell 3' Solution V2 and V3 guide (Teich Lab)</a></li>
<li>10X Chromium V2 CLS sequences are 26bp long.</li>
<li>10X Chromium V3 CLS sequences are 28bp long.</li>
</ul>
<p>The benefit of providing the name of the CLS bead versions in the sample sheet is that you can combine runs that utilise different beads together in the same workflow. Keep in mind that if you do this though, there may be some bead-related batch effects to address and correct downstream &ndash; it is always important to check for these effects when combining sequencing runs in any case. The current options are:</p>
<table>
<thead>
<tr>
<th>CLS option</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>BD_Original</td>
<td>The original BD rhapsody beads and linker sequences</td>
</tr>
<tr>
<td>BD_Enhanced_V1</td>
<td>First version of enhanced beads with polyT and 5prime capture oligo types, shorter linker sequences, longer polyT, and 0-3 diversity insert bases at the beginning of the sequence</td>
</tr>
<tr>
<td>BD_Enhanced_V2</td>
<td>Same structure as the enhanced (V1) beads, but with increased CLS diversity (384 vs. 96)</td>
</tr>
<tr>
<td>10X_Chromium_V2</td>
<td>Feature a 16 bp cell barcode and a 10 bp unique molecular identifier (UMI)</td>
</tr>
<tr>
<td>10X_Chromium_V3</td>
<td>Enhanced sequencing accuracy and resolution with a 16 bp cell barcode and a 12 bp UMI</td>
</tr>
</tbody>
</table>
<p>Further, we also need to provide the path to the STAR genome index folder for each sample &ndash; while in many cases this value will remain constant, the benefit of providing this information is that you can process runs with different R2 sequence lengths at the same time. Recall from above that the genome index you use should use an <code>--sjdbOverhang</code> length that of your R2 sequences minus 1.</p>
<p>Your sample sheet should look as follows, <strong>ensuring you use the exact column names as below</strong>. </p>
<div class="admonition warning">
<p class="admonition-title">File paths on M3</p>
<p>Remember that on the M3 MASSIVE cluster, you need to use the <strong>full file path</strong> &ndash; relative file paths don't usually work.</p>
</div>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-9-1"><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a>sample,fastq_1,fastq_2,CLS,GenomeIndex
</span><span id="__span-9-2"><a id="__codelineno-9-2" name="__codelineno-9-2" href="#__codelineno-9-2"></a>CONTROL_S1,CONTROL_S1_R1.fastq.gz,CONTROL_S1_R2.fastq.gz,BD_Enhanced_V2,mf33/Databases/ensembl/human/STARgenomeIndex79
</span><span id="__span-9-3"><a id="__codelineno-9-3" name="__codelineno-9-3" href="#__codelineno-9-3"></a>CONTROL_S2,CONTROL_S2_R1.fastq.gz,CONTROL_S1_R2.fastq.gz,BD_Enhanced_V2,mf33/Databases/ensembl/human/STARgenomeIndex79
</span><span id="__span-9-4"><a id="__codelineno-9-4" name="__codelineno-9-4" href="#__codelineno-9-4"></a>TREATMENT_S1,TREATMENT_S1_R1.fastq.gz,TREATMENT_S1_R2.fastq.gz,BD_Enhanced_V2,mf33/Databases/ensembl/human/STARgenomeIndex79
</span></code></pre></div>
<p>An example is provided <a href="../../assets/NextFlow/samplesheet_test.csv">here</a>.</p>
<h3 id="running-the-pipeline">Running the pipeline 🏃</h3>
<p>Now you can run the pipeline. You will need to set up a parent job to run each of the individual jobs &ndash; this can be either an interactive session, or an sbatch job. For example:</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-10-1"><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a><span class="c1"># Start an interactive session with minimal resources</span>
</span><span id="__span-10-2"><a id="__codelineno-10-2" name="__codelineno-10-2" href="#__codelineno-10-2"></a>smux<span class="w"> </span>n<span class="w"> </span>--time<span class="o">=</span><span class="m">3</span>-00:00:00<span class="w"> </span>--mem<span class="o">=</span>16GB<span class="w"> </span>--ntasks<span class="o">=</span><span class="m">1</span><span class="w"> </span>--cpuspertask<span class="o">=</span><span class="m">2</span><span class="w"> </span>-J<span class="w"> </span>nf-STARsolo
</span></code></pre></div>
<div class="admonition warning">
<p class="admonition-title">Set the correct sample sheet location</p>
<p>Make sure you alter the <code>nextflow.config</code> file to provide the path to your sample sheet, unless it is <code>./data/samplesheet.csv</code> which is the default for the cluster profile. Stay within the top <code>cluster</code> profile section to alter settings for Slurm-submitted jobs.</p>
</div>
<p>Inside your interactive session, be sure to activate your <code>nextflow-scrnaseq</code> environment from above. Then, <strong>inside the scRNAseq folder</strong>, begin the pipeline using the following command (ensuring you use the <code>cluster</code> profile to make use of the Slurm workflow manager).</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-11-1"><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a><span class="c1"># Activate conda environment</span>
</span><span id="__span-11-2"><a id="__codelineno-11-2" name="__codelineno-11-2" href="#__codelineno-11-2"></a>mamba<span class="w"> </span>activate<span class="w"> </span>nextflow-scrnaseq
</span><span id="__span-11-3"><a id="__codelineno-11-3" name="__codelineno-11-3" href="#__codelineno-11-3"></a>
</span><span id="__span-11-4"><a id="__codelineno-11-4" name="__codelineno-11-4" href="#__codelineno-11-4"></a><span class="c1"># Begin running the pipeline</span>
</span><span id="__span-11-5"><a id="__codelineno-11-5" name="__codelineno-11-5" href="#__codelineno-11-5"></a>nextflow<span class="w"> </span>run<span class="w"> </span>process_raw_reads.nf<span class="w"> </span>-resume<span class="w"> </span>-profile<span class="w"> </span>cluster
</span></code></pre></div>
<h4 id="customisation">Customisation ⚙️</h4>
<p>There are several customisation options that are available within the <code>nextflow.config</code> file. While the defaults should be suitable for those with access to the M3 MASSIVE cluster genomics partition, for those without access, of for those who require different amounts of resources, there are ways to change these.</p>
<p>In order to work with different technologies, and accommodate for differences in cell label structure (CLS), the STAR parameters <code>--soloType</code> and <code>--soloCBmatchWLtype</code> are set in a CLS-dependent manner. This is required, because the BD Rhapsody system has a complex barcode structure. The 10X Chromium system on the other hand has a simple barcode structure with a single barcode and single UMI. Additionally, the <code>--soloCBmatchWLtype = EditDist2</code> only works with <code>--soloType = CB_UMI_Complex</code>, and therefore <code>--soloCBmatchWLtype = 1MM multi Nbase pseudocounts</code> is used for 10X Chromium runs.</p>
<ul>
<li>For BD Rhapsody sequencing: <code>--soloType = CB_UMI_Complex</code> and <code>--soloCBmatchWLtype = EditDist2</code>.</li>
<li>For 10X Chromium sequencing: <code>--soloType = CB_UMI_Simple</code> and <code>--soloCBmatchWLtype = 1MM multi Nbase pseudocounts</code>.</li>
<li>Additionally, 10X Chromium runs use <code>--clipAdapterType = CellRanger4</code>.</li>
</ul>
<p>To adjust the <code>cluster</code> profile settings, stay within the appropriate section at the top of the file.</p>
<details class="info">
<summary>Parameters</summary>
<p>Visit <a href="https://github.com/alexdobin/STAR/">STAR documentation</a> for explanations of all available options for STARsolo.</p>
<table>
<thead>
<tr>
<th>Option</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>samples_csv</td>
<td>The file path to your sample sheet</td>
</tr>
<tr>
<td>outdir</td>
<td>A new folder name to be created for your results</td>
</tr>
<tr>
<td><em>trimgalore</em>.quality</td>
<td>The minimum quality before a sequence is truncated (default: <code>20</code>)</td>
</tr>
<tr>
<td><em>trimgalore</em>.adapter</td>
<td>A custom adapter sequence for the R1 sequences (default: <code>'AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC'</code>)</td>
</tr>
<tr>
<td><em>trimgalore</em>.adapter2</td>
<td>A custom adapter sequence for the R2 sequences (default: <code>'AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT'</code>)</td>
</tr>
<tr>
<td><em>starsolo</em>.soloUMIdedup</td>
<td>The type of UMI deduplication (default: <code>'1MM_CR'</code>)</td>
</tr>
<tr>
<td><em>starsolo</em>.soloUMIfiltering</td>
<td>The type of UMI filtering for reads uniquely mapping to genes (default: <code>'MultiGeneUMI_CR'</code>)</td>
</tr>
<tr>
<td><em>starsolo</em>.soloCellFilter</td>
<td>The method type and parameters for cell filtering (default: <code>'EmptyDrops_CR'</code>)</td>
</tr>
<tr>
<td><em>starsolo</em>.soloMultiMappers</td>
<td>The counting method for reads mapping for multiple genes (default: <code>'EM'</code>)</td>
</tr>
</tbody>
</table>
</details>
<details class="info">
<summary>Process</summary>
<p>These settings relate to resource allocation and cluster settings. FASTQC and TRIMGALORE steps can take longer than 4 hours for typical single-cell RNAseq file, and therefore the default option is to run these steps on the <code>comp</code> partition.</p>
<table>
<thead>
<tr>
<th>Option</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>executor</td>
<td>The workload manager (default: <code>'slurm'</code>)</td>
</tr>
<tr>
<td>conda</td>
<td>The conda environment to use (default: <code>'./environment.yaml'</code>)</td>
</tr>
<tr>
<td>queueSize</td>
<td>The maximum number of jobs to be submitted at any time (default: <code>12</code>)</td>
</tr>
<tr>
<td>submitRateLimit</td>
<td>The rate allowed for job submission &ndash; either a number of jobs per second (e.g. 20sec) or a number of jobs per time period (e.g. 20/5min) (default: <code>'1/2sec'</code>)</td>
</tr>
<tr>
<td>memory</td>
<td>The maximum global memory allowed for Nextflow to use (default: <code>'320 GB'</code>)</td>
</tr>
<tr>
<td><em>FASTQC</em>.memory</td>
<td>Memory for FASTQC step to use (default: <code>'80 GB'</code>)</td>
</tr>
<tr>
<td><em>FASTQC</em>.cpus</td>
<td>Number of CPUs for FASTQC step to use (default: <code>8</code>)</td>
</tr>
<tr>
<td><em>FASTQC</em>.clusterOptions</td>
<td>Specific cluster options for FASTQC step (default: <code>'--time=8:00:00'</code>)</td>
</tr>
<tr>
<td><em>TRIMGALORE</em>.memory</td>
<td>Memory for TRIMGALORE step to use (default: <code>'80 GB'</code>)</td>
</tr>
<tr>
<td><em>TRIMGALORE</em>.cpus</td>
<td>Number of CPUs for TRIMGALORE step to use (default: <code>8</code>)</td>
</tr>
<tr>
<td><em>TRIMGALORE</em>.clusterOptions</td>
<td>Specific cluster options for TRIMGALORE step (default : <code>'--time=8:00:00'</code>)</td>
</tr>
<tr>
<td><em>STARSOLO</em>.memory</td>
<td>Memory for STARSOLO step to use (default: <code>'80 GB'</code>)</td>
</tr>
<tr>
<td><em>STARSOLO</em>.cpus</td>
<td>Number of CPUs for STARSOLO step to use (default: <code>12</code>)</td>
</tr>
<tr>
<td><em>STARSOLO</em>.clusterOptions</td>
<td>Specific cluster options for STARSOLO step (default : <code>'--time=4:00:00 --partition=genomics --qos=genomics'</code>)</td>
</tr>
<tr>
<td><em>COLLECT_EXPORT_FILES</em>.memory</td>
<td>Memory for COLLECT_EXPORT_FILES step to use (default: <code>'32 GB'</code>)</td>
</tr>
<tr>
<td><em>COLLECT_EXPORT_FILES</em>.cpus</td>
<td>Number of CPUs for COLLECT_EXPORT_FILES step to use (default: <code>8</code>)</td>
</tr>
<tr>
<td><em>COLLECT_EXPORT_FILES</em>.clusterOptions</td>
<td>Specific cluster options for COLLECT_EXPORT_FILES step (default : <code>'--time=4:00:00 --partition=genomics --qos=genomics'</code>)</td>
</tr>
</tbody>
</table>
</details>
<h2 id="outputs">Outputs</h2>
<p>Several outputs will be copied from their respective Nextflow <code>work</code> directories to the output folder of your choice (default: <code>results</code>).</p>
<p><strong>Alignment summary utility script</strong></p>
<p><img alt="🌈" class="emojione" src="https://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/png/1f308.png" title=":rainbow:" /> There is also a utility script in the main <code>scRNAseq</code> directory called <code>collect_alignment_summaries.sh</code>. This will navigate into each of the sample folders inside <code>results/STARsolo</code>, and retrieve some key information for you to validate that the alignment worked successfully (from the <code>GeneFull_Ex50pAS</code> subfolder). This can otherwise take quite some time to go through each folder if you have a lot of samples.</p>
<ul>
<li>After running this, a new file called <code>AlignmentSummary.txt</code> will be generated in the <code>scRNAseq</code> directory. Each sample will be listed by name, with the number of reads, percentage of reads with valid barcodes, and estimated number of cells.</li>
<li>It will be immediately obvious that something has gone wrong if you see that the percentage of reads with valid barcodes is very low (e.g. <code>0.02</code> = 2% valid barcodes) &ndash; this is usually paired with a very low estimated cell number.</li>
<li>This could indicate that you have used the wrong barcode version for your runs, and therefore the associated barcode whitelist used by the pipeline was incorrect.</li>
</ul>
<p>A successful example is shown below <img alt="✨" class="emojione" src="https://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/png/2728.png" title=":sparkles:" /></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-12-1"><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a>Sample:<span class="w"> </span>Healthy1
</span><span id="__span-12-2"><a id="__codelineno-12-2" name="__codelineno-12-2" href="#__codelineno-12-2"></a>Number<span class="w"> </span>of<span class="w"> </span>Reads,353152389
</span><span id="__span-12-3"><a id="__codelineno-12-3" name="__codelineno-12-3" href="#__codelineno-12-3"></a>Reads<span class="w"> </span>With<span class="w"> </span>Valid<span class="w"> </span>Barcodes,0.950799
</span><span id="__span-12-4"><a id="__codelineno-12-4" name="__codelineno-12-4" href="#__codelineno-12-4"></a>Estimated<span class="w"> </span>Number<span class="w"> </span>of<span class="w"> </span>Cells,6623
</span><span id="__span-12-5"><a id="__codelineno-12-5" name="__codelineno-12-5" href="#__codelineno-12-5"></a>
</span><span id="__span-12-6"><a id="__codelineno-12-6" name="__codelineno-12-6" href="#__codelineno-12-6"></a>Sample:<span class="w"> </span>Healthy2
</span><span id="__span-12-7"><a id="__codelineno-12-7" name="__codelineno-12-7" href="#__codelineno-12-7"></a>Number<span class="w"> </span>of<span class="w"> </span>Reads,344989615
</span><span id="__span-12-8"><a id="__codelineno-12-8" name="__codelineno-12-8" href="#__codelineno-12-8"></a>Reads<span class="w"> </span>With<span class="w"> </span>Valid<span class="w"> </span>Barcodes,0.948577
</span><span id="__span-12-9"><a id="__codelineno-12-9" name="__codelineno-12-9" href="#__codelineno-12-9"></a>Estimated<span class="w"> </span>Number<span class="w"> </span>of<span class="w"> </span>Cells,6631
</span><span id="__span-12-10"><a id="__codelineno-12-10" name="__codelineno-12-10" href="#__codelineno-12-10"></a><span class="c1"># etc...</span>
</span></code></pre></div>
<h3 id="collected-export-files">Collected export files 📦</h3>
<p>The main output will be a single archive file called <code>export_files.tar.gz</code> that you will take for further downstream pre-processing. It contains STARsolo outputs for each sample, with the respective subfolders described below.</p>
<h3 id="reports">Reports 📄</h3>
<p>Within the <code>reports</code> folder, you will find the MultiQC outputs from pre- and post-trimming.</p>
<h3 id="starsolo">STARsolo ⭐</h3>
<p>Contains the outputs for each sample from STARsolo, including various log files and package version information.</p>
<p>The main output of interest here is a folder called <code>{sample}.Solo.out</code>, which houses subfolders called <code>Gene</code>, <code>GeneFull_Ex50pAS</code>, and <code>Velocyto</code>. It is this main folder for each sample that is added to <code>export_files.tar.gz</code>.
* As you will use the gene count data from <code>GeneFull_Ex50pAS</code> downstream, it is a good idea to check the <code>Summary.csv</code> within this folder for each sample to ensure mapping was successful (or use the utility script above).
  * One of the key values to inspect is <code>Reads With Valid Barcodes</code>, which should be &gt;0.8 (indicating at least 80% of reads had valid barcodes).
  * If you note that this value is closer to 0.02 (i.e. ~2% had valid barcodes), you should double-check to make sure you specified the correct BD Rhapsody beads version. For instance, if you specified <code>BD_Enhanced_V1</code> but actually required <code>BD_Enhanced_V2</code>, the majority of your reads will not match the whitelist, and therefore the reads will be considered invalid.</p>
<p><strong>Folder structure</strong></p>
<p>Below is an example of the output structure for running one sample. The STARsolo folder would contain additional samples as required.</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-13-1"><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a>scRNAseq
</span><span id="__span-13-2"><a id="__codelineno-13-2" name="__codelineno-13-2" href="#__codelineno-13-2"></a>└──<span class="w"> </span>results/
</span><span id="__span-13-3"><a id="__codelineno-13-3" name="__codelineno-13-3" href="#__codelineno-13-3"></a><span class="w">    </span>├──<span class="w"> </span>export_files.tar.gz
</span><span id="__span-13-4"><a id="__codelineno-13-4" name="__codelineno-13-4" href="#__codelineno-13-4"></a><span class="w">    </span>├──<span class="w"> </span>reports/
</span><span id="__span-13-5"><a id="__codelineno-13-5" name="__codelineno-13-5" href="#__codelineno-13-5"></a><span class="w">    </span>│<span class="w">   </span>├──<span class="w"> </span>pretrim_multiqc_report.html
</span><span id="__span-13-6"><a id="__codelineno-13-6" name="__codelineno-13-6" href="#__codelineno-13-6"></a><span class="w">    </span>│<span class="w">   </span>└──<span class="w"> </span>posttrim_multiqc_report.html
</span><span id="__span-13-7"><a id="__codelineno-13-7" name="__codelineno-13-7" href="#__codelineno-13-7"></a><span class="w">    </span>└──<span class="w"> </span>STARsolo/
</span><span id="__span-13-8"><a id="__codelineno-13-8" name="__codelineno-13-8" href="#__codelineno-13-8"></a><span class="w">        </span>└──<span class="w"> </span>sample1/
</span><span id="__span-13-9"><a id="__codelineno-13-9" name="__codelineno-13-9" href="#__codelineno-13-9"></a><span class="w">            </span>├──<span class="w"> </span>sample1.Solo.out/
</span><span id="__span-13-10"><a id="__codelineno-13-10" name="__codelineno-13-10" href="#__codelineno-13-10"></a><span class="w">            </span>│<span class="w">   </span>├──<span class="w"> </span>Gene/
</span><span id="__span-13-11"><a id="__codelineno-13-11" name="__codelineno-13-11" href="#__codelineno-13-11"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>filtered/
</span><span id="__span-13-12"><a id="__codelineno-13-12" name="__codelineno-13-12" href="#__codelineno-13-12"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>barcodes.tsv.gz
</span><span id="__span-13-13"><a id="__codelineno-13-13" name="__codelineno-13-13" href="#__codelineno-13-13"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>features.tsv.gz
</span><span id="__span-13-14"><a id="__codelineno-13-14" name="__codelineno-13-14" href="#__codelineno-13-14"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>│<span class="w">   </span>└──<span class="w"> </span>matrix.mtx.gz
</span><span id="__span-13-15"><a id="__codelineno-13-15" name="__codelineno-13-15" href="#__codelineno-13-15"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>raw/
</span><span id="__span-13-16"><a id="__codelineno-13-16" name="__codelineno-13-16" href="#__codelineno-13-16"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>barcodes.tsv.gz
</span><span id="__span-13-17"><a id="__codelineno-13-17" name="__codelineno-13-17" href="#__codelineno-13-17"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>features.tsv.gz
</span><span id="__span-13-18"><a id="__codelineno-13-18" name="__codelineno-13-18" href="#__codelineno-13-18"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>matrix.mtx.gz
</span><span id="__span-13-19"><a id="__codelineno-13-19" name="__codelineno-13-19" href="#__codelineno-13-19"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>│<span class="w">   </span>└──<span class="w"> </span>UniqueAndMult-EM.mtx.gz
</span><span id="__span-13-20"><a id="__codelineno-13-20" name="__codelineno-13-20" href="#__codelineno-13-20"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>Features.stats
</span><span id="__span-13-21"><a id="__codelineno-13-21" name="__codelineno-13-21" href="#__codelineno-13-21"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>Summary.csv
</span><span id="__span-13-22"><a id="__codelineno-13-22" name="__codelineno-13-22" href="#__codelineno-13-22"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>└──<span class="w"> </span>UMIperCellSorted.txt
</span><span id="__span-13-23"><a id="__codelineno-13-23" name="__codelineno-13-23" href="#__codelineno-13-23"></a><span class="w">            </span>│<span class="w">   </span>├──<span class="w"> </span>GeneFull_Ex50pAS/
</span><span id="__span-13-24"><a id="__codelineno-13-24" name="__codelineno-13-24" href="#__codelineno-13-24"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>filtered/
</span><span id="__span-13-25"><a id="__codelineno-13-25" name="__codelineno-13-25" href="#__codelineno-13-25"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>barcodes.tsv.gz
</span><span id="__span-13-26"><a id="__codelineno-13-26" name="__codelineno-13-26" href="#__codelineno-13-26"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>features.tsv.gz
</span><span id="__span-13-27"><a id="__codelineno-13-27" name="__codelineno-13-27" href="#__codelineno-13-27"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>│<span class="w">   </span>└──<span class="w"> </span>matrix.mtx.gz
</span><span id="__span-13-28"><a id="__codelineno-13-28" name="__codelineno-13-28" href="#__codelineno-13-28"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>raw/
</span><span id="__span-13-29"><a id="__codelineno-13-29" name="__codelineno-13-29" href="#__codelineno-13-29"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>barcodes.tsv.gz
</span><span id="__span-13-30"><a id="__codelineno-13-30" name="__codelineno-13-30" href="#__codelineno-13-30"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>features.tsv.gz
</span><span id="__span-13-31"><a id="__codelineno-13-31" name="__codelineno-13-31" href="#__codelineno-13-31"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>matrix.mtx.gz
</span><span id="__span-13-32"><a id="__codelineno-13-32" name="__codelineno-13-32" href="#__codelineno-13-32"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>│<span class="w">   </span>└──<span class="w"> </span>UniqueAndMult-EM.mtx.gz
</span><span id="__span-13-33"><a id="__codelineno-13-33" name="__codelineno-13-33" href="#__codelineno-13-33"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>Features.stats
</span><span id="__span-13-34"><a id="__codelineno-13-34" name="__codelineno-13-34" href="#__codelineno-13-34"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>Summary.csv
</span><span id="__span-13-35"><a id="__codelineno-13-35" name="__codelineno-13-35" href="#__codelineno-13-35"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>└──<span class="w"> </span>UMIperCellSorted.txt
</span><span id="__span-13-36"><a id="__codelineno-13-36" name="__codelineno-13-36" href="#__codelineno-13-36"></a><span class="w">            </span>│<span class="w">   </span>├──<span class="w"> </span>Velocyto/
</span><span id="__span-13-37"><a id="__codelineno-13-37" name="__codelineno-13-37" href="#__codelineno-13-37"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>filtered/
</span><span id="__span-13-38"><a id="__codelineno-13-38" name="__codelineno-13-38" href="#__codelineno-13-38"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>ambiguous.mtx.gz
</span><span id="__span-13-39"><a id="__codelineno-13-39" name="__codelineno-13-39" href="#__codelineno-13-39"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>barcodes.tsv.gz
</span><span id="__span-13-40"><a id="__codelineno-13-40" name="__codelineno-13-40" href="#__codelineno-13-40"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>features.tsv.gz
</span><span id="__span-13-41"><a id="__codelineno-13-41" name="__codelineno-13-41" href="#__codelineno-13-41"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>spliced.mtx.gz
</span><span id="__span-13-42"><a id="__codelineno-13-42" name="__codelineno-13-42" href="#__codelineno-13-42"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>│<span class="w">   </span>└──<span class="w"> </span>unspliced.mtx.gz
</span><span id="__span-13-43"><a id="__codelineno-13-43" name="__codelineno-13-43" href="#__codelineno-13-43"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>raw/
</span><span id="__span-13-44"><a id="__codelineno-13-44" name="__codelineno-13-44" href="#__codelineno-13-44"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>ambiguous.mtx.gz
</span><span id="__span-13-45"><a id="__codelineno-13-45" name="__codelineno-13-45" href="#__codelineno-13-45"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>barcodes.tsv.gz
</span><span id="__span-13-46"><a id="__codelineno-13-46" name="__codelineno-13-46" href="#__codelineno-13-46"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>features.tsv.gz
</span><span id="__span-13-47"><a id="__codelineno-13-47" name="__codelineno-13-47" href="#__codelineno-13-47"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>spliced.mtx.gz
</span><span id="__span-13-48"><a id="__codelineno-13-48" name="__codelineno-13-48" href="#__codelineno-13-48"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>│<span class="w">   </span>└──<span class="w"> </span>unspliced.mtx.gz
</span><span id="__span-13-49"><a id="__codelineno-13-49" name="__codelineno-13-49" href="#__codelineno-13-49"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>├──<span class="w"> </span>Features.stats
</span><span id="__span-13-50"><a id="__codelineno-13-50" name="__codelineno-13-50" href="#__codelineno-13-50"></a><span class="w">            </span>│<span class="w">   </span>│<span class="w">   </span>└──<span class="w"> </span>Summary.csv
</span><span id="__span-13-51"><a id="__codelineno-13-51" name="__codelineno-13-51" href="#__codelineno-13-51"></a><span class="w">            </span>│<span class="w">   </span>└──<span class="w"> </span>Barcodes.stats
</span><span id="__span-13-52"><a id="__codelineno-13-52" name="__codelineno-13-52" href="#__codelineno-13-52"></a><span class="w">            </span>├──<span class="w"> </span>sample1.Log.final.out
</span><span id="__span-13-53"><a id="__codelineno-13-53" name="__codelineno-13-53" href="#__codelineno-13-53"></a><span class="w">            </span>├──<span class="w"> </span>sample1.Log.out
</span><span id="__span-13-54"><a id="__codelineno-13-54" name="__codelineno-13-54" href="#__codelineno-13-54"></a><span class="w">            </span>├──<span class="w"> </span>sample1.Log.progress.out
</span><span id="__span-13-55"><a id="__codelineno-13-55" name="__codelineno-13-55" href="#__codelineno-13-55"></a><span class="w">            </span>└──<span class="w"> </span>versions.yml
</span></code></pre></div>
<h2 id="rights">Rights</h2>
<ul>
<li>Copyright ©️ 2024 Mucosal Immunology Lab, Monash University, Melbourne, Australia.</li>
<li>Licence: This pipeline is provided under the MIT license.</li>
<li>Authors: M. Macowan</li>
</ul>
<div class="admonition info">
<ul>
<li>FastQC: <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">Babraham Bioinformatics</a></li>
<li>MultiQC: <a href="https://github.com/MultiQC/MultiQC">National Genomics Infrastructure (SciLifeLab, Sweden), Seqera Labs et al.</a></li>
<li>TrimGalore!: <a href="https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/">Babraham Bioinformatics</a></li>
<li>STARsolo: <a href="https://github.com/alexdobin/STAR/tree/master">Alex Dobin et al.</a></li>
</ul>
</div>









  




                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": ["content.code.copy"], "search": "../../assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.50899def.min.js"></script>
      
    
  </body>
</html>